// frosted-ui/dist/cjs/components/accordion/accordion.stories.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Multiple = exports.Single = void 0;
const tslib_1 = require("tslib");
const react_1 = tslib_1.__importDefault(require("react"));
const __1 = require("..");
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Layout/Accordion',
    component: __1.Accordion.Root,
    args: {},
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
};
exports.default = meta;
// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
exports.Single = {
    args: {
        defaultValue: 'item-1',
        type: 'single',
        collapsible: false,
    },
    render: (args) => (react_1.default.createElement("div", null,
        react_1.default.createElement(__1.Accordion.Root
        // collapsible
        , { 
            // collapsible
            style: { width: 600 }, ...args },
            react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-4)' } },
                react_1.default.createElement(__1.Accordion.Item, { value: "item-1" },
                    react_1.default.createElement(__1.Accordion.Trigger, null, "Is it accessible?"),
                    react_1.default.createElement(__1.Accordion.Content, null, "Yes. It adheres to the WAI-ARIA design pattern.")),
                react_1.default.createElement(__1.Accordion.Item, { value: "item-2" },
                    react_1.default.createElement(__1.Accordion.Trigger, null, "Is it unstyled?"),
                    react_1.default.createElement(__1.Accordion.Content, null, "Yes. It's unstyled by default, giving you freedom over the look and feel.")),
                react_1.default.createElement(__1.Accordion.Item, { value: "item-3" },
                    react_1.default.createElement(__1.Accordion.Trigger, null, "Can it be animated?"),
                    react_1.default.createElement(__1.Accordion.Content, null, "Yes! You can animate the Accordion with CSS or JavaScript.")))))),
};
exports.Multiple = {
    args: {
        defaultValue: ['item-1', 'item-2'],
        type: 'multiple',
    },
    render: (args) => (react_1.default.createElement("div", null,
        react_1.default.createElement(__1.Accordion.Root, { style: { width: 600 }, ...args },
            react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-4)' } },
                react_1.default.createElement(__1.Accordion.Item, { value: "item-1" },
                    react_1.default.createElement(__1.Accordion.Trigger, null, "Is it accessible?"),
                    react_1.default.createElement(__1.Accordion.Content, null, "Yes. It adheres to the WAI-ARIA design pattern.")),
                react_1.default.createElement(__1.Accordion.Item, { value: "item-2" },
                    react_1.default.createElement(__1.Accordion.Trigger, null, "Is it unstyled?"),
                    react_1.default.createElement(__1.Accordion.Content, null, "Yes. It's unstyled by default, giving you freedom over the look and feel.")),
                react_1.default.createElement(__1.Accordion.Item, { value: "item-3" },
                    react_1.default.createElement(__1.Accordion.Trigger, null, "Can it be animated?"),
                    react_1.default.createElement(__1.Accordion.Content, null, "Yes! You can animate the Accordion with CSS or JavaScript.")))))),
};
