// frosted-ui/dist/cjs/components/scroll-area/scroll-area.stories.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Scrollbars = exports.Size = exports.Default = void 0;
const tslib_1 = require("tslib");
const react_1 = tslib_1.__importDefault(require("react"));
const __1 = require("..");
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Components/ScrollArea',
    component: __1.ScrollArea,
    args: {
        type: 'always',
    },
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
};
exports.default = meta;
// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
exports.Default = {
    args: {
        size: __1.scrollAreaPropDefs.size.default,
        scrollbars: 'vertical',
    },
    render: (args) => (react_1.default.createElement(__1.ScrollArea, { type: "always", scrollbars: "vertical", style: { height: 180 }, ...args },
        react_1.default.createElement("div", { style: { padding: '8px 8px 54px 8px' } },
            react_1.default.createElement(__1.Heading, { size: "4", style: { marginBottom: 8 }, trim: "start" }, "Principles of the typographic craft"),
            react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-4)' } },
                react_1.default.createElement(__1.Text, { as: "p" }, "Three fundamental aspects of typography are legibility, readability, and aesthetics. Although in a non-technical sense \"legible\" and \"readable\" are often used synonymously, typographically they are separate but related concepts."),
                react_1.default.createElement(__1.Text, { as: "p" }, "Legibility describes how easily individual characters can be distinguished from one another. It is described by Walter Tracy as \"the quality of being decipherable and recognisable\". For instance, if a \"b\" and an \"h\", or a \"3\" and an \"8\", are difficult to distinguish at small sizes, this is a problem of legibility."),
                react_1.default.createElement(__1.Text, { as: "p" }, "Typographers are concerned with legibility insofar as it is their job to select the correct font to use. Brush Script is an example of a font containing many characters that might be difficult to distinguish. The selection of cases influences the legibility of typography because using only uppercase letters (all-caps) reduces legibility."))))),
};
exports.Size = {
    args: {
        size: __1.scrollAreaPropDefs.size.default,
        scrollbars: 'vertical',
    },
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-2)' } },
        react_1.default.createElement(__1.ScrollArea, { ...args, size: "1", type: "always", scrollbars: "horizontal", style: { width: 300, height: 12 } },
            react_1.default.createElement("div", { style: { width: 800, height: 1 } })),
        react_1.default.createElement(__1.ScrollArea, { ...args, size: "2", type: "always", scrollbars: "horizontal", style: { width: 350, height: 16 } },
            react_1.default.createElement("div", { style: { width: 900, height: 1 } })),
        react_1.default.createElement(__1.ScrollArea, { ...args, size: "3", type: "always", scrollbars: "horizontal", style: { width: 400, height: 20 } },
            react_1.default.createElement("div", { style: { width: 1000, height: 1 } })))),
};
exports.Scrollbars = {
    args: {
        size: __1.scrollAreaPropDefs.size.default,
        type: 'always',
        scrollbars: 'vertical',
    },
    render: (args) => (react_1.default.createElement("div", { style: { display: 'grid', gridTemplateColumns: 'repeat(2, 1fr)', gap: 'var(--space-2)' } },
        react_1.default.createElement(__1.ScrollArea, { ...args, type: "always", scrollbars: "vertical", style: { height: 150 } },
            react_1.default.createElement("div", { style: {
                    display: 'flex',
                    flexDirection: 'column',
                    gap: 'var(--space-4)',
                    padding: 'var(--space-2)',
                    paddingRight: 'var(--space-8)',
                } },
                react_1.default.createElement(__1.Text, { size: "2", trim: "both" }, "Three fundamental aspects of typography are legibility, readability, and aesthetics. Although in a non-technical sense \"legible\" and \"readable\" are often used synonymously, typographically they are separate but related concepts."),
                react_1.default.createElement(__1.Text, { size: "2", trim: "both" }, "Legibility describes how easily individual characters can be distinguished from one another. It is described by Walter Tracy as \"the quality of being decipherable and recognisable\". For instance, if a \"b\" and an \"h\", or a \"3\" and an \"8\", are difficult to distinguish at small sizes, this is a problem of legibility."))),
        react_1.default.createElement(__1.ScrollArea, { ...args, type: "always", scrollbars: "horizontal", style: { height: 150 } },
            react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-4)', padding: 'var(--space-2)', width: 700 } },
                react_1.default.createElement(__1.Text, { size: "2", trim: "both" }, "Three fundamental aspects of typography are legibility, readability, and aesthetics. Although in a non-technical sense \"legible\" and \"readable\" are often used synonymously, typographically they are separate but related concepts."),
                react_1.default.createElement(__1.Text, { size: "2", trim: "both" }, "Legibility describes how easily individual characters can be distinguished from one another. It is described by Walter Tracy as \"the quality of being decipherable and recognisable\". For instance, if a \"b\" and an \"h\", or a \"3\" and an \"8\", are difficult to distinguish at small sizes, this is a problem of legibility."))))),
};
