// frosted-ui/dist/esm/components/kbd/kbd.stories.js
import React from 'react';
import { Kbd, kbdPropDefs } from '../../../src/components/';
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Typography/Kbd',
    component: Kbd,
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
};
export default meta;
// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
export const Default = {
    args: {
        children: 'Shift + Tab',
        size: kbdPropDefs.size.default,
    },
};
export const Size = {
    render: (args) => (React.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-3)' } },
        React.createElement("div", null,
            React.createElement(Kbd, { ...args, size: "1" }, "Shift + Tab")),
        React.createElement("div", null,
            React.createElement(Kbd, { ...args, size: "2" }, "Shift + Tab")),
        React.createElement("div", null,
            React.createElement(Kbd, { ...args, size: "3" }, "Shift + Tab")),
        React.createElement("div", null,
            React.createElement(Kbd, { ...args, size: "4" }, "Shift + Tab")),
        React.createElement("div", null,
            React.createElement(Kbd, { ...args, size: "5" }, "Shift + Tab")),
        React.createElement("div", null,
            React.createElement(Kbd, { ...args, size: "6" }, "Shift + Tab")),
        React.createElement("div", null,
            React.createElement(Kbd, { ...args, size: "7" }, "Shift + Tab")),
        React.createElement("div", null,
            React.createElement(Kbd, { ...args, size: "8" }, "Shift + Tab")),
        React.createElement("div", null,
            React.createElement(Kbd, { ...args, size: "9" }, "Shift + Tab")))),
};
