// frosted-ui/dist/esm/components/slider/slider.stories.js
import React from 'react';
import { Slider, sliderPropDefs } from '..';
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Controls/Slider',
    component: Slider,
    args: {
        size: sliderPropDefs.size.default,
        color: sliderPropDefs.color.default,
        highContrast: sliderPropDefs.highContrast.default,
        disabled: false,
    },
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
};
export default meta;
// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
export const Default = {
    render: (args) => (React.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-4)', width: 300 } },
        React.createElement(Slider, { ...args, defaultValue: [50] }))),
};
export const Size = {
    render: (args) => (React.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-4)', width: 300 } },
        React.createElement(Slider, { ...args, defaultValue: [25], size: "1" }),
        React.createElement(Slider, { ...args, defaultValue: [50], size: "2" }),
        React.createElement(Slider, { ...args, defaultValue: [75], size: "3" }))),
};
export const Color = {
    render: (args) => (React.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-4)', width: 300 } },
        React.createElement(Slider, { ...args, defaultValue: [20], color: "indigo" }),
        React.createElement(Slider, { ...args, defaultValue: [40], color: "cyan" }),
        React.createElement(Slider, { ...args, defaultValue: [60], color: "orange" }),
        React.createElement(Slider, { ...args, defaultValue: [80], color: "crimson" }))),
};
export const HighContrast = {
    name: 'High Contrast',
    render: (args) => (React.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-4)', width: 300 } },
        React.createElement(Slider, { ...args, defaultValue: [25], highContrast: false }),
        React.createElement(Slider, { ...args, defaultValue: [50], highContrast: true }))),
};
export const Range = {
    render: (args) => (React.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-4)', width: 300 } },
        React.createElement(Slider, { ...args, defaultValue: [25, 75] }))),
};
