// frosted-ui/dist/cjs/components/badge/badge.stories.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.HighContrast = exports.SemanticColor = exports.Color = exports.Variant = exports.Size = exports.Default = void 0;
const tslib_1 = require("tslib");
const react_1 = tslib_1.__importDefault(require("react"));
const components_1 = require("../../../src/components");
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Components/Badge',
    component: components_1.Badge,
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
};
exports.default = meta;
// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
exports.Default = {
    args: {
        children: 'Badge',
        size: components_1.badgePropDefs.size.default,
        variant: components_1.badgePropDefs.variant.default,
        color: components_1.badgePropDefs.color.default,
    },
};
exports.Size = {
    args: {
        children: 'Badge',
        size: components_1.badgePropDefs.size.default,
        variant: components_1.badgePropDefs.variant.default,
        color: components_1.badgePropDefs.color.default,
    },
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', alignItems: 'center', gap: 'var(--space-2)' } },
        react_1.default.createElement(components_1.Badge, { ...args, size: "2" }),
        react_1.default.createElement(components_1.Badge, { ...args, size: "1" }))),
};
exports.Variant = {
    args: {
        children: 'Badge',
        size: components_1.badgePropDefs.size.default,
        color: components_1.badgePropDefs.color.default,
    },
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', alignItems: 'center', gap: 'var(--space-2)' } },
        react_1.default.createElement(components_1.Badge, { ...args, variant: "solid" }),
        react_1.default.createElement(components_1.Badge, { ...args, variant: "soft" }),
        react_1.default.createElement(components_1.Badge, { ...args, variant: "outline" }),
        react_1.default.createElement(components_1.Badge, { ...args, variant: "surface" }))),
};
exports.Color = {
    args: {
        children: 'Badge',
        size: components_1.badgePropDefs.size.default,
    },
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', alignItems: 'center', gap: 'var(--space-2)' } },
        react_1.default.createElement(components_1.Badge, { ...args, color: "indigo" }),
        react_1.default.createElement(components_1.Badge, { ...args, color: "cyan" }),
        react_1.default.createElement(components_1.Badge, { ...args, color: "orange" }),
        react_1.default.createElement(components_1.Badge, { ...args, color: "crimson" }))),
};
exports.SemanticColor = {
    name: 'Semantic color',
    args: {
        size: components_1.badgePropDefs.size.default,
    },
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', alignItems: 'center', gap: 'var(--space-2)' } },
        react_1.default.createElement(components_1.Badge, { ...args, color: "info" }, "Info"),
        react_1.default.createElement(components_1.Badge, { ...args, color: "success" }, "Success"),
        react_1.default.createElement(components_1.Badge, { ...args, color: "warning" }, "Warning"),
        react_1.default.createElement(components_1.Badge, { ...args, color: "danger" }, "Danger"))),
};
exports.HighContrast = {
    name: 'High Contrast',
    args: {
        children: 'Badge',
    },
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-2)' } },
        react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-2)' } },
            react_1.default.createElement(components_1.Badge, { ...args, highContrast: false, variant: "solid" }),
            react_1.default.createElement(components_1.Badge, { ...args, highContrast: false, variant: "soft" }),
            react_1.default.createElement(components_1.Badge, { ...args, highContrast: false, variant: "outline" }),
            react_1.default.createElement(components_1.Badge, { ...args, highContrast: false, variant: "surface" })),
        react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-2)' } },
            react_1.default.createElement(components_1.Badge, { ...args, variant: "solid", highContrast: true }),
            react_1.default.createElement(components_1.Badge, { ...args, variant: "soft", highContrast: true }),
            react_1.default.createElement(components_1.Badge, { ...args, variant: "outline", highContrast: true }),
            react_1.default.createElement(components_1.Badge, { ...args, variant: "surface", highContrast: true })))),
};
