// frosted-ui/dist/cjs/components/tabs/tabs.stories.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Default = void 0;
const tslib_1 = require("tslib");
const react_1 = tslib_1.__importDefault(require("react"));
const __1 = require("..");
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Components/Tabs',
    component: __1.Tabs.List,
    args: {
        size: __1.tabsListPropDefs.size.default,
    },
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
};
exports.default = meta;
// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
// TODO: Tabs content jumping in size due to font-weight change
// This doesn't happen in Frosted UI (might be due to a font in use?)
exports.Default = {
    render: (args) => (react_1.default.createElement("div", { style: { width: 600 } },
        react_1.default.createElement(__1.Tabs.Root, { defaultValue: "account" },
            react_1.default.createElement(__1.Tabs.List, { ...args },
                react_1.default.createElement(__1.Tabs.Trigger, { value: "account" }, "Account"),
                react_1.default.createElement(__1.Tabs.Trigger, { value: "documents" }, "Documents"),
                react_1.default.createElement(__1.Tabs.Trigger, { value: "settings" }, "Settings")),
            react_1.default.createElement("div", { style: { padding: '12px 16px 8px 16px' } },
                react_1.default.createElement(__1.Tabs.Content, { value: "account" },
                    react_1.default.createElement(__1.Text, { size: "2" }, "Make changes to your account.")),
                react_1.default.createElement(__1.Tabs.Content, { value: "documents" },
                    react_1.default.createElement(__1.Text, { size: "2" }, "Access and update your documents.")),
                react_1.default.createElement(__1.Tabs.Content, { value: "settings" },
                    react_1.default.createElement(__1.Text, { size: "2" }, "Edit your profile or update contact information.")))))),
};
