// frosted-ui/dist/cjs/components/dropdown-menu/dropdown-menu.stories.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Color = exports.Size = exports.Default = void 0;
const tslib_1 = require("tslib");
const react_1 = tslib_1.__importDefault(require("react"));
const __1 = require("..");
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Controls/DropdownMenu',
    component: __1.DropdownMenu.Content,
    args: {},
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
};
exports.default = meta;
exports.Default = {
    args: {
        size: __1.dropdownMenuContentPropDefs.size.default,
        variant: __1.dropdownMenuContentPropDefs.variant.default,
    },
    render: (args) => {
        const [order, setOrder] = react_1.default.useState('ascending');
        const [showHiddenFiles, setShowHiddenFiles] = react_1.default.useState(true);
        return (react_1.default.createElement(__1.DropdownMenu.Root, null,
            react_1.default.createElement(__1.DropdownMenu.Trigger, null,
                react_1.default.createElement(__1.Button, { variant: "soft" }, "Options")),
            react_1.default.createElement(__1.DropdownMenu.Content, { ...args },
                react_1.default.createElement(__1.DropdownMenu.Label, null, "Swag"),
                react_1.default.createElement(__1.DropdownMenu.Item, { shortcut: "\u2318 E" }, "Edit"),
                react_1.default.createElement(__1.DropdownMenu.Item, { shortcut: "\u2318 D" }, "Duplicate"),
                react_1.default.createElement(__1.DropdownMenu.Separator, null),
                react_1.default.createElement(__1.DropdownMenu.Item, { shortcut: "\u2318 N" }, "Archive"),
                react_1.default.createElement(__1.DropdownMenu.Sub, null,
                    react_1.default.createElement(__1.DropdownMenu.SubTrigger, null, "More"),
                    react_1.default.createElement(__1.DropdownMenu.SubContent, null,
                        react_1.default.createElement(__1.DropdownMenu.Item, null, "Move to project\u2026"),
                        react_1.default.createElement(__1.DropdownMenu.Item, null, "Move to folder\u2026"),
                        react_1.default.createElement(__1.DropdownMenu.Separator, null),
                        react_1.default.createElement(__1.DropdownMenu.Item, null, "Advanced options\u2026"))),
                react_1.default.createElement(__1.DropdownMenu.Separator, null),
                react_1.default.createElement(__1.DropdownMenu.RadioGroup, { value: order, onValueChange: (value) => setOrder(value) },
                    react_1.default.createElement(__1.DropdownMenu.RadioItem, { value: "ascending" }, "Ascending"),
                    react_1.default.createElement(__1.DropdownMenu.RadioItem, { value: "descending" }, "Descending")),
                react_1.default.createElement(__1.DropdownMenu.Separator, null),
                react_1.default.createElement(__1.DropdownMenu.CheckboxItem, { checked: showHiddenFiles, onCheckedChange: setShowHiddenFiles, shortcut: "S+ H" }, "Show hidden files"),
                react_1.default.createElement(__1.DropdownMenu.Separator, null),
                react_1.default.createElement(__1.DropdownMenu.Item, { shortcut: "\u2318 \u232B", color: "danger" }, "Delete"))));
    },
};
exports.Size = {
    args: {
        variant: __1.dropdownMenuContentPropDefs.variant.default,
    },
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-3)', alignItems: 'center' } },
        react_1.default.createElement(__1.DropdownMenu.Root, null,
            react_1.default.createElement(__1.DropdownMenu.Trigger, null,
                react_1.default.createElement(__1.Button, { variant: "soft", size: "3" }, "Large")),
            react_1.default.createElement(__1.DropdownMenu.Content, { ...args, size: "3" },
                react_1.default.createElement(__1.DropdownMenu.Item, { shortcut: "\u2318 E" }, "Edit"),
                react_1.default.createElement(__1.DropdownMenu.Item, { shortcut: "\u2318 D" }, "Duplicate"),
                react_1.default.createElement(__1.DropdownMenu.Separator, null),
                react_1.default.createElement(__1.DropdownMenu.Item, { shortcut: "\u2318 N" }, "Archive"),
                react_1.default.createElement(__1.DropdownMenu.Separator, null),
                react_1.default.createElement(__1.DropdownMenu.Item, { shortcut: "\u2318 \u232B", color: "danger" }, "Delete"))),
        react_1.default.createElement(__1.DropdownMenu.Root, null,
            react_1.default.createElement(__1.DropdownMenu.Trigger, null,
                react_1.default.createElement(__1.Button, { variant: "soft", size: "2" }, "Default")),
            react_1.default.createElement(__1.DropdownMenu.Content, { ...args, size: "2" },
                react_1.default.createElement(__1.DropdownMenu.Item, { shortcut: "\u2318 E" }, "Edit"),
                react_1.default.createElement(__1.DropdownMenu.Item, { shortcut: "\u2318 D" }, "Duplicate"),
                react_1.default.createElement(__1.DropdownMenu.Separator, null),
                react_1.default.createElement(__1.DropdownMenu.Item, { shortcut: "\u2318 N" }, "Archive"),
                react_1.default.createElement(__1.DropdownMenu.Separator, null),
                react_1.default.createElement(__1.DropdownMenu.Item, { shortcut: "\u2318 \u232B", color: "danger" }, "Delete"))),
        react_1.default.createElement(__1.DropdownMenu.Root, null,
            react_1.default.createElement(__1.DropdownMenu.Trigger, null,
                react_1.default.createElement(__1.Button, { variant: "soft", size: "1" }, "Small")),
            react_1.default.createElement(__1.DropdownMenu.Content, { ...args, size: "1" },
                react_1.default.createElement(__1.DropdownMenu.Item, { shortcut: "\u2318 E" }, "Edit"),
                react_1.default.createElement(__1.DropdownMenu.Item, { shortcut: "\u2318 D" }, "Duplicate"),
                react_1.default.createElement(__1.DropdownMenu.Separator, null),
                react_1.default.createElement(__1.DropdownMenu.Item, { shortcut: "\u2318 N" }, "Archive"),
                react_1.default.createElement(__1.DropdownMenu.Separator, null),
                react_1.default.createElement(__1.DropdownMenu.Item, { shortcut: "\u2318 \u232B", color: "danger" }, "Delete"))))),
};
exports.Color = {
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-3)' } },
        react_1.default.createElement(__1.DropdownMenu.Root, null,
            react_1.default.createElement(__1.DropdownMenu.Trigger, null,
                react_1.default.createElement(__1.Button, { variant: "soft", color: "gray" }, "Options")),
            react_1.default.createElement(__1.DropdownMenu.Content, { ...args },
                react_1.default.createElement(__1.DropdownMenu.Item, { shortcut: "\u2318 E", color: "info" }, "Edit"),
                react_1.default.createElement(__1.DropdownMenu.Item, { shortcut: "\u2318 D", color: "success" }, "Duplicate"),
                react_1.default.createElement(__1.DropdownMenu.Separator, null),
                react_1.default.createElement(__1.DropdownMenu.Item, { shortcut: "\u2318 N", color: "danger" }, "Archive"))))),
};
