// frosted-ui/dist/esm/components/select/select.stories.js
import React from 'react';
import { Select, selectTriggerPropDefs } from '..';
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Controls/Select',
    component: Select.Trigger,
    args: {},
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
};
export default meta;
export const Default = {
    args: {
        variant: selectTriggerPropDefs.variant.default,
        color: selectTriggerPropDefs.color.default,
    },
    render: (args) => (React.createElement(Select.Root, { defaultValue: "apple" },
        React.createElement(Select.Trigger, { ...args }),
        React.createElement(Select.Content, null,
            React.createElement(Select.Group, null,
                React.createElement(Select.Label, null, "Fruits"),
                React.createElement(Select.Item, { value: "orange" }, "Orange"),
                React.createElement(Select.Item, { value: "apple" }, "Apple"),
                React.createElement(Select.Item, { value: "grape", disabled: true }, "Grape")),
            React.createElement(Select.Separator, null),
            React.createElement(Select.Group, null,
                React.createElement(Select.Label, null, "Vegetables"),
                React.createElement(Select.Item, { value: "carrot" }, "Carrot"),
                React.createElement(Select.Item, { value: "potato" }, "Potato"))))),
};
export const Size = {
    args: {
        variant: selectTriggerPropDefs.variant.default,
        color: selectTriggerPropDefs.color.default,
    },
    render: (args) => (React.createElement("div", { style: { display: 'flex', gap: 'var(--space-3)', alignItems: 'center' } },
        React.createElement(Select.Root, { size: "1", defaultValue: "apple" },
            React.createElement(Select.Trigger, { ...args }),
            React.createElement(Select.Content, null,
                React.createElement(Select.Item, { value: "apple" }, "Apple"),
                React.createElement(Select.Item, { value: "orange" }, "Orange"))),
        React.createElement(Select.Root, { size: "2", defaultValue: "apple" },
            React.createElement(Select.Trigger, { ...args }),
            React.createElement(Select.Content, null,
                React.createElement(Select.Item, { value: "apple" }, "Apple"),
                React.createElement(Select.Item, { value: "orange" }, "Orange"))),
        React.createElement(Select.Root, { size: "3", defaultValue: "apple" },
            React.createElement(Select.Trigger, { ...args }),
            React.createElement(Select.Content, null,
                React.createElement(Select.Item, { value: "apple" }, "Apple"),
                React.createElement(Select.Item, { value: "orange" }, "Orange"))))),
};
export const TriggerVariant = {
    args: {
        variant: selectTriggerPropDefs.variant.default,
        color: selectTriggerPropDefs.color.default,
    },
    render: (args) => (React.createElement("div", { style: { display: 'flex', gap: 'var(--space-3)', alignItems: 'center', flexDirection: 'column' } },
        React.createElement("div", { style: { display: 'flex', gap: 'var(--space-3)', alignItems: 'center' } },
            React.createElement(Select.Root, { defaultValue: "apple" },
                React.createElement(Select.Trigger, { ...args, variant: "surface" }),
                React.createElement(Select.Content, null,
                    React.createElement(Select.Item, { value: "apple" }, "Apple"),
                    React.createElement(Select.Item, { value: "orange" }, "Orange"))),
            React.createElement(Select.Root, { defaultValue: "apple" },
                React.createElement(Select.Trigger, { ...args, variant: "soft" }),
                React.createElement(Select.Content, null,
                    React.createElement(Select.Item, { value: "apple" }, "Apple"),
                    React.createElement(Select.Item, { value: "orange" }, "Orange"))),
            React.createElement(Select.Root, { defaultValue: "apple" },
                React.createElement(Select.Trigger, { ...args, variant: "ghost" }),
                React.createElement(Select.Content, null,
                    React.createElement(Select.Item, { value: "apple" }, "Apple"),
                    React.createElement(Select.Item, { value: "orange" }, "Orange")))),
        React.createElement("div", { style: { display: 'flex', gap: 'var(--space-3)', alignItems: 'center' } },
            React.createElement(Select.Root, { defaultValue: "apple", disabled: true },
                React.createElement(Select.Trigger, { ...args, variant: "surface" }),
                React.createElement(Select.Content, null,
                    React.createElement(Select.Item, { value: "apple" }, "Apple"),
                    React.createElement(Select.Item, { value: "orange" }, "Orange"))),
            React.createElement(Select.Root, { defaultValue: "apple", disabled: true },
                React.createElement(Select.Trigger, { ...args, variant: "soft" }),
                React.createElement(Select.Content, null,
                    React.createElement(Select.Item, { value: "apple" }, "Apple"),
                    React.createElement(Select.Item, { value: "orange" }, "Orange"))),
            React.createElement(Select.Root, { defaultValue: "apple", disabled: true },
                React.createElement(Select.Trigger, { ...args, variant: "ghost" }),
                React.createElement(Select.Content, null,
                    React.createElement(Select.Item, { value: "apple" }, "Apple"),
                    React.createElement(Select.Item, { value: "orange" }, "Orange")))))),
};
export const Color = {
    args: {
        variant: selectTriggerPropDefs.variant.default,
        color: selectTriggerPropDefs.color.default,
    },
    render: (args) => (React.createElement("div", { style: { display: 'flex', gap: 'var(--space-3)' } },
        React.createElement(Select.Root, { defaultValue: "apple" },
            React.createElement(Select.Trigger, { ...args, color: "indigo" }),
            React.createElement(Select.Content, null,
                React.createElement(Select.Item, { value: "apple" }, "Apple"),
                React.createElement(Select.Item, { value: "orange" }, "Orange"))),
        React.createElement(Select.Root, { defaultValue: "apple" },
            React.createElement(Select.Trigger, { ...args, color: "cyan" }),
            React.createElement(Select.Content, null,
                React.createElement(Select.Item, { value: "apple" }, "Apple"),
                React.createElement(Select.Item, { value: "orange" }, "Orange"))),
        React.createElement(Select.Root, { defaultValue: "apple" },
            React.createElement(Select.Trigger, { ...args, color: "orange" }),
            React.createElement(Select.Content, null,
                React.createElement(Select.Item, { value: "apple" }, "Apple"),
                React.createElement(Select.Item, { value: "orange" }, "Orange"))),
        React.createElement(Select.Root, { defaultValue: "apple" },
            React.createElement(Select.Trigger, { ...args, color: "crimson" }),
            React.createElement(Select.Content, null,
                React.createElement(Select.Item, { value: "apple" }, "Apple"),
                React.createElement(Select.Item, { value: "orange" }, "Orange"))))),
};
export const Placeholder = {
    args: {
        variant: selectTriggerPropDefs.variant.default,
        color: selectTriggerPropDefs.color.default,
    },
    render: (args) => (React.createElement(Select.Root, null,
        React.createElement(Select.Trigger, { ...args, placeholder: "Pick a fruit" }),
        React.createElement(Select.Content, null,
            React.createElement(Select.Group, null,
                React.createElement(Select.Label, null, "Fruits"),
                React.createElement(Select.Item, { value: "orange" }, "Orange"),
                React.createElement(Select.Item, { value: "apple" }, "Apple"),
                React.createElement(Select.Item, { value: "grape", disabled: true }, "Grape")),
            React.createElement(Select.Separator, null),
            React.createElement(Select.Group, null,
                React.createElement(Select.Label, null, "Vegetables"),
                React.createElement(Select.Item, { value: "carrot" }, "Carrot"),
                React.createElement(Select.Item, { value: "potato" }, "Potato"))))),
};
export const HighContrast = {
    name: 'High Contrast',
    args: {
        variant: selectTriggerPropDefs.variant.default,
        color: selectTriggerPropDefs.color.default,
    },
    render: (args) => (React.createElement("div", { style: { display: 'flex', gap: 'var(--space-3)' } },
        React.createElement(Select.Root, { defaultValue: "apple" },
            React.createElement(Select.Trigger, { ...args }),
            React.createElement(Select.Content, null,
                React.createElement(Select.Item, { value: "apple" }, "Apple"),
                React.createElement(Select.Item, { value: "orange" }, "Orange"))),
        React.createElement(Select.Root, { defaultValue: "apple" },
            React.createElement(Select.Trigger, { ...args }),
            React.createElement(Select.Content, { highContrast: true },
                React.createElement(Select.Item, { value: "apple" }, "Apple"),
                React.createElement(Select.Item, { value: "orange" }, "Orange"))))),
};
export const AlternativePositioning = {
    name: 'Alternative positioning',
    args: {
        variant: selectTriggerPropDefs.variant.default,
        color: selectTriggerPropDefs.color.default,
    },
    render: (args) => (React.createElement(Select.Root, { defaultValue: "apple" },
        React.createElement(Select.Trigger, { ...args }),
        React.createElement(Select.Content, { position: "popper" },
            React.createElement(Select.Item, { value: "apple" }, "Apple"),
            React.createElement(Select.Item, { value: "orange" }, "Orange")))),
};
