// frosted-ui/dist/esm/components/select/select.js
'use client';
import classNames from 'classnames';
import { ScrollArea as ScrollAreaPrimitive, Select as SelectPrimitive } from 'radix-ui';
import * as React from 'react';
import { ThickCheckIcon } from '../../icons';
import { Theme } from '../../theme';
import { selectContentPropDefs, selectRootPropDefs, selectTriggerPropDefs } from './select.props';
const SelectContext = React.createContext({});
const SelectRoot = (props) => {
    const { children, size = selectRootPropDefs.size.default, ...rootProps } = props;
    return (React.createElement(SelectPrimitive.Root, { ...rootProps },
        React.createElement(SelectContext.Provider, { value: React.useMemo(() => ({ size }), [size]) }, children)));
};
SelectRoot.displayName = 'SelectRoot';
const SelectTrigger = (props) => {
    const { className, variant = selectTriggerPropDefs.variant.default, color = selectTriggerPropDefs.color.default, placeholder, ...triggerProps } = props;
    const { size } = React.useContext(SelectContext);
    return (React.createElement(SelectPrimitive.Trigger, { asChild: true },
        React.createElement("button", { "data-accent-color": color, ...triggerProps, className: classNames('fui-reset', 'fui-SelectTrigger', className, `fui-r-size-${size}`, `fui-variant-${variant}`) },
            React.createElement("span", { className: "fui-SelectTriggerInner" },
                React.createElement(SelectPrimitive.Value, { placeholder: placeholder })),
            React.createElement(SelectPrimitive.Icon, { asChild: true },
                React.createElement("svg", { className: "fui-SelectIcon", xmlns: "http://www.w3.org/2000/svg", width: "16", height: "16", viewBox: "3.25 5.25 9.5 5.5", fill: "none" },
                    React.createElement("path", { d: "M4 6L8 10L12 6", stroke: "currentColor", strokeWidth: "1.5", strokeLinecap: "round", strokeLinejoin: "round" }))))));
};
SelectTrigger.displayName = 'SelectTrigger';
const SelectContent = (props) => {
    const { className, children, highContrast = selectContentPropDefs.highContrast.default, container, ...contentProps } = props;
    const { size } = React.useContext(SelectContext);
    return (React.createElement(SelectPrimitive.Portal, { container: container },
        React.createElement(Theme, { asChild: true },
            React.createElement(SelectPrimitive.Content, { sideOffset: 4, ...contentProps, className: classNames({ 'fui-PopperContent': contentProps.position === 'popper' }, 'fui-SelectContent', className, `fui-r-size-${size}`, { 'fui-high-contrast': highContrast }) },
                React.createElement(ScrollAreaPrimitive.Root, { type: "auto", className: "fui-ScrollAreaRoot" },
                    React.createElement(SelectPrimitive.Viewport, { asChild: true, className: "fui-SelectViewport" },
                        React.createElement(ScrollAreaPrimitive.Viewport, { className: "fui-ScrollAreaViewport", style: { overflowY: undefined } }, children)),
                    React.createElement(ScrollAreaPrimitive.Scrollbar, { className: "fui-ScrollAreaScrollbar fui-r-size-1", orientation: "vertical" },
                        React.createElement(ScrollAreaPrimitive.Thumb, { className: "fui-ScrollAreaThumb" })))))));
};
SelectContent.displayName = 'SelectContent';
const SelectItem = (props) => {
    const { className, children, ...itemProps } = props;
    return (React.createElement(SelectPrimitive.Item, { ...itemProps, className: classNames('fui-SelectItem', className) },
        React.createElement(SelectPrimitive.ItemIndicator, { className: "fui-SelectItemIndicator" },
            React.createElement(ThickCheckIcon, { className: "fui-SelectItemIndicatorIcon" })),
        React.createElement(SelectPrimitive.ItemText, null, children)));
};
SelectItem.displayName = 'SelectItem';
const SelectGroup = (props) => (React.createElement(SelectPrimitive.Group, { ...props, className: classNames('fui-SelectGroup', props.className) }));
SelectGroup.displayName = 'SelectGroup';
const SelectLabel = (props) => (React.createElement(SelectPrimitive.Label, { ...props, className: classNames('fui-SelectLabel', props.className) }));
SelectLabel.displayName = 'SelectLabel';
const SelectSeparator = (props) => (React.createElement(SelectPrimitive.Separator, { ...props, className: classNames('fui-SelectSeparator', props.className) }));
SelectSeparator.displayName = 'SelectSeparator';
export { SelectContent as Content, SelectGroup as Group, SelectItem as Item, SelectLabel as Label, SelectRoot as Root, SelectSeparator as Separator, SelectTrigger as Trigger, };
