// frosted-ui/dist/esm/components/context-menu/context-menu.stories.js
import React from 'react';
import { ContextMenu, contextMenuContentPropDefs } from '..';
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Controls/ContextMenu',
    component: ContextMenu.Content,
    args: {
        variant: contextMenuContentPropDefs.variant.default,
    },
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'centered',
    },
    render: (args) => (React.createElement(ContextMenu.Root, null,
        React.createElement(ContextMenu.Trigger, null,
            React.createElement("div", { style: {
                    background: 'var(--gray-a2)',
                    borderRadius: 'var(--radius-3)',
                    width: 200,
                    height: 200,
                    display: 'flex',
                    alignItems: 'center',
                    justifyContent: 'center',
                } }, "Right-click here")),
        React.createElement(ContextMenu.Content, { ...args },
            React.createElement(ContextMenu.Item, { shortcut: "\u2318 E" }, "Edit"),
            React.createElement(ContextMenu.Item, { shortcut: "\u2318 D" }, "Duplicate"),
            React.createElement(ContextMenu.Separator, null),
            React.createElement(ContextMenu.Item, { shortcut: "\u2318 N" }, "Archive"),
            React.createElement(ContextMenu.Sub, null,
                React.createElement(ContextMenu.SubTrigger, null, "More"),
                React.createElement(ContextMenu.SubContent, null,
                    React.createElement(ContextMenu.Item, null, "Move to project\u2026"),
                    React.createElement(ContextMenu.Item, null, "Move to folder\u2026"),
                    React.createElement(ContextMenu.Separator, null),
                    React.createElement(ContextMenu.Item, null, "Advanced options\u2026"))),
            React.createElement(ContextMenu.Separator, null),
            React.createElement(ContextMenu.Item, null, "Share"),
            React.createElement(ContextMenu.Item, null, "Add to favorites"),
            React.createElement(ContextMenu.Separator, null),
            React.createElement(ContextMenu.Item, { shortcut: "\u2318 \u232B", color: "danger" }, "Delete")))),
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
};
export default meta;
// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
export const Default = {};
