// frosted-ui/dist/cjs/components/tabs-nav/tabs-nav.stories.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Default = void 0;
const tslib_1 = require("tslib");
const react_1 = tslib_1.__importDefault(require("react"));
const __1 = require("..");
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Components/TabsNav',
    component: __1.TabsNav.Root,
    args: {
        size: __1.tabsNavPropDefs.size.default,
    },
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
};
exports.default = meta;
// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
exports.Default = {
    render: (args) => (react_1.default.createElement("div", { style: { width: 600 } },
        react_1.default.createElement(__1.TabsNav.Root, { ...args },
            react_1.default.createElement(__1.TabsNav.Link, { active: true, href: "#" }, "Account"),
            react_1.default.createElement(__1.TabsNav.Link, { href: "#" }, "Documents"),
            react_1.default.createElement(__1.TabsNav.Link, { href: "#" }, "Settings")))),
};
