// frosted-ui/dist/cjs/components/dialog/dialog.stories.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.InsetContent = exports.Sizes = exports.Default = void 0;
const tslib_1 = require("tslib");
const react_1 = tslib_1.__importDefault(require("react"));
const __1 = require("..");
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Components/Dialog',
    component: __1.Dialog.Content,
    args: {
        size: __1.dialogContentPropDefs.size.default,
    },
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
};
exports.default = meta;
// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
exports.Default = {
    render: (args) => (react_1.default.createElement(__1.Dialog.Root, null,
        react_1.default.createElement(__1.Dialog.Trigger, null,
            react_1.default.createElement(__1.Button, null, "Edit profile")),
        react_1.default.createElement(__1.Dialog.Content, { style: { maxWidth: 450 }, ...args },
            react_1.default.createElement(__1.Dialog.Title, null, "Edit profile"),
            react_1.default.createElement(__1.Dialog.Description, null, "Make changes to your profile."),
            react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-3)' } },
                react_1.default.createElement("label", null,
                    react_1.default.createElement(__1.Text, { as: "div", size: "2", style: { marginBottom: 4 }, weight: "bold" }, "Name"),
                    react_1.default.createElement(__1.TextField.Input, { defaultValue: "Freja Johnsen", placeholder: "Enter your full name" })),
                react_1.default.createElement("label", null,
                    react_1.default.createElement(__1.Text, { as: "div", size: "2", style: { marginBottom: 4 }, weight: "bold" }, "Email"),
                    react_1.default.createElement(__1.TextField.Input, { defaultValue: "freja@example.com", placeholder: "Enter your email" }))),
            react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-3)', marginTop: 'var(--space-4)', justifyContent: 'flex-end' } },
                react_1.default.createElement(__1.Dialog.Close, null,
                    react_1.default.createElement(__1.Button, { variant: "soft", color: "gray", onClick: () => alert('Cancel') }, "Cancel")),
                react_1.default.createElement(__1.Dialog.Close, null,
                    react_1.default.createElement(__1.Button, { onClick: () => alert('Save') }, "Save")))))),
};
exports.Sizes = {
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-4)' } },
        react_1.default.createElement(__1.Dialog.Root, null,
            react_1.default.createElement(__1.Dialog.Trigger, null,
                react_1.default.createElement(__1.Button, null, "Size 1")),
            react_1.default.createElement(__1.Dialog.Content, { style: { maxWidth: 350 }, ...args, size: "1" },
                react_1.default.createElement(__1.Dialog.Title, null, "Edit profile"),
                react_1.default.createElement(__1.Dialog.Description, null, "Make changes to your profile."),
                react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-2)', justifyContent: 'flex-end' } },
                    react_1.default.createElement(__1.Dialog.Close, null,
                        react_1.default.createElement(__1.Button, { size: "1", variant: "soft", color: "gray", onClick: () => alert('Cancel') }, "Cancel")),
                    react_1.default.createElement(__1.Dialog.Close, null,
                        react_1.default.createElement(__1.Button, { size: "1", onClick: () => alert('Save'), variant: "classic" }, "Save"))))),
        react_1.default.createElement(__1.Dialog.Root, null,
            react_1.default.createElement(__1.Dialog.Trigger, null,
                react_1.default.createElement(__1.Button, null, "Size 2")),
            react_1.default.createElement(__1.Dialog.Content, { style: { maxWidth: 350 }, ...args, size: "2" },
                react_1.default.createElement(__1.Dialog.Title, null, "Edit profile"),
                react_1.default.createElement(__1.Dialog.Description, null, "Make changes to your profile."),
                react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-2)', justifyContent: 'flex-end' } },
                    react_1.default.createElement(__1.Dialog.Close, null,
                        react_1.default.createElement(__1.Button, { size: "2", variant: "soft", color: "gray", onClick: () => alert('Cancel') }, "Cancel")),
                    react_1.default.createElement(__1.Dialog.Close, null,
                        react_1.default.createElement(__1.Button, { size: "2", onClick: () => alert('Save'), variant: "classic" }, "Save"))))),
        react_1.default.createElement(__1.Dialog.Root, null,
            react_1.default.createElement(__1.Dialog.Trigger, null,
                react_1.default.createElement(__1.Button, null, "Size 3")),
            react_1.default.createElement(__1.Dialog.Content, { style: { maxWidth: 350 }, ...args, size: "3" },
                react_1.default.createElement(__1.Dialog.Title, null, "Edit profile"),
                react_1.default.createElement(__1.Dialog.Description, null, "Make changes to your profile."),
                react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-3)', justifyContent: 'flex-end' } },
                    react_1.default.createElement(__1.Dialog.Close, null,
                        react_1.default.createElement(__1.Button, { size: "2", variant: "soft", color: "gray", onClick: () => alert('Cancel') }, "Cancel")),
                    react_1.default.createElement(__1.Dialog.Close, null,
                        react_1.default.createElement(__1.Button, { size: "2", onClick: () => alert('Save'), variant: "classic" }, "Save"))))),
        react_1.default.createElement(__1.Dialog.Root, null,
            react_1.default.createElement(__1.Dialog.Trigger, null,
                react_1.default.createElement(__1.Button, null, "Size 4")),
            react_1.default.createElement(__1.Dialog.Content, { style: { maxWidth: 350 }, ...args, size: "4" },
                react_1.default.createElement(__1.Dialog.Title, null, "Edit profile"),
                react_1.default.createElement(__1.Dialog.Description, null, "Make changes to your profile."),
                react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-3)', justifyContent: 'flex-end', alignItems: 'center' } },
                    react_1.default.createElement(__1.Dialog.Close, null,
                        react_1.default.createElement(__1.Button, { size: "3", variant: "soft", color: "gray", onClick: () => alert('Cancel') }, "Cancel")),
                    react_1.default.createElement(__1.Dialog.Close, null,
                        react_1.default.createElement(__1.Button, { size: "3", onClick: () => alert('Save'), variant: "classic" }, "Save"))))))),
};
exports.InsetContent = {
    name: 'With inset content',
    render: (args) => (react_1.default.createElement(__1.Dialog.Root, null,
        react_1.default.createElement(__1.Dialog.Trigger, null,
            react_1.default.createElement(__1.Button, null, "View users")),
        react_1.default.createElement(__1.Dialog.Content, { ...args },
            react_1.default.createElement(__1.Dialog.Title, null, "Users"),
            react_1.default.createElement(__1.Dialog.Description, null, "The following users have access to this project."),
            react_1.default.createElement(__1.Inset, { side: "x", style: { marginTop: 20 } },
                react_1.default.createElement(__1.Table.Root, { variant: "ghost", size: "1" },
                    react_1.default.createElement(__1.ScrollArea, { scrollbars: "horizontal" },
                        react_1.default.createElement(__1.Table.Table, null,
                            react_1.default.createElement(__1.Table.Header, null,
                                react_1.default.createElement(__1.Table.Row, null,
                                    react_1.default.createElement(__1.Table.ColumnHeaderCell, null,
                                        react_1.default.createElement("div", { style: { display: 'flex', alignItems: 'center', gap: 'var(--space-3)' } },
                                            react_1.default.createElement(__1.Checkbox, null),
                                            "Full name")),
                                    react_1.default.createElement(__1.Table.ColumnHeaderCell, null, "Email"),
                                    react_1.default.createElement(__1.Table.ColumnHeaderCell, null, "Group"))),
                            react_1.default.createElement(__1.Table.Body, null,
                                react_1.default.createElement(__1.Table.Row, null,
                                    react_1.default.createElement(__1.Table.RowHeaderCell, null,
                                        react_1.default.createElement("div", { style: { display: 'flex', alignItems: 'center', gap: 'var(--space-3)' } },
                                            react_1.default.createElement(__1.Checkbox, null),
                                            "Danilo Sousa")),
                                    react_1.default.createElement(__1.Table.Cell, null, "danilo@example.com"),
                                    react_1.default.createElement(__1.Table.Cell, null,
                                        react_1.default.createElement(__1.Badge, { color: "green" }, "Developer"))),
                                react_1.default.createElement(__1.Table.Row, null,
                                    react_1.default.createElement(__1.Table.RowHeaderCell, null,
                                        react_1.default.createElement("div", { style: { display: 'flex', alignItems: 'center', gap: 'var(--space-3)' } },
                                            react_1.default.createElement(__1.Checkbox, null),
                                            "Zahra Ambessa")),
                                    react_1.default.createElement(__1.Table.Cell, null, "zahra@example.com"),
                                    react_1.default.createElement(__1.Table.Cell, null,
                                        react_1.default.createElement(__1.Badge, { color: "amber" }, "Admin"))),
                                react_1.default.createElement(__1.Table.Row, null,
                                    react_1.default.createElement(__1.Table.RowHeaderCell, null,
                                        react_1.default.createElement("div", { style: { display: 'flex', alignItems: 'center', gap: 'var(--space-3)' } },
                                            react_1.default.createElement(__1.Checkbox, null),
                                            "Jasper Eriksson")),
                                    react_1.default.createElement(__1.Table.Cell, null, "jasper@example.com"),
                                    react_1.default.createElement(__1.Table.Cell, null,
                                        react_1.default.createElement(__1.Badge, { color: "green" }, "Developer"))))))))))),
};
