// frosted-ui/dist/esm/components/heading/heading.stories.js
import React from 'react';
import { Heading, headingPropDefs } from '../../../src/components/heading';
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Typography/Heading',
    component: Heading,
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
};
export default meta;
// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
export const Default = {
    args: {
        children: 'The quick brown fox jumps over the lazy dog.',
        size: headingPropDefs.size.default,
    },
};
export const Size = {
    render: (args) => (React.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-3)' } },
        React.createElement(Heading, { ...args, size: "0" }, "The quick brown fox jumps over the lazy dog."),
        React.createElement(Heading, { ...args, size: "1" }, "The quick brown fox jumps over the lazy dog."),
        React.createElement(Heading, { ...args, size: "2" }, "The quick brown fox jumps over the lazy dog."),
        React.createElement(Heading, { ...args, size: "3" }, "The quick brown fox jumps over the lazy dog."),
        React.createElement(Heading, { ...args, size: "4" }, "The quick brown fox jumps over the lazy dog."),
        React.createElement(Heading, { ...args, size: "5" }, "The quick brown fox jumps over the lazy dog."),
        React.createElement(Heading, { ...args, size: "6" }, "The quick brown fox jumps over the lazy dog."),
        React.createElement(Heading, { ...args, size: "7" }, "The quick brown fox jumps over the lazy dog."),
        React.createElement(Heading, { ...args, size: "8" }, "The quick brown fox jumps over the lazy dog."),
        React.createElement(Heading, { ...args, size: "9" }, "The quick brown fox jumps over the lazy dog."))),
};
export const Color = {
    render: (args) => (React.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-3)' } },
        React.createElement(Heading, { ...args, color: "indigo" }, "The quick brown fox jumps over the lazy dog."),
        React.createElement(Heading, { ...args, color: "cyan" }, "The quick brown fox jumps over the lazy dog."),
        React.createElement(Heading, { ...args, color: "orange" }, "The quick brown fox jumps over the lazy dog."),
        React.createElement(Heading, { ...args, color: "crimson" }, "The quick brown fox jumps over the lazy dog."))),
};
export const Align = {
    render: (args) => (React.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-3)', width: 500 } },
        React.createElement(Heading, { ...args, align: "left" }, "Left-aligned"),
        React.createElement(Heading, { ...args, align: "center" }, "Center-aligned"),
        React.createElement(Heading, { ...args, align: "right" }, "Right-aligned"))),
};
export const Trim = {
    render: (args) => (React.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-3)' } },
        React.createElement(Heading, { ...args, trim: "normal", style: {
                background: 'var(--gray-a2)',
                borderTop: '1px dashed var(--gray-a7)',
                borderBottom: '1px dashed var(--gray-a7)',
            } }, "Without trim"),
        React.createElement(Heading, { ...args, trim: "both", style: {
                background: 'var(--gray-a2)',
                borderTop: '1px dashed var(--gray-a7)',
                borderBottom: '1px dashed var(--gray-a7)',
            } }, "With trim"))),
};
export const HighContrast = {
    name: 'High Contrast',
    render: (args) => (React.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-3)' } },
        React.createElement(Heading, { ...args, highContrast: true, color: "indigo" }, "The quick brown fox jumps over the lazy dog."),
        React.createElement(Heading, { ...args, highContrast: true, color: "cyan" }, "The quick brown fox jumps over the lazy dog."),
        React.createElement(Heading, { ...args, highContrast: true, color: "orange" }, "The quick brown fox jumps over the lazy dog."),
        React.createElement(Heading, { ...args, highContrast: true, color: "crimson" }, "The quick brown fox jumps over the lazy dog."))),
};
