// frosted-ui/dist/esm/components/checkbox/checkbox.stories.js
import React from 'react';
import { Checkbox, Code, Text, checkboxPropDefs } from '..';
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Controls/Checkbox',
    component: Checkbox,
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
};
export default meta;
// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
export const Default = {
    args: {
        size: checkboxPropDefs.size.default,
        color: checkboxPropDefs.color.default,
        highContrast: checkboxPropDefs.highContrast.default,
    },
    render: (args) => (React.createElement("div", { style: { display: 'flex', gap: 'var(--space-2)', flexDirection: 'column' } },
        React.createElement(Checkbox, { checked: true, ...args }, "Checked"),
        React.createElement(Checkbox, { checked: true, disabled: true, ...args }, "Checked disabled"),
        React.createElement(Checkbox, { checked: false, ...args }, "Unchecked"),
        React.createElement(Checkbox, { checked: false, disabled: true, ...args }, "Unchecked disabled"),
        React.createElement(Checkbox, { checked: 'indeterminate', ...args }, "Indeterminate"),
        React.createElement(Checkbox, { checked: 'indeterminate', disabled: true, ...args }, "Indeterminate disabled"))),
};
export const Composed = {
    args: {
        size: checkboxPropDefs.size.default,
        color: checkboxPropDefs.color.default,
        highContrast: checkboxPropDefs.highContrast.default,
    },
    render: (args) => (React.createElement("div", { style: { display: 'flex', gap: 'var(--space-2)', flexDirection: 'column' } },
        React.createElement(Text, { as: "label", size: "2" },
            React.createElement("div", { style: { display: 'flex', gap: 'var(--space-2)' } },
                React.createElement(Checkbox, { defaultChecked: true, ...args }),
                "Default")),
        React.createElement(Text, { as: "label", size: "2" },
            React.createElement("div", { style: { display: 'flex', gap: 'var(--space-2)' } },
                React.createElement(Checkbox, { defaultChecked: true, disabled: true, ...args }),
                "Disabled checked")),
        React.createElement(Text, { as: "label", size: "2" },
            React.createElement("div", { style: { display: 'flex', gap: 'var(--space-2)' } },
                React.createElement(Checkbox, { defaultChecked: false, disabled: true, ...args }),
                "Disabled unchecked")))),
};
export const Size = {
    render: (args) => (React.createElement("div", { style: { display: 'flex', gap: 'var(--space-2)' } },
        React.createElement(Checkbox, { defaultChecked: true, ...args, size: "1" }),
        React.createElement(Checkbox, { defaultChecked: true, ...args, size: "2" }),
        React.createElement(Checkbox, { defaultChecked: true, ...args, size: "3" }))),
};
export const Color = {
    render: (args) => (React.createElement("div", { style: { display: 'flex', gap: 'var(--space-2)' } },
        React.createElement(Checkbox, { ...args, color: "indigo", defaultChecked: true }),
        React.createElement(Checkbox, { ...args, color: "cyan", defaultChecked: true }),
        React.createElement(Checkbox, { ...args, color: "orange", defaultChecked: true }),
        React.createElement(Checkbox, { ...args, color: "crimson", defaultChecked: true }))),
};
export const HighContrast = {
    name: 'High Contrast',
    render: (args) => (React.createElement("div", { style: {
            display: 'inline-grid',
            gridTemplateColumns: 'repeat(2, 1fr)',
            gap: '8px',
        } },
        React.createElement(Checkbox, { ...args, color: "indigo", defaultChecked: true }),
        React.createElement(Checkbox, { ...args, color: "indigo", defaultChecked: true, highContrast: true }),
        React.createElement(Checkbox, { ...args, color: "cyan", defaultChecked: true }),
        React.createElement(Checkbox, { ...args, color: "cyan", defaultChecked: true, highContrast: true }),
        React.createElement(Checkbox, { ...args, color: "orange", defaultChecked: true }),
        React.createElement(Checkbox, { ...args, color: "orange", defaultChecked: true, highContrast: true }),
        React.createElement(Checkbox, { ...args, color: "crimson", defaultChecked: true }),
        React.createElement(Checkbox, { ...args, color: "crimson", defaultChecked: true, highContrast: true }))),
};
export const Alignment = {
    name: 'Alignment with text',
    render: (args) => (React.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-3)' } },
        React.createElement(Text, { style: { marginBottom: 12 } },
            "Composing ",
            React.createElement(Code, null, "Checkbox"),
            " within ",
            React.createElement(Code, null, "Text"),
            " automatically centers it with the first line of text."),
        React.createElement(Text, { as: "label", size: "3" },
            React.createElement("div", { style: { display: 'flex', gap: 'var(--space-2)' } },
                React.createElement(Checkbox, { ...args, size: "1", defaultChecked: true }),
                " Agree to Terms and Conditions")),
        React.createElement(Text, { as: "label", size: "4" },
            React.createElement("div", { style: { display: 'flex', gap: 'var(--space-2)' } },
                React.createElement(Checkbox, { ...args, size: "2", defaultChecked: true }),
                " Agree to Terms and Conditions")),
        React.createElement(Text, { as: "label", size: "5" },
            React.createElement("div", { style: { display: 'flex', gap: 'var(--space-3)' } },
                React.createElement(Checkbox, { ...args, size: "3", defaultChecked: true }),
                " Agree to Terms and Conditions")),
        React.createElement("div", { style: { maxWidth: 300 } },
            React.createElement(Text, { as: "label", size: "4" },
                React.createElement("div", { style: { display: 'flex', gap: 'var(--space-2)' } },
                    React.createElement(Checkbox, { defaultChecked: true }),
                    " It is automatically well-aligned with multi-line text too."))))),
};
