// frosted-ui/dist/cjs/components/checkbox/checkbox.stories.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Alignment = exports.HighContrast = exports.Color = exports.Size = exports.Composed = exports.Default = void 0;
const tslib_1 = require("tslib");
const react_1 = tslib_1.__importDefault(require("react"));
const __1 = require("..");
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Controls/Checkbox',
    component: __1.Checkbox,
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
};
exports.default = meta;
// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
exports.Default = {
    args: {
        size: __1.checkboxPropDefs.size.default,
        color: __1.checkboxPropDefs.color.default,
        highContrast: __1.checkboxPropDefs.highContrast.default,
    },
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-2)', flexDirection: 'column' } },
        react_1.default.createElement(__1.Checkbox, { checked: true, ...args }, "Checked"),
        react_1.default.createElement(__1.Checkbox, { checked: true, disabled: true, ...args }, "Checked disabled"),
        react_1.default.createElement(__1.Checkbox, { checked: false, ...args }, "Unchecked"),
        react_1.default.createElement(__1.Checkbox, { checked: false, disabled: true, ...args }, "Unchecked disabled"),
        react_1.default.createElement(__1.Checkbox, { checked: 'indeterminate', ...args }, "Indeterminate"),
        react_1.default.createElement(__1.Checkbox, { checked: 'indeterminate', disabled: true, ...args }, "Indeterminate disabled"))),
};
exports.Composed = {
    args: {
        size: __1.checkboxPropDefs.size.default,
        color: __1.checkboxPropDefs.color.default,
        highContrast: __1.checkboxPropDefs.highContrast.default,
    },
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-2)', flexDirection: 'column' } },
        react_1.default.createElement(__1.Text, { as: "label", size: "2" },
            react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-2)' } },
                react_1.default.createElement(__1.Checkbox, { defaultChecked: true, ...args }),
                "Default")),
        react_1.default.createElement(__1.Text, { as: "label", size: "2" },
            react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-2)' } },
                react_1.default.createElement(__1.Checkbox, { defaultChecked: true, disabled: true, ...args }),
                "Disabled checked")),
        react_1.default.createElement(__1.Text, { as: "label", size: "2" },
            react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-2)' } },
                react_1.default.createElement(__1.Checkbox, { defaultChecked: false, disabled: true, ...args }),
                "Disabled unchecked")))),
};
exports.Size = {
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-2)' } },
        react_1.default.createElement(__1.Checkbox, { defaultChecked: true, ...args, size: "1" }),
        react_1.default.createElement(__1.Checkbox, { defaultChecked: true, ...args, size: "2" }),
        react_1.default.createElement(__1.Checkbox, { defaultChecked: true, ...args, size: "3" }))),
};
exports.Color = {
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-2)' } },
        react_1.default.createElement(__1.Checkbox, { ...args, color: "indigo", defaultChecked: true }),
        react_1.default.createElement(__1.Checkbox, { ...args, color: "cyan", defaultChecked: true }),
        react_1.default.createElement(__1.Checkbox, { ...args, color: "orange", defaultChecked: true }),
        react_1.default.createElement(__1.Checkbox, { ...args, color: "crimson", defaultChecked: true }))),
};
exports.HighContrast = {
    name: 'High Contrast',
    render: (args) => (react_1.default.createElement("div", { style: {
            display: 'inline-grid',
            gridTemplateColumns: 'repeat(2, 1fr)',
            gap: '8px',
        } },
        react_1.default.createElement(__1.Checkbox, { ...args, color: "indigo", defaultChecked: true }),
        react_1.default.createElement(__1.Checkbox, { ...args, color: "indigo", defaultChecked: true, highContrast: true }),
        react_1.default.createElement(__1.Checkbox, { ...args, color: "cyan", defaultChecked: true }),
        react_1.default.createElement(__1.Checkbox, { ...args, color: "cyan", defaultChecked: true, highContrast: true }),
        react_1.default.createElement(__1.Checkbox, { ...args, color: "orange", defaultChecked: true }),
        react_1.default.createElement(__1.Checkbox, { ...args, color: "orange", defaultChecked: true, highContrast: true }),
        react_1.default.createElement(__1.Checkbox, { ...args, color: "crimson", defaultChecked: true }),
        react_1.default.createElement(__1.Checkbox, { ...args, color: "crimson", defaultChecked: true, highContrast: true }))),
};
exports.Alignment = {
    name: 'Alignment with text',
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-3)' } },
        react_1.default.createElement(__1.Text, { style: { marginBottom: 12 } },
            "Composing ",
            react_1.default.createElement(__1.Code, null, "Checkbox"),
            " within ",
            react_1.default.createElement(__1.Code, null, "Text"),
            " automatically centers it with the first line of text."),
        react_1.default.createElement(__1.Text, { as: "label", size: "3" },
            react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-2)' } },
                react_1.default.createElement(__1.Checkbox, { ...args, size: "1", defaultChecked: true }),
                " Agree to Terms and Conditions")),
        react_1.default.createElement(__1.Text, { as: "label", size: "4" },
            react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-2)' } },
                react_1.default.createElement(__1.Checkbox, { ...args, size: "2", defaultChecked: true }),
                " Agree to Terms and Conditions")),
        react_1.default.createElement(__1.Text, { as: "label", size: "5" },
            react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-3)' } },
                react_1.default.createElement(__1.Checkbox, { ...args, size: "3", defaultChecked: true }),
                " Agree to Terms and Conditions")),
        react_1.default.createElement("div", { style: { maxWidth: 300 } },
            react_1.default.createElement(__1.Text, { as: "label", size: "4" },
                react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-2)' } },
                    react_1.default.createElement(__1.Checkbox, { defaultChecked: true }),
                    " It is automatically well-aligned with multi-line text too."))))),
};
