// frosted-ui/dist/esm/theme-options.js
import { radixColorScales, radixColorScalesBright, radixColorScalesMetal, radixColorScalesRegular, 
//
radixGetMatchingGrayScale, 
//
radixGrayScalePure, radixGrayScales, radixGrayScalesDesaturated, } from './helpers/radix-colors';
const colorPanelElevationColors = ['color-panel-elevation'];
const semanticColors = ['danger', 'warning', 'success', 'info'];
const appearances = ['inherit', 'light', 'dark'];
const accentColors = [...radixColorScales, 'gray'];
const grayColors = [...radixGrayScales, 'auto'];
const dangerColors = ['tomato', 'red', 'ruby'];
const warningColors = ['yellow', 'amber'];
const successColors = ['teal', 'jade', 'green', 'grass'];
const infoColors = ['blue', 'sky'];
const themePropDefs = {
    hasBackground: { type: 'boolean', default: true },
    appearance: { type: 'enum', values: appearances, default: 'inherit' },
    accentColor: { type: 'enum', values: accentColors, default: 'blue' },
    grayColor: { type: 'enum', values: grayColors, default: 'gray' },
    dangerColor: { type: 'enum', values: dangerColors, default: 'red' },
    warningColor: { type: 'enum', values: warningColors, default: 'amber' },
    successColor: { type: 'enum', values: successColors, default: 'green' },
    infoColor: { type: 'enum', values: infoColors, default: 'sky' },
};
const themeAccentColorsGrouped = [
    {
        label: 'Regulars',
        values: [...radixColorScalesRegular],
    },
    {
        label: 'Brights',
        values: [...radixColorScalesBright],
    },
    { label: 'Metals', values: [...radixColorScalesMetal] },
    { label: 'Gray', values: ['gray'] },
];
const themeAccentColorsOrdered = [
    'gray',
    'gold',
    'bronze',
    'brown',
    'yellow',
    'amber',
    'tomato',
    'red',
    'ruby',
    'crimson',
    'pink',
    'plum',
    'purple',
    'violet',
    'iris',
    'cyan',
    'teal',
    'jade',
    'green',
    'grass',
    'mint',
    'sky',
    // Whop Brand Colors
    'blue',
    'orange',
    'indigo',
    'magenta',
    'lemon',
    'lime',
];
const themeGrayColorsGrouped = [
    { label: 'Pure', values: [radixGrayScalePure] },
    {
        label: 'Desaturated',
        values: ['auto', ...radixGrayScalesDesaturated],
    },
];
function getMatchingGrayColor(accentColor) {
    if (accentColor === 'gray')
        return 'gray';
    return radixGetMatchingGrayScale(accentColor);
}
export { colorPanelElevationColors, dangerColors, getMatchingGrayColor, infoColors, semanticColors, successColors, 
//
themeAccentColorsGrouped, themeAccentColorsOrdered, themeGrayColorsGrouped, themePropDefs, warningColors, };
