// frosted-ui/dist/esm/components/breadcrumbs/breadcrumbs.stories.js
import React from 'react';
import { Breadcrumbs, breadcrumbsPropDefs } from '..';
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Components/Breadcrumbs',
    component: Breadcrumbs.Root,
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
};
export default meta;
export const WithLinks = {
    name: 'With links',
    args: {
        color: breadcrumbsPropDefs.color.default,
    },
    render: (args) => (React.createElement(Breadcrumbs.Root, { ...args },
        React.createElement(Breadcrumbs.Item, { asChild: true },
            React.createElement("a", { href: "#" }, "Home")),
        React.createElement(Breadcrumbs.Item, { asChild: true },
            React.createElement("a", { href: "#" }, "User Profiles")),
        React.createElement(Breadcrumbs.Item, null, "Products"))),
};
export const WithOnClick = {
    name: 'With onClick',
    args: {
        color: breadcrumbsPropDefs.color.default,
    },
    render: (args) => (React.createElement(Breadcrumbs.Root, { ...args },
        React.createElement(Breadcrumbs.Item, { onClick: () => alert('Home') }, "Home"),
        React.createElement(Breadcrumbs.Item, { onClick: () => alert('Products') }, "Products"),
        React.createElement(Breadcrumbs.Item, { onClick: () => alert('Sneaker Bot') }, "Sneaker Bot"))),
};
export const Truncated = {
    name: 'Truncated',
    args: {
        color: breadcrumbsPropDefs.color.default,
    },
    render: (args) => (React.createElement(Breadcrumbs.Root, { ...args },
        React.createElement(Breadcrumbs.Item, { asChild: true },
            React.createElement("a", { href: "#" }, "Home")),
        React.createElement(Breadcrumbs.Dropdown, null,
            React.createElement(Breadcrumbs.DropdownItem, { asChild: true },
                React.createElement("a", { href: "#" }, "Products")),
            React.createElement(Breadcrumbs.DropdownItem, { asChild: true },
                React.createElement("a", { href: "#" }, "Categories")),
            React.createElement(Breadcrumbs.DropdownItem, { asChild: true },
                React.createElement("a", { href: "#" }, "Software"))),
        React.createElement(Breadcrumbs.Item, { asChild: true },
            React.createElement("a", { href: "#" }, "Bots")),
        React.createElement(Breadcrumbs.Item, null, "Sneaker Bot"))),
};
