// frosted-ui/dist/cjs/components/otp-field/otp-field.stories.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Controlled = exports.Separator = exports.Pattern = exports.Default = void 0;
const tslib_1 = require("tslib");
const input_otp_1 = require("input-otp");
const react_1 = tslib_1.__importDefault(require("react"));
const __1 = require("..");
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Controls/OTPField',
    component: __1.OTPField.Root,
    args: {},
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
};
exports.default = meta;
// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
exports.Default = {
    render: (args) => {
        // eslint-disable-next-line @typescript-eslint/no-unused-vars
        const { children, ...otherArgs } = args;
        return (react_1.default.createElement("div", null,
            react_1.default.createElement(__1.OTPField.Root, { ...otherArgs, maxLength: 6, render: ({ slots }) => (react_1.default.createElement(react_1.default.Fragment, null,
                    react_1.default.createElement(__1.OTPField.Group, null,
                        slots.slice(0, 3).map((slot, index) => (react_1.default.createElement(__1.OTPField.Slot, { key: index, ...slot }))),
                        ' '),
                    react_1.default.createElement(__1.OTPField.Separator, null),
                    react_1.default.createElement(__1.OTPField.Group, null, slots.slice(3).map((slot, index) => (react_1.default.createElement(__1.OTPField.Slot, { key: index, ...slot })))))) })));
    },
};
exports.Pattern = {
    render: (args) => {
        // eslint-disable-next-line @typescript-eslint/no-unused-vars
        const { children, ...otherArgs } = args;
        return (react_1.default.createElement("div", null,
            react_1.default.createElement(__1.OTPField.Root, { ...otherArgs, maxLength: 6, pattern: input_otp_1.REGEXP_ONLY_DIGITS_AND_CHARS, render: ({ slots }) => (react_1.default.createElement(__1.OTPField.Group, null,
                    slots.map((slot, index) => (react_1.default.createElement(__1.OTPField.Slot, { key: index, ...slot }))),
                    ' ')) })));
    },
};
exports.Separator = {
    render: (args) => {
        // eslint-disable-next-line @typescript-eslint/no-unused-vars
        const { children, ...otherArgs } = args;
        return (react_1.default.createElement("div", null,
            react_1.default.createElement(__1.OTPField.Root, { ...otherArgs, maxLength: 6, render: ({ slots }) => (react_1.default.createElement(__1.OTPField.Group, { style: { gap: 4 } },
                    slots.map((slot, index) => (react_1.default.createElement(react_1.default.Fragment, { key: index },
                        react_1.default.createElement(__1.OTPField.Slot, { style: { borderRadius: 10 }, ...slot }),
                        index !== slots.length - 1 && react_1.default.createElement(__1.OTPField.Separator, null)))),
                    ' ')) })));
    },
};
exports.Controlled = {
    render: (args) => {
        // eslint-disable-next-line @typescript-eslint/no-unused-vars
        const { children, ...otherArgs } = args;
        const [value, setValue] = react_1.default.useState('');
        return (react_1.default.createElement("div", null,
            react_1.default.createElement("div", { style: { marginBottom: 16 } },
                react_1.default.createElement(__1.OTPField.Root, { ...otherArgs, maxLength: 6, value: value, onChange: (value) => setValue(value), render: ({ slots }) => (react_1.default.createElement(__1.OTPField.Group, null,
                        slots.map((slot, index) => (react_1.default.createElement(__1.OTPField.Slot, { key: index, ...slot }))),
                        ' ')) })),
            react_1.default.createElement(__1.Text, { align: "center", color: "gray", size: "2", as: "div" }, value === '' ? react_1.default.createElement(react_1.default.Fragment, null, "Enter your one-time password.") : react_1.default.createElement(react_1.default.Fragment, null,
                "You entered: ",
                value))));
    },
};
