// frosted-ui/dist/esm/components/tabs-nav/tabs-nav.stories.js
import React from 'react';
import { TabsNav, tabsNavPropDefs } from '..';
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Components/TabsNav',
    component: TabsNav.Root,
    args: {
        size: tabsNavPropDefs.size.default,
    },
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
};
export default meta;
// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
export const Default = {
    render: (args) => (React.createElement("div", { style: { width: 600 } },
        React.createElement(TabsNav.Root, { ...args },
            React.createElement(TabsNav.Link, { active: true, href: "#" }, "Account"),
            React.createElement(TabsNav.Link, { href: "#" }, "Documents"),
            React.createElement(TabsNav.Link, { href: "#" }, "Settings")))),
};
