// frosted-ui/dist/esm/components/button/button.stories.js
import { Download16 } from '@frosted-ui/icons';
import React from 'react';
import { Button, Code, Spinner, Text, buttonPropDefs } from '..';
const ExampleIcon = () => (React.createElement("svg", { width: "16", height: "16", viewBox: "0 0 15 15", fill: "none", xmlns: "http://www.w3.org/2000/svg" },
    React.createElement("path", { d: "M3 2.5C3 2.22386 3.22386 2 3.5 2H11.5C11.7761 2 12 2.22386 12 2.5V13.5C12 13.6818 11.9014 13.8492 11.7424 13.9373C11.5834 14.0254 11.3891 14.0203 11.235 13.924L7.5 11.5896L3.765 13.924C3.61087 14.0203 3.41659 14.0254 3.25762 13.9373C3.09864 13.8492 3 13.6818 3 13.5V2.5ZM4 3V12.5979L6.97 10.7416C7.29427 10.539 7.70573 10.539 8.03 10.7416L11 12.5979V3H4Z", fill: "currentColor", fillRule: "evenodd", clipRule: "evenodd" })));
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Controls/Button',
    component: Button,
    args: {
        disabled: false,
    },
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
};
export default meta;
// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
export const Default = {
    args: {
        children: 'Button',
        size: buttonPropDefs.size.default,
        variant: buttonPropDefs.variant.default,
        color: buttonPropDefs.color.default,
    },
};
export const Size = {
    args: {
        children: 'Button',
        size: buttonPropDefs.size.default,
        variant: buttonPropDefs.variant.default,
        color: buttonPropDefs.color.default,
    },
    render: (args) => (React.createElement("div", { style: { display: 'flex', alignItems: 'center', gap: 'var(--space-2)' } },
        React.createElement(Button, { ...args, size: "4" }),
        React.createElement(Button, { ...args, size: "3" }),
        React.createElement(Button, { ...args, size: "2" }),
        React.createElement(Button, { ...args, size: "1" }))),
};
export const Variant = {
    args: {
        children: 'Button',
        size: buttonPropDefs.size.default,
        color: buttonPropDefs.color.default,
    },
    render: (args) => (React.createElement("div", { style: { display: 'flex', alignItems: 'center', gap: 'var(--space-4)' } },
        React.createElement(Button, { ...args, variant: "classic" }),
        React.createElement(Button, { ...args, variant: "solid" }),
        React.createElement(Button, { ...args, variant: "soft" }),
        React.createElement(Button, { ...args, variant: "surface" }),
        React.createElement(Button, { ...args, variant: "ghost" }))),
};
export const Color = {
    args: {
        children: 'Button',
        size: buttonPropDefs.size.default,
    },
    render: (args) => (React.createElement("div", { style: { display: 'flex', alignItems: 'center', gap: 'var(--space-2)' } },
        React.createElement(Button, { ...args, color: "indigo" }),
        React.createElement(Button, { ...args, color: "cyan" }),
        React.createElement(Button, { ...args, color: "orange" }),
        React.createElement(Button, { ...args, color: "crimson" }))),
};
export const SemanticColor = {
    name: 'Semantic color',
    args: {
        size: buttonPropDefs.size.default,
    },
    render: (args) => (React.createElement("div", { style: { display: 'flex', alignItems: 'center', gap: 'var(--space-2)' } },
        React.createElement(Button, { ...args, color: "info" }, "Info"),
        React.createElement(Button, { ...args, color: "success" }, "Success"),
        React.createElement(Button, { ...args, color: "warning" }, "Warning"),
        React.createElement(Button, { ...args, color: "danger" }, "Danger"))),
};
export const HighContrast = {
    name: 'High Contrast',
    args: {
        children: 'Button',
    },
    render: (args) => (React.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-2)' } },
        React.createElement("div", { style: { display: 'flex', gap: 'var(--space-2)' } },
            React.createElement(Button, { ...args, highContrast: false, variant: "classic" }),
            React.createElement(Button, { ...args, highContrast: false, variant: "solid" }),
            React.createElement(Button, { ...args, highContrast: false, variant: "soft" }),
            React.createElement(Button, { ...args, highContrast: false, variant: "surface" }),
            React.createElement(Button, { ...args, highContrast: false, variant: "ghost" })),
        React.createElement("div", { style: { display: 'flex', gap: 'var(--space-2)' } },
            React.createElement(Button, { ...args, variant: "classic", highContrast: true }),
            React.createElement(Button, { ...args, variant: "solid", highContrast: true }),
            React.createElement(Button, { ...args, variant: "soft", highContrast: true }),
            React.createElement(Button, { ...args, variant: "surface", highContrast: true }),
            React.createElement(Button, { ...args, variant: "ghost", highContrast: true })))),
};
export const WithIcons = {
    name: 'With Icons',
    args: {
        size: buttonPropDefs.size.default,
    },
    render: (args) => (React.createElement("div", { style: { display: 'flex', alignItems: 'center', gap: 'var(--space-2)' } },
        React.createElement(Button, { ...args },
            React.createElement(ExampleIcon, null),
            " With icon"))),
};
export const Loading = {
    args: {
        children: 'Button',
        size: buttonPropDefs.size.default,
        color: buttonPropDefs.color.default,
        disabled: undefined,
        loading: true,
    },
    render: (args) => (React.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-5)', maxWidth: 650 } },
        React.createElement(Text, null,
            "Buttons have their own ",
            React.createElement(Code, null, "loading"),
            " prop that automatically composes a spinner."),
        React.createElement("div", { style: { display: 'flex', alignItems: 'center', gap: 'var(--space-4)' } },
            React.createElement(Button, { ...args, variant: "classic" }),
            React.createElement(Button, { ...args, variant: "solid" }),
            React.createElement(Button, { ...args, variant: "soft" }),
            React.createElement(Button, { ...args, variant: "surface" }),
            React.createElement(Button, { ...args, variant: "ghost" })),
        React.createElement(Text, null,
            "If you have an icon inside the button, you can use the button`s ",
            React.createElement(Code, null, "disabled"),
            " state and wrap the icon in a standalone ",
            React.createElement(Code, null, `<Spinner>`),
            " to achieve a more sophisticated design."),
        React.createElement("div", { style: { display: 'flex', alignItems: 'center', gap: 'var(--space-4)' } },
            React.createElement(Button, { ...args, loading: false, variant: "classic" },
                React.createElement(Spinner, { loading: false },
                    React.createElement(Download16, null)),
                "Download"),
            React.createElement(Button, { ...args, loading: false, variant: "classic", disabled: true },
                React.createElement(Spinner, { loading: true },
                    React.createElement(Download16, null)),
                "Download")),
        React.createElement("div", { style: { display: 'flex', alignItems: 'center', gap: 'var(--space-4)' } },
            React.createElement(Button, { ...args, loading: false, variant: "soft" },
                React.createElement(Spinner, { loading: false },
                    React.createElement(Download16, null)),
                "Download"),
            React.createElement(Button, { ...args, loading: false, variant: "soft", disabled: true },
                React.createElement(Spinner, { loading: true },
                    React.createElement(Download16, null)),
                "Download")),
        React.createElement("div", { style: { display: 'flex', alignItems: 'center', gap: 'var(--space-4)' } },
            React.createElement(Button, { ...args, loading: false, variant: "solid" },
                React.createElement(Spinner, { loading: false },
                    React.createElement(Download16, null)),
                "Download"),
            React.createElement(Button, { ...args, loading: false, variant: "solid", disabled: true },
                React.createElement(Spinner, { loading: true },
                    React.createElement(Download16, null)),
                "Download")),
        React.createElement("div", { style: { display: 'flex', alignItems: 'center', gap: 'var(--space-4)' } },
            React.createElement(Button, { ...args, loading: false, variant: "surface" },
                React.createElement(Spinner, { loading: false },
                    React.createElement(Download16, null)),
                "Download"),
            React.createElement(Button, { ...args, loading: false, variant: "surface", disabled: true },
                React.createElement(Spinner, { loading: true },
                    React.createElement(Download16, null)),
                "Download")))),
};
