// frosted-ui/dist/cjs/components/link/link.stories.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.HighContrast = exports.Underline = exports.Color = exports.Size = exports.Default = void 0;
const tslib_1 = require("tslib");
const react_1 = tslib_1.__importDefault(require("react"));
const __1 = require("..");
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Typography/Link',
    component: __1.Link,
    args: {
        children: 'Frosted-UI',
        href: 'https://storybook.whop.dev/',
        target: '_blank',
        color: __1.linkPropDefs.color.default,
        size: __1.linkPropDefs.size.default,
    },
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
};
exports.default = meta;
// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
exports.Default = {};
exports.Size = {
    args: {
        children: 'Frosted-UI',
        href: 'https://storybook.whop.dev/',
        target: '_blank',
        color: __1.linkPropDefs.color.default,
    },
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-3)' } },
        react_1.default.createElement(__1.Link, { ...args, size: "0" }),
        react_1.default.createElement(__1.Link, { ...args, size: "1" }),
        react_1.default.createElement(__1.Link, { ...args, size: "2" }),
        react_1.default.createElement(__1.Link, { ...args, size: "3" }),
        react_1.default.createElement(__1.Link, { ...args, size: "4" }),
        react_1.default.createElement(__1.Link, { ...args, size: "5" }),
        react_1.default.createElement(__1.Link, { ...args, size: "6" }),
        react_1.default.createElement(__1.Link, { ...args, size: "7" }),
        react_1.default.createElement(__1.Link, { ...args, size: "8" }),
        react_1.default.createElement(__1.Link, { ...args, size: "9" }))),
};
exports.Color = {
    args: {
        children: 'Frosted-UI',
        href: 'https://storybook.whop.dev/',
        target: '_blank',
        color: __1.linkPropDefs.color.default,
    },
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', alignItems: 'start', gap: 'var(--space-2)' } },
        react_1.default.createElement(__1.Link, { ...args, color: "indigo" }),
        react_1.default.createElement(__1.Link, { ...args, color: "cyan" }),
        react_1.default.createElement(__1.Link, { ...args, color: "orange" }),
        react_1.default.createElement(__1.Link, { ...args, color: "crimson" }))),
};
exports.Underline = {
    args: {
        children: 'Frosted-UI',
        href: 'https://storybook.whop.dev/',
        target: '_blank',
        color: __1.linkPropDefs.color.default,
    },
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', alignItems: 'start', gap: 'var(--space-2)' } },
        react_1.default.createElement(__1.Link, { ...args, underline: "auto" }),
        react_1.default.createElement(__1.Link, { ...args, underline: "hover" }),
        react_1.default.createElement(__1.Link, { ...args, underline: "always" }))),
};
exports.HighContrast = {
    name: 'High Contrast',
    args: {
        children: 'Frosted-UI',
        href: 'https://storybook.whop.dev/',
        target: '_blank',
        color: __1.linkPropDefs.color.default,
    },
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column' } },
        react_1.default.createElement(__1.Link, { ...args }),
        react_1.default.createElement(__1.Link, { ...args, highContrast: true }))),
};
