// frosted-ui/dist/esm/components/link/link.stories.js
import React from 'react';
import { Link, linkPropDefs } from '..';
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Typography/Link',
    component: Link,
    args: {
        children: 'Frosted-UI',
        href: 'https://storybook.whop.dev/',
        target: '_blank',
        color: linkPropDefs.color.default,
        size: linkPropDefs.size.default,
    },
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
};
export default meta;
// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
export const Default = {};
export const Size = {
    args: {
        children: 'Frosted-UI',
        href: 'https://storybook.whop.dev/',
        target: '_blank',
        color: linkPropDefs.color.default,
    },
    render: (args) => (React.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-3)' } },
        React.createElement(Link, { ...args, size: "0" }),
        React.createElement(Link, { ...args, size: "1" }),
        React.createElement(Link, { ...args, size: "2" }),
        React.createElement(Link, { ...args, size: "3" }),
        React.createElement(Link, { ...args, size: "4" }),
        React.createElement(Link, { ...args, size: "5" }),
        React.createElement(Link, { ...args, size: "6" }),
        React.createElement(Link, { ...args, size: "7" }),
        React.createElement(Link, { ...args, size: "8" }),
        React.createElement(Link, { ...args, size: "9" }))),
};
export const Color = {
    args: {
        children: 'Frosted-UI',
        href: 'https://storybook.whop.dev/',
        target: '_blank',
        color: linkPropDefs.color.default,
    },
    render: (args) => (React.createElement("div", { style: { display: 'flex', flexDirection: 'column', alignItems: 'start', gap: 'var(--space-2)' } },
        React.createElement(Link, { ...args, color: "indigo" }),
        React.createElement(Link, { ...args, color: "cyan" }),
        React.createElement(Link, { ...args, color: "orange" }),
        React.createElement(Link, { ...args, color: "crimson" }))),
};
export const Underline = {
    args: {
        children: 'Frosted-UI',
        href: 'https://storybook.whop.dev/',
        target: '_blank',
        color: linkPropDefs.color.default,
    },
    render: (args) => (React.createElement("div", { style: { display: 'flex', flexDirection: 'column', alignItems: 'start', gap: 'var(--space-2)' } },
        React.createElement(Link, { ...args, underline: "auto" }),
        React.createElement(Link, { ...args, underline: "hover" }),
        React.createElement(Link, { ...args, underline: "always" }))),
};
export const HighContrast = {
    name: 'High Contrast',
    args: {
        children: 'Frosted-UI',
        href: 'https://storybook.whop.dev/',
        target: '_blank',
        color: linkPropDefs.color.default,
    },
    render: (args) => (React.createElement("div", { style: { display: 'flex', flexDirection: 'column' } },
        React.createElement(Link, { ...args }),
        React.createElement(Link, { ...args, highContrast: true }))),
};
