// frosted-ui/dist/cjs/components/visually-hidden/visually-hidden.stories.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Default = void 0;
const tslib_1 = require("tslib");
const react_1 = tslib_1.__importDefault(require("react"));
const __1 = require("..");
const CrossIcon = () => (react_1.default.createElement("svg", { viewBox: "0 0 32 32", width: 24, height: 24, fill: "none", stroke: "currentColor" },
    react_1.default.createElement("path", { d: "M2 30 L30 2 M30 30 L2 2" })));
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Utilities/VisuallyHidden',
    component: __1.VisuallyHidden.Root,
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
};
exports.default = meta;
// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
exports.Default = {
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-4)' } },
        react_1.default.createElement(__1.Text, null,
            "Anything you put inside this component will be hidden visually but will be announced by screen readers. See",
            ' ',
            react_1.default.createElement(__1.Link, { href: "https://www.radix-ui.com/primitives/docs/utilities/visually-hidden" }, "Radix UI docs"),
            "."),
        react_1.default.createElement(__1.Text, null,
            "This is useful in certain scenarios as an alternative to traditional labelling with ",
            react_1.default.createElement(__1.Code, null, "aria-label"),
            " or",
            ' ',
            react_1.default.createElement(__1.Code, null, "aria-labelledby"),
            "."),
        react_1.default.createElement("div", null,
            react_1.default.createElement(__1.Button, { size: "4" },
                react_1.default.createElement(CrossIcon, null),
                react_1.default.createElement(__1.VisuallyHidden.Root, { ...args }, "Close"))))),
};
