// frosted-ui/dist/cjs/components/card/card.stories.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AsAnotherElement = exports.InsetContent = exports.Variant = exports.Size = exports.Default = void 0;
const tslib_1 = require("tslib");
const react_1 = tslib_1.__importDefault(require("react"));
const __1 = require("..");
const CardContentExample = () => (react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-3)', alignItems: 'center' } },
    react_1.default.createElement(__1.Avatar, { size: "3", fallback: "IM", color: "indigo" }),
    react_1.default.createElement("div", null,
        react_1.default.createElement(__1.Text, { as: "div", size: "2", weight: "bold" }, "Ilya Miskov"),
        react_1.default.createElement(__1.Text, { as: "div", size: "2", color: "gray" }, "I love how we have the freedom to explore skeuomorphism"))));
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Components/Card',
    component: __1.Card,
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
};
exports.default = meta;
// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
exports.Default = {
    args: {
        children: react_1.default.createElement(CardContentExample, null),
        size: __1.cardPropDefs.size.default,
    },
};
exports.Size = {
    args: {
        size: __1.cardPropDefs.size.default,
    },
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-3)', flexDirection: 'column' } },
        react_1.default.createElement(__1.Card, { ...args, size: "1", style: { width: 350 } },
            react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-3)', alignItems: 'center' } },
                react_1.default.createElement(__1.Avatar, { size: "3", fallback: "T", color: "indigo" }),
                react_1.default.createElement("div", null,
                    react_1.default.createElement(__1.Text, { as: "div", size: "2", weight: "bold" }, "Teodros Girmay"),
                    react_1.default.createElement(__1.Text, { as: "div", size: "2", color: "gray" }, "Engineering")))),
        react_1.default.createElement(__1.Card, { ...args, size: "2", style: { width: 425 } },
            react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-4)', alignItems: 'center' } },
                react_1.default.createElement(__1.Avatar, { size: "4", fallback: "T", color: "indigo" }),
                react_1.default.createElement("div", null,
                    react_1.default.createElement(__1.Text, { as: "div", weight: "bold" }, "Teodros Girmay"),
                    react_1.default.createElement(__1.Text, { as: "div", color: "gray" }, "Engineering")))),
        react_1.default.createElement(__1.Card, { ...args, size: "3", style: { width: 500 } },
            react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-4)', alignItems: 'center' } },
                react_1.default.createElement(__1.Avatar, { size: "5", fallback: "T", color: "indigo" }),
                react_1.default.createElement("div", null,
                    react_1.default.createElement(__1.Text, { as: "div", size: "4", weight: "bold" }, "Teodros Girmay"),
                    react_1.default.createElement(__1.Text, { as: "div", size: "4", color: "gray" }, "Engineering")))),
        react_1.default.createElement(__1.Card, { ...args, size: "4", style: { width: 500 } },
            react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-4)', alignItems: 'center' } },
                react_1.default.createElement(__1.Avatar, { size: "5", fallback: "T", color: "indigo" }),
                react_1.default.createElement("div", null,
                    react_1.default.createElement(__1.Text, { as: "div", size: "4", weight: "bold" }, "Teodros Girmay"),
                    react_1.default.createElement(__1.Text, { as: "div", size: "4", color: "gray" }, "Engineering")))),
        react_1.default.createElement(__1.Card, { ...args, size: "5", style: { width: 500 } },
            react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-4)', alignItems: 'center' } },
                react_1.default.createElement(__1.Avatar, { size: "5", fallback: "T", color: "indigo" }),
                react_1.default.createElement("div", null,
                    react_1.default.createElement(__1.Text, { as: "div", size: "4", weight: "bold" }, "Teodros Girmay"),
                    react_1.default.createElement(__1.Text, { as: "div", size: "4", color: "gray" }, "Engineering")))))),
};
exports.Variant = {
    args: {
        children: react_1.default.createElement(CardContentExample, null),
    },
    render: ({ children, ...args }) => (react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-6)' } },
        react_1.default.createElement(__1.Card, { ...args, variant: "classic" }, children),
        react_1.default.createElement(__1.Card, { ...args, variant: "ghost" }, children),
        react_1.default.createElement(__1.Card, { ...args, variant: "surface" }, children))),
};
exports.InsetContent = {
    name: 'Inset Content',
    render: (args) => (react_1.default.createElement(__1.Card, { size: "2", style: { maxWidth: 240 }, ...args },
        react_1.default.createElement(__1.Inset, { clip: "padding-box", side: "top", pb: "current" },
            react_1.default.createElement("img", { src: "https://images.unsplash.com/photo-1617050318658-a9a3175e34cb?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=600&q=80", alt: "Bold typography", style: {
                    display: 'block',
                    objectFit: 'cover',
                    width: '100%',
                    height: 140,
                    backgroundColor: 'var(--gray-5)',
                } })),
        react_1.default.createElement(__1.Text, { as: "p", size: "3" },
            "Use ",
            react_1.default.createElement(__1.Code, null, '<Inset />'),
            " component to align content flush with the sides of the card."))),
};
exports.AsAnotherElement = {
    name: 'As another element',
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-6)' } },
        react_1.default.createElement("div", null,
            react_1.default.createElement(__1.Text, null,
                "Use the ",
                react_1.default.createElement(__1.Code, null, "asChild"),
                " prop to render the card as a link or a button. This prop adds styles for the interactive states, like hover and focus.")),
        react_1.default.createElement("div", null,
            react_1.default.createElement(__1.Card, { ...args, asChild: true, style: { maxWidth: 350 } },
                react_1.default.createElement("a", { href: "#" },
                    react_1.default.createElement(CardContentExample, null)))))),
};
