// frosted-ui/dist/esm/components/visually-hidden/visually-hidden.stories.js
import React from 'react';
import { Button, Code, Link, Text, VisuallyHidden } from '..';
const CrossIcon = () => (React.createElement("svg", { viewBox: "0 0 32 32", width: 24, height: 24, fill: "none", stroke: "currentColor" },
    React.createElement("path", { d: "M2 30 L30 2 M30 30 L2 2" })));
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Utilities/VisuallyHidden',
    component: VisuallyHidden.Root,
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
};
export default meta;
// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
export const Default = {
    render: (args) => (React.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-4)' } },
        React.createElement(Text, null,
            "Anything you put inside this component will be hidden visually but will be announced by screen readers. See",
            ' ',
            React.createElement(Link, { href: "https://www.radix-ui.com/primitives/docs/utilities/visually-hidden" }, "Radix UI docs"),
            "."),
        React.createElement(Text, null,
            "This is useful in certain scenarios as an alternative to traditional labelling with ",
            React.createElement(Code, null, "aria-label"),
            " or",
            ' ',
            React.createElement(Code, null, "aria-labelledby"),
            "."),
        React.createElement("div", null,
            React.createElement(Button, { size: "4" },
                React.createElement(CrossIcon, null),
                React.createElement(VisuallyHidden.Root, { ...args }, "Close"))))),
};
