// frosted-ui/dist/cjs/components/radio-button-group/radio-button-group.stories.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Color = exports.WithCard = exports.HighContrast = exports.Default = void 0;
const tslib_1 = require("tslib");
const react_1 = tslib_1.__importDefault(require("react"));
const __1 = require("..");
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Controls/RadioButtonGroup',
    component: __1.RadioButtonGroup.Root,
    args: {
        disabled: false,
    },
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
};
exports.default = meta;
// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
exports.Default = {
    args: {
        color: __1.radioGroupPropDefs.color.default,
        highContrast: __1.radioGroupPropDefs.highContrast.default,
    },
    render: (args) => (react_1.default.createElement(__1.RadioButtonGroup.Root, { defaultValue: "1", ...args },
        react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-2)', flexDirection: 'row' } },
            react_1.default.createElement(__1.RadioButtonGroup.Item, { value: "1" },
                react_1.default.createElement("div", { style: {
                        display: 'flex',
                        padding: '8px 40px 8px 24px',
                        borderRadius: 8,
                        border: '1px solid var(--gray-a7)',
                        gap: 'var(--space-2)',
                        alignItems: 'center',
                    } },
                    react_1.default.createElement(__1.RadioButtonGroup.Icon, { style: { marginLeft: -12 } }),
                    react_1.default.createElement(__1.Text, null, "One"))),
            react_1.default.createElement(__1.RadioButtonGroup.Item, { value: "2" },
                react_1.default.createElement("div", { style: {
                        display: 'flex',
                        padding: '8px 40px 8px 24px',
                        borderRadius: 8,
                        border: '1px solid var(--gray-a7)',
                        gap: 'var(--space-2)',
                        alignItems: 'center',
                    } },
                    react_1.default.createElement(__1.RadioButtonGroup.Icon, { style: { marginLeft: -12 } }),
                    react_1.default.createElement(__1.Text, null, "Two"))),
            react_1.default.createElement(__1.RadioButtonGroup.Item, { value: "3" },
                react_1.default.createElement("div", { style: {
                        display: 'flex',
                        padding: '8px 40px 8px 24px',
                        borderRadius: 8,
                        border: '1px solid var(--gray-a7)',
                        gap: 'var(--space-2)',
                        alignItems: 'center',
                    } },
                    react_1.default.createElement(__1.RadioButtonGroup.Icon, { style: { marginLeft: -12 } }),
                    react_1.default.createElement(__1.Text, null, "Three")))))),
};
exports.HighContrast = {
    args: {
        color: __1.radioGroupPropDefs.color.default,
        highContrast: true,
    },
    render: (args) => (react_1.default.createElement(__1.RadioButtonGroup.Root, { defaultValue: "1", ...args },
        react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-2)', flexDirection: 'row' } },
            react_1.default.createElement(__1.Tooltip, { content: "Lime" },
                react_1.default.createElement("div", null,
                    react_1.default.createElement(__1.RadioButtonGroup.Item, { value: "1" },
                        react_1.default.createElement("div", { style: {
                                width: 48,
                                height: 48,
                                borderRadius: 16,
                                background: 'var(--lime-9)',
                            } })))),
            react_1.default.createElement(__1.Tooltip, { content: "Teal" },
                react_1.default.createElement("div", null,
                    react_1.default.createElement(__1.RadioButtonGroup.Item, { value: "2" },
                        react_1.default.createElement("div", { style: {
                                width: 48,
                                height: 48,
                                borderRadius: 16,
                                background: 'var(--teal-9)',
                            } })))),
            react_1.default.createElement(__1.Tooltip, { content: "Gold" },
                react_1.default.createElement("div", null,
                    react_1.default.createElement(__1.RadioButtonGroup.Item, { value: "3" },
                        react_1.default.createElement("div", { style: {
                                width: 48,
                                height: 48,
                                borderRadius: 16,
                                background: 'var(--gold-9)',
                            } }))))))),
};
exports.WithCard = {
    args: {
        color: __1.radioGroupPropDefs.color.default,
        highContrast: __1.radioGroupPropDefs.highContrast.default,
    },
    render: (args) => (react_1.default.createElement(__1.RadioButtonGroup.Root, { defaultValue: "1", ...args },
        react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-2)', flexDirection: 'row' } },
            react_1.default.createElement(__1.RadioButtonGroup.Item, { value: "1" },
                react_1.default.createElement(__1.Card, { size: "2", variant: "classic" },
                    react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-3)', alignItems: 'center' } },
                        react_1.default.createElement(__1.Avatar, { size: "3", fallback: "AB", color: "lime" }),
                        react_1.default.createElement("div", null,
                            react_1.default.createElement(__1.Text, { as: "div", size: "2", weight: "bold" }, "Artur Bie\u0144"),
                            react_1.default.createElement(__1.Text, { as: "div", size: "2", color: "gray" }, "UI engineer"))))),
            react_1.default.createElement(__1.RadioButtonGroup.Item, { value: "2" },
                react_1.default.createElement(__1.Card, { size: "2", variant: "classic" },
                    react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-3)', alignItems: 'center' } },
                        react_1.default.createElement(__1.Avatar, { size: "3", fallback: "IM", color: "sky" }),
                        react_1.default.createElement("div", null,
                            react_1.default.createElement(__1.Text, { as: "div", size: "2", weight: "bold" }, "Ilya Miskov"),
                            react_1.default.createElement(__1.Text, { as: "div", size: "2", color: "gray" }, "Designer"))))),
            react_1.default.createElement(__1.RadioButtonGroup.Item, { value: "3" },
                react_1.default.createElement(__1.Card, { size: "2", variant: "classic" },
                    react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-3)', alignItems: 'center' } },
                        react_1.default.createElement(__1.Avatar, { size: "3", fallback: "SS", color: "orange" }),
                        react_1.default.createElement("div", null,
                            react_1.default.createElement(__1.Text, { as: "div", size: "2", weight: "bold" }, "Steven Schwartz"),
                            react_1.default.createElement(__1.Text, { as: "div", size: "2", color: "gray" }, "CEO")))))))),
};
exports.Color = {
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-2)', flexDirection: 'column' } }, ['indigo', 'cyan', 'orange', 'lime'].map((color) => (react_1.default.createElement(__1.RadioButtonGroup.Root, { ...args, color: color, key: color, defaultValue: "1" },
        react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-2)' } }, ['1', '2', '3'].map((value) => (react_1.default.createElement(__1.RadioButtonGroup.Item, { value: value, key: value },
            react_1.default.createElement(__1.Card, { size: "3", variant: "classic", style: {
                    width: 100,
                    height: 100,
                    '--card-border-radius': '20px',
                } },
                react_1.default.createElement("div", { style: {
                        display: 'flex',
                        alignItems: 'center',
                        justifyContent: 'center',
                        height: '100%',
                        width: '100%',
                    } }, value),
                react_1.default.createElement(__1.RadioButtonGroup.Icon, { style: {
                        position: 'absolute',
                        top: 12,
                        right: 12,
                    } }))))))))))),
};
