// frosted-ui/dist/esm/components/segmented-control/segmented-control.stories.js
import React from 'react';
import { Button, Drawer, SegmentedControl, Text } from '..';
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Components/SegmentedControl',
    component: SegmentedControl.List,
    args: {},
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
};
export default meta;
// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
export const Default = {
    render: (args) => (React.createElement("div", { style: { width: 'auto' } },
        React.createElement(SegmentedControl.Root, { defaultValue: "account" },
            React.createElement(SegmentedControl.List, { ...args },
                React.createElement(SegmentedControl.Trigger, { value: "account" }, "Account"),
                React.createElement(SegmentedControl.Trigger, { value: "documents" }, "Documents"),
                React.createElement(SegmentedControl.Trigger, { value: "settings" }, "Settings")),
            React.createElement("div", { style: { padding: '12px 16px 8px 16px' } },
                React.createElement(SegmentedControl.Content, { value: "account" },
                    React.createElement(Text, { size: "2" }, "Your account.")),
                React.createElement(SegmentedControl.Content, { value: "documents" },
                    React.createElement(Text, { size: "2" }, "Your documents.")),
                React.createElement(SegmentedControl.Content, { value: "settings" },
                    React.createElement(Text, { size: "2" }, "Your profile.")))))),
};
export const InDrawer = {
    name: 'In a drawer',
    render: (args) => (React.createElement(Drawer.Root, { defaultOpen: true },
        React.createElement(Drawer.Trigger, null,
            React.createElement(Button, null, "View users")),
        React.createElement(Drawer.Content, { style: { width: 400 } },
            React.createElement(Drawer.Header, null,
                React.createElement(Drawer.Title, null, "Users")),
            React.createElement(Drawer.Body, null,
                React.createElement(SegmentedControl.Root, { defaultValue: "account" },
                    React.createElement(SegmentedControl.List, { ...args },
                        React.createElement(SegmentedControl.Trigger, { value: "account" }, "Account"),
                        React.createElement(SegmentedControl.Trigger, { value: "documents" }, "Documents"),
                        React.createElement(SegmentedControl.Trigger, { value: "settings" }, "Settings")),
                    React.createElement("div", { style: { padding: '12px 16px 8px 16px' } },
                        React.createElement(SegmentedControl.Content, { value: "account" },
                            React.createElement(Text, { size: "2" }, "Make changes to your account.")),
                        React.createElement(SegmentedControl.Content, { value: "documents" },
                            React.createElement(Text, { size: "2" }, "Access and update your documents.")),
                        React.createElement(SegmentedControl.Content, { value: "settings" },
                            React.createElement(Text, { size: "2" }, "Edit your profile.")))))))),
};
