// frosted-ui/dist/cjs/components/aspect-ratio/aspect-ratio.stories.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Default = void 0;
const tslib_1 = require("tslib");
const react_1 = tslib_1.__importDefault(require("react"));
const __1 = require("..");
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Components/AspectRatio',
    component: __1.AspectRatio.Root,
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
    args: {
        children: (react_1.default.createElement("img", { className: "Image", src: "https://images.unsplash.com/photo-1535025183041-0991a977e25b?w=300&dpr=2&q=80", alt: "Landscape photograph by Tobias Tullius", style: {
                objectFit: 'cover',
                width: '100%',
                height: '100%',
                borderRadius: 'var(--radius-2)',
            } })),
    },
};
exports.default = meta;
// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
exports.Default = {
    args: {
        ratio: 1.5,
    },
    render: (args) => (react_1.default.createElement("div", { style: { width: 300 } },
        react_1.default.createElement(__1.AspectRatio.Root, { ...args }))),
};
