// frosted-ui/dist/esm/components/card/card.stories.js
import React from 'react';
import { Avatar, Card, Code, Inset, Text, cardPropDefs } from '..';
const CardContentExample = () => (React.createElement("div", { style: { display: 'flex', gap: 'var(--space-3)', alignItems: 'center' } },
    React.createElement(Avatar, { size: "3", fallback: "IM", color: "indigo" }),
    React.createElement("div", null,
        React.createElement(Text, { as: "div", size: "2", weight: "bold" }, "Ilya Miskov"),
        React.createElement(Text, { as: "div", size: "2", color: "gray" }, "I love how we have the freedom to explore skeuomorphism"))));
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Components/Card',
    component: Card,
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
};
export default meta;
// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
export const Default = {
    args: {
        children: React.createElement(CardContentExample, null),
        size: cardPropDefs.size.default,
    },
};
export const Size = {
    args: {
        size: cardPropDefs.size.default,
    },
    render: (args) => (React.createElement("div", { style: { display: 'flex', gap: 'var(--space-3)', flexDirection: 'column' } },
        React.createElement(Card, { ...args, size: "1", style: { width: 350 } },
            React.createElement("div", { style: { display: 'flex', gap: 'var(--space-3)', alignItems: 'center' } },
                React.createElement(Avatar, { size: "3", fallback: "T", color: "indigo" }),
                React.createElement("div", null,
                    React.createElement(Text, { as: "div", size: "2", weight: "bold" }, "Teodros Girmay"),
                    React.createElement(Text, { as: "div", size: "2", color: "gray" }, "Engineering")))),
        React.createElement(Card, { ...args, size: "2", style: { width: 425 } },
            React.createElement("div", { style: { display: 'flex', gap: 'var(--space-4)', alignItems: 'center' } },
                React.createElement(Avatar, { size: "4", fallback: "T", color: "indigo" }),
                React.createElement("div", null,
                    React.createElement(Text, { as: "div", weight: "bold" }, "Teodros Girmay"),
                    React.createElement(Text, { as: "div", color: "gray" }, "Engineering")))),
        React.createElement(Card, { ...args, size: "3", style: { width: 500 } },
            React.createElement("div", { style: { display: 'flex', gap: 'var(--space-4)', alignItems: 'center' } },
                React.createElement(Avatar, { size: "5", fallback: "T", color: "indigo" }),
                React.createElement("div", null,
                    React.createElement(Text, { as: "div", size: "4", weight: "bold" }, "Teodros Girmay"),
                    React.createElement(Text, { as: "div", size: "4", color: "gray" }, "Engineering")))),
        React.createElement(Card, { ...args, size: "4", style: { width: 500 } },
            React.createElement("div", { style: { display: 'flex', gap: 'var(--space-4)', alignItems: 'center' } },
                React.createElement(Avatar, { size: "5", fallback: "T", color: "indigo" }),
                React.createElement("div", null,
                    React.createElement(Text, { as: "div", size: "4", weight: "bold" }, "Teodros Girmay"),
                    React.createElement(Text, { as: "div", size: "4", color: "gray" }, "Engineering")))),
        React.createElement(Card, { ...args, size: "5", style: { width: 500 } },
            React.createElement("div", { style: { display: 'flex', gap: 'var(--space-4)', alignItems: 'center' } },
                React.createElement(Avatar, { size: "5", fallback: "T", color: "indigo" }),
                React.createElement("div", null,
                    React.createElement(Text, { as: "div", size: "4", weight: "bold" }, "Teodros Girmay"),
                    React.createElement(Text, { as: "div", size: "4", color: "gray" }, "Engineering")))))),
};
export const Variant = {
    args: {
        children: React.createElement(CardContentExample, null),
    },
    render: ({ children, ...args }) => (React.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-6)' } },
        React.createElement(Card, { ...args, variant: "classic" }, children),
        React.createElement(Card, { ...args, variant: "ghost" }, children),
        React.createElement(Card, { ...args, variant: "surface" }, children))),
};
export const InsetContent = {
    name: 'Inset Content',
    render: (args) => (React.createElement(Card, { size: "2", style: { maxWidth: 240 }, ...args },
        React.createElement(Inset, { clip: "padding-box", side: "top", pb: "current" },
            React.createElement("img", { src: "https://images.unsplash.com/photo-1617050318658-a9a3175e34cb?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=600&q=80", alt: "Bold typography", style: {
                    display: 'block',
                    objectFit: 'cover',
                    width: '100%',
                    height: 140,
                    backgroundColor: 'var(--gray-5)',
                } })),
        React.createElement(Text, { as: "p", size: "3" },
            "Use ",
            React.createElement(Code, null, '<Inset />'),
            " component to align content flush with the sides of the card."))),
};
export const AsAnotherElement = {
    name: 'As another element',
    render: (args) => (React.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-6)' } },
        React.createElement("div", null,
            React.createElement(Text, null,
                "Use the ",
                React.createElement(Code, null, "asChild"),
                " prop to render the card as a link or a button. This prop adds styles for the interactive states, like hover and focus.")),
        React.createElement("div", null,
            React.createElement(Card, { ...args, asChild: true, style: { maxWidth: 350 } },
                React.createElement("a", { href: "#" },
                    React.createElement(CardContentExample, null)))))),
};
