// frosted-ui/dist/esm/components/tabs/tabs.stories.js
import React from 'react';
import { Tabs, Text, tabsListPropDefs } from '..';
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Components/Tabs',
    component: Tabs.List,
    args: {
        size: tabsListPropDefs.size.default,
    },
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
};
export default meta;
// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
// TODO: Tabs content jumping in size due to font-weight change
// This doesn't happen in Frosted UI (might be due to a font in use?)
export const Default = {
    render: (args) => (React.createElement("div", { style: { width: 600 } },
        React.createElement(Tabs.Root, { defaultValue: "account" },
            React.createElement(Tabs.List, { ...args },
                React.createElement(Tabs.Trigger, { value: "account" }, "Account"),
                React.createElement(Tabs.Trigger, { value: "documents" }, "Documents"),
                React.createElement(Tabs.Trigger, { value: "settings" }, "Settings")),
            React.createElement("div", { style: { padding: '12px 16px 8px 16px' } },
                React.createElement(Tabs.Content, { value: "account" },
                    React.createElement(Text, { size: "2" }, "Make changes to your account.")),
                React.createElement(Tabs.Content, { value: "documents" },
                    React.createElement(Text, { size: "2" }, "Access and update your documents.")),
                React.createElement(Tabs.Content, { value: "settings" },
                    React.createElement(Text, { size: "2" }, "Edit your profile or update contact information.")))))),
};
