// frosted-ui/dist/cjs/components/select/select.stories.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AlternativePositioning = exports.HighContrast = exports.Placeholder = exports.Color = exports.TriggerVariant = exports.Size = exports.Default = void 0;
const tslib_1 = require("tslib");
const react_1 = tslib_1.__importDefault(require("react"));
const __1 = require("..");
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Controls/Select',
    component: __1.Select.Trigger,
    args: {},
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
};
exports.default = meta;
exports.Default = {
    args: {
        variant: __1.selectTriggerPropDefs.variant.default,
        color: __1.selectTriggerPropDefs.color.default,
    },
    render: (args) => (react_1.default.createElement(__1.Select.Root, { defaultValue: "apple" },
        react_1.default.createElement(__1.Select.Trigger, { ...args }),
        react_1.default.createElement(__1.Select.Content, null,
            react_1.default.createElement(__1.Select.Group, null,
                react_1.default.createElement(__1.Select.Label, null, "Fruits"),
                react_1.default.createElement(__1.Select.Item, { value: "orange" }, "Orange"),
                react_1.default.createElement(__1.Select.Item, { value: "apple" }, "Apple"),
                react_1.default.createElement(__1.Select.Item, { value: "grape", disabled: true }, "Grape")),
            react_1.default.createElement(__1.Select.Separator, null),
            react_1.default.createElement(__1.Select.Group, null,
                react_1.default.createElement(__1.Select.Label, null, "Vegetables"),
                react_1.default.createElement(__1.Select.Item, { value: "carrot" }, "Carrot"),
                react_1.default.createElement(__1.Select.Item, { value: "potato" }, "Potato"))))),
};
exports.Size = {
    args: {
        variant: __1.selectTriggerPropDefs.variant.default,
        color: __1.selectTriggerPropDefs.color.default,
    },
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-3)', alignItems: 'center' } },
        react_1.default.createElement(__1.Select.Root, { size: "1", defaultValue: "apple" },
            react_1.default.createElement(__1.Select.Trigger, { ...args }),
            react_1.default.createElement(__1.Select.Content, null,
                react_1.default.createElement(__1.Select.Item, { value: "apple" }, "Apple"),
                react_1.default.createElement(__1.Select.Item, { value: "orange" }, "Orange"))),
        react_1.default.createElement(__1.Select.Root, { size: "2", defaultValue: "apple" },
            react_1.default.createElement(__1.Select.Trigger, { ...args }),
            react_1.default.createElement(__1.Select.Content, null,
                react_1.default.createElement(__1.Select.Item, { value: "apple" }, "Apple"),
                react_1.default.createElement(__1.Select.Item, { value: "orange" }, "Orange"))),
        react_1.default.createElement(__1.Select.Root, { size: "3", defaultValue: "apple" },
            react_1.default.createElement(__1.Select.Trigger, { ...args }),
            react_1.default.createElement(__1.Select.Content, null,
                react_1.default.createElement(__1.Select.Item, { value: "apple" }, "Apple"),
                react_1.default.createElement(__1.Select.Item, { value: "orange" }, "Orange"))))),
};
exports.TriggerVariant = {
    args: {
        variant: __1.selectTriggerPropDefs.variant.default,
        color: __1.selectTriggerPropDefs.color.default,
    },
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-3)', alignItems: 'center', flexDirection: 'column' } },
        react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-3)', alignItems: 'center' } },
            react_1.default.createElement(__1.Select.Root, { defaultValue: "apple" },
                react_1.default.createElement(__1.Select.Trigger, { ...args, variant: "surface" }),
                react_1.default.createElement(__1.Select.Content, null,
                    react_1.default.createElement(__1.Select.Item, { value: "apple" }, "Apple"),
                    react_1.default.createElement(__1.Select.Item, { value: "orange" }, "Orange"))),
            react_1.default.createElement(__1.Select.Root, { defaultValue: "apple" },
                react_1.default.createElement(__1.Select.Trigger, { ...args, variant: "soft" }),
                react_1.default.createElement(__1.Select.Content, null,
                    react_1.default.createElement(__1.Select.Item, { value: "apple" }, "Apple"),
                    react_1.default.createElement(__1.Select.Item, { value: "orange" }, "Orange"))),
            react_1.default.createElement(__1.Select.Root, { defaultValue: "apple" },
                react_1.default.createElement(__1.Select.Trigger, { ...args, variant: "ghost" }),
                react_1.default.createElement(__1.Select.Content, null,
                    react_1.default.createElement(__1.Select.Item, { value: "apple" }, "Apple"),
                    react_1.default.createElement(__1.Select.Item, { value: "orange" }, "Orange")))),
        react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-3)', alignItems: 'center' } },
            react_1.default.createElement(__1.Select.Root, { defaultValue: "apple", disabled: true },
                react_1.default.createElement(__1.Select.Trigger, { ...args, variant: "surface" }),
                react_1.default.createElement(__1.Select.Content, null,
                    react_1.default.createElement(__1.Select.Item, { value: "apple" }, "Apple"),
                    react_1.default.createElement(__1.Select.Item, { value: "orange" }, "Orange"))),
            react_1.default.createElement(__1.Select.Root, { defaultValue: "apple", disabled: true },
                react_1.default.createElement(__1.Select.Trigger, { ...args, variant: "soft" }),
                react_1.default.createElement(__1.Select.Content, null,
                    react_1.default.createElement(__1.Select.Item, { value: "apple" }, "Apple"),
                    react_1.default.createElement(__1.Select.Item, { value: "orange" }, "Orange"))),
            react_1.default.createElement(__1.Select.Root, { defaultValue: "apple", disabled: true },
                react_1.default.createElement(__1.Select.Trigger, { ...args, variant: "ghost" }),
                react_1.default.createElement(__1.Select.Content, null,
                    react_1.default.createElement(__1.Select.Item, { value: "apple" }, "Apple"),
                    react_1.default.createElement(__1.Select.Item, { value: "orange" }, "Orange")))))),
};
exports.Color = {
    args: {
        variant: __1.selectTriggerPropDefs.variant.default,
        color: __1.selectTriggerPropDefs.color.default,
    },
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-3)' } },
        react_1.default.createElement(__1.Select.Root, { defaultValue: "apple" },
            react_1.default.createElement(__1.Select.Trigger, { ...args, color: "indigo" }),
            react_1.default.createElement(__1.Select.Content, null,
                react_1.default.createElement(__1.Select.Item, { value: "apple" }, "Apple"),
                react_1.default.createElement(__1.Select.Item, { value: "orange" }, "Orange"))),
        react_1.default.createElement(__1.Select.Root, { defaultValue: "apple" },
            react_1.default.createElement(__1.Select.Trigger, { ...args, color: "cyan" }),
            react_1.default.createElement(__1.Select.Content, null,
                react_1.default.createElement(__1.Select.Item, { value: "apple" }, "Apple"),
                react_1.default.createElement(__1.Select.Item, { value: "orange" }, "Orange"))),
        react_1.default.createElement(__1.Select.Root, { defaultValue: "apple" },
            react_1.default.createElement(__1.Select.Trigger, { ...args, color: "orange" }),
            react_1.default.createElement(__1.Select.Content, null,
                react_1.default.createElement(__1.Select.Item, { value: "apple" }, "Apple"),
                react_1.default.createElement(__1.Select.Item, { value: "orange" }, "Orange"))),
        react_1.default.createElement(__1.Select.Root, { defaultValue: "apple" },
            react_1.default.createElement(__1.Select.Trigger, { ...args, color: "crimson" }),
            react_1.default.createElement(__1.Select.Content, null,
                react_1.default.createElement(__1.Select.Item, { value: "apple" }, "Apple"),
                react_1.default.createElement(__1.Select.Item, { value: "orange" }, "Orange"))))),
};
exports.Placeholder = {
    args: {
        variant: __1.selectTriggerPropDefs.variant.default,
        color: __1.selectTriggerPropDefs.color.default,
    },
    render: (args) => (react_1.default.createElement(__1.Select.Root, null,
        react_1.default.createElement(__1.Select.Trigger, { ...args, placeholder: "Pick a fruit" }),
        react_1.default.createElement(__1.Select.Content, null,
            react_1.default.createElement(__1.Select.Group, null,
                react_1.default.createElement(__1.Select.Label, null, "Fruits"),
                react_1.default.createElement(__1.Select.Item, { value: "orange" }, "Orange"),
                react_1.default.createElement(__1.Select.Item, { value: "apple" }, "Apple"),
                react_1.default.createElement(__1.Select.Item, { value: "grape", disabled: true }, "Grape")),
            react_1.default.createElement(__1.Select.Separator, null),
            react_1.default.createElement(__1.Select.Group, null,
                react_1.default.createElement(__1.Select.Label, null, "Vegetables"),
                react_1.default.createElement(__1.Select.Item, { value: "carrot" }, "Carrot"),
                react_1.default.createElement(__1.Select.Item, { value: "potato" }, "Potato"))))),
};
exports.HighContrast = {
    name: 'High Contrast',
    args: {
        variant: __1.selectTriggerPropDefs.variant.default,
        color: __1.selectTriggerPropDefs.color.default,
    },
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-3)' } },
        react_1.default.createElement(__1.Select.Root, { defaultValue: "apple" },
            react_1.default.createElement(__1.Select.Trigger, { ...args }),
            react_1.default.createElement(__1.Select.Content, null,
                react_1.default.createElement(__1.Select.Item, { value: "apple" }, "Apple"),
                react_1.default.createElement(__1.Select.Item, { value: "orange" }, "Orange"))),
        react_1.default.createElement(__1.Select.Root, { defaultValue: "apple" },
            react_1.default.createElement(__1.Select.Trigger, { ...args }),
            react_1.default.createElement(__1.Select.Content, { highContrast: true },
                react_1.default.createElement(__1.Select.Item, { value: "apple" }, "Apple"),
                react_1.default.createElement(__1.Select.Item, { value: "orange" }, "Orange"))))),
};
exports.AlternativePositioning = {
    name: 'Alternative positioning',
    args: {
        variant: __1.selectTriggerPropDefs.variant.default,
        color: __1.selectTriggerPropDefs.color.default,
    },
    render: (args) => (react_1.default.createElement(__1.Select.Root, { defaultValue: "apple" },
        react_1.default.createElement(__1.Select.Trigger, { ...args }),
        react_1.default.createElement(__1.Select.Content, { position: "popper" },
            react_1.default.createElement(__1.Select.Item, { value: "apple" }, "Apple"),
            react_1.default.createElement(__1.Select.Item, { value: "orange" }, "Orange")))),
};
