// frosted-ui/dist/cjs/components/data-list/data-list.props.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.dataListRootPropDefs = exports.dataListLabelPropDefs = exports.dataListItemPropDefs = void 0;
const helpers_1 = require("../../helpers");
const alignValues = ['start', 'center', 'end', 'baseline', 'stretch'];
const orientationValues = ['horizontal', 'vertical'];
const sizes = ['1', '2', '3'];
const dataListRootPropDefs = {
    orientation: {
        type: 'enum',
        values: orientationValues,
        default: 'horizontal',
    },
    size: {
        type: 'enum',
        values: sizes,
        default: '2',
    },
    trim: {
        ...helpers_1.trimProp,
    },
};
exports.dataListRootPropDefs = dataListRootPropDefs;
const dataListItemPropDefs = {
    align: {
        type: 'enum',
        values: alignValues,
        default: undefined,
    },
};
exports.dataListItemPropDefs = dataListItemPropDefs;
const dataListLabelPropDefs = {
    color: helpers_1.colorProp,
    highContrast: helpers_1.highContrastProp,
};
exports.dataListLabelPropDefs = dataListLabelPropDefs;
