// frosted-ui/dist/esm/components/radio-button-group/radio-button-group.stories.js
import React from 'react';
import { Avatar, Card, RadioButtonGroup, Text, Tooltip, radioGroupPropDefs } from '..';
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Controls/RadioButtonGroup',
    component: RadioButtonGroup.Root,
    args: {
        disabled: false,
    },
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
};
export default meta;
// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
export const Default = {
    args: {
        color: radioGroupPropDefs.color.default,
        highContrast: radioGroupPropDefs.highContrast.default,
    },
    render: (args) => (React.createElement(RadioButtonGroup.Root, { defaultValue: "1", ...args },
        React.createElement("div", { style: { display: 'flex', gap: 'var(--space-2)', flexDirection: 'row' } },
            React.createElement(RadioButtonGroup.Item, { value: "1" },
                React.createElement("div", { style: {
                        display: 'flex',
                        padding: '8px 40px 8px 24px',
                        borderRadius: 8,
                        border: '1px solid var(--gray-a7)',
                        gap: 'var(--space-2)',
                        alignItems: 'center',
                    } },
                    React.createElement(RadioButtonGroup.Icon, { style: { marginLeft: -12 } }),
                    React.createElement(Text, null, "One"))),
            React.createElement(RadioButtonGroup.Item, { value: "2" },
                React.createElement("div", { style: {
                        display: 'flex',
                        padding: '8px 40px 8px 24px',
                        borderRadius: 8,
                        border: '1px solid var(--gray-a7)',
                        gap: 'var(--space-2)',
                        alignItems: 'center',
                    } },
                    React.createElement(RadioButtonGroup.Icon, { style: { marginLeft: -12 } }),
                    React.createElement(Text, null, "Two"))),
            React.createElement(RadioButtonGroup.Item, { value: "3" },
                React.createElement("div", { style: {
                        display: 'flex',
                        padding: '8px 40px 8px 24px',
                        borderRadius: 8,
                        border: '1px solid var(--gray-a7)',
                        gap: 'var(--space-2)',
                        alignItems: 'center',
                    } },
                    React.createElement(RadioButtonGroup.Icon, { style: { marginLeft: -12 } }),
                    React.createElement(Text, null, "Three")))))),
};
export const HighContrast = {
    args: {
        color: radioGroupPropDefs.color.default,
        highContrast: true,
    },
    render: (args) => (React.createElement(RadioButtonGroup.Root, { defaultValue: "1", ...args },
        React.createElement("div", { style: { display: 'flex', gap: 'var(--space-2)', flexDirection: 'row' } },
            React.createElement(Tooltip, { content: "Lime" },
                React.createElement("div", null,
                    React.createElement(RadioButtonGroup.Item, { value: "1" },
                        React.createElement("div", { style: {
                                width: 48,
                                height: 48,
                                borderRadius: 16,
                                background: 'var(--lime-9)',
                            } })))),
            React.createElement(Tooltip, { content: "Teal" },
                React.createElement("div", null,
                    React.createElement(RadioButtonGroup.Item, { value: "2" },
                        React.createElement("div", { style: {
                                width: 48,
                                height: 48,
                                borderRadius: 16,
                                background: 'var(--teal-9)',
                            } })))),
            React.createElement(Tooltip, { content: "Gold" },
                React.createElement("div", null,
                    React.createElement(RadioButtonGroup.Item, { value: "3" },
                        React.createElement("div", { style: {
                                width: 48,
                                height: 48,
                                borderRadius: 16,
                                background: 'var(--gold-9)',
                            } }))))))),
};
export const WithCard = {
    args: {
        color: radioGroupPropDefs.color.default,
        highContrast: radioGroupPropDefs.highContrast.default,
    },
    render: (args) => (React.createElement(RadioButtonGroup.Root, { defaultValue: "1", ...args },
        React.createElement("div", { style: { display: 'flex', gap: 'var(--space-2)', flexDirection: 'row' } },
            React.createElement(RadioButtonGroup.Item, { value: "1" },
                React.createElement(Card, { size: "2", variant: "classic" },
                    React.createElement("div", { style: { display: 'flex', gap: 'var(--space-3)', alignItems: 'center' } },
                        React.createElement(Avatar, { size: "3", fallback: "AB", color: "lime" }),
                        React.createElement("div", null,
                            React.createElement(Text, { as: "div", size: "2", weight: "bold" }, "Artur Bie\u0144"),
                            React.createElement(Text, { as: "div", size: "2", color: "gray" }, "UI engineer"))))),
            React.createElement(RadioButtonGroup.Item, { value: "2" },
                React.createElement(Card, { size: "2", variant: "classic" },
                    React.createElement("div", { style: { display: 'flex', gap: 'var(--space-3)', alignItems: 'center' } },
                        React.createElement(Avatar, { size: "3", fallback: "IM", color: "sky" }),
                        React.createElement("div", null,
                            React.createElement(Text, { as: "div", size: "2", weight: "bold" }, "Ilya Miskov"),
                            React.createElement(Text, { as: "div", size: "2", color: "gray" }, "Designer"))))),
            React.createElement(RadioButtonGroup.Item, { value: "3" },
                React.createElement(Card, { size: "2", variant: "classic" },
                    React.createElement("div", { style: { display: 'flex', gap: 'var(--space-3)', alignItems: 'center' } },
                        React.createElement(Avatar, { size: "3", fallback: "SS", color: "orange" }),
                        React.createElement("div", null,
                            React.createElement(Text, { as: "div", size: "2", weight: "bold" }, "Steven Schwartz"),
                            React.createElement(Text, { as: "div", size: "2", color: "gray" }, "CEO")))))))),
};
export const Color = {
    render: (args) => (React.createElement("div", { style: { display: 'flex', gap: 'var(--space-2)', flexDirection: 'column' } }, ['indigo', 'cyan', 'orange', 'lime'].map((color) => (React.createElement(RadioButtonGroup.Root, { ...args, color: color, key: color, defaultValue: "1" },
        React.createElement("div", { style: { display: 'flex', gap: 'var(--space-2)' } }, ['1', '2', '3'].map((value) => (React.createElement(RadioButtonGroup.Item, { value: value, key: value },
            React.createElement(Card, { size: "3", variant: "classic", style: {
                    width: 100,
                    height: 100,
                    '--card-border-radius': '20px',
                } },
                React.createElement("div", { style: {
                        display: 'flex',
                        alignItems: 'center',
                        justifyContent: 'center',
                        height: '100%',
                        width: '100%',
                    } }, value),
                React.createElement(RadioButtonGroup.Icon, { style: {
                        position: 'absolute',
                        top: 12,
                        right: 12,
                    } }))))))))))),
};
