// frosted-ui/dist/cjs/components/heading/heading.stories.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.HighContrast = exports.Trim = exports.Align = exports.Color = exports.Size = exports.Default = void 0;
const tslib_1 = require("tslib");
const react_1 = tslib_1.__importDefault(require("react"));
const heading_1 = require("../../../src/components/heading");
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Typography/Heading',
    component: heading_1.Heading,
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
};
exports.default = meta;
// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
exports.Default = {
    args: {
        children: 'The quick brown fox jumps over the lazy dog.',
        size: heading_1.headingPropDefs.size.default,
    },
};
exports.Size = {
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-3)' } },
        react_1.default.createElement(heading_1.Heading, { ...args, size: "0" }, "The quick brown fox jumps over the lazy dog."),
        react_1.default.createElement(heading_1.Heading, { ...args, size: "1" }, "The quick brown fox jumps over the lazy dog."),
        react_1.default.createElement(heading_1.Heading, { ...args, size: "2" }, "The quick brown fox jumps over the lazy dog."),
        react_1.default.createElement(heading_1.Heading, { ...args, size: "3" }, "The quick brown fox jumps over the lazy dog."),
        react_1.default.createElement(heading_1.Heading, { ...args, size: "4" }, "The quick brown fox jumps over the lazy dog."),
        react_1.default.createElement(heading_1.Heading, { ...args, size: "5" }, "The quick brown fox jumps over the lazy dog."),
        react_1.default.createElement(heading_1.Heading, { ...args, size: "6" }, "The quick brown fox jumps over the lazy dog."),
        react_1.default.createElement(heading_1.Heading, { ...args, size: "7" }, "The quick brown fox jumps over the lazy dog."),
        react_1.default.createElement(heading_1.Heading, { ...args, size: "8" }, "The quick brown fox jumps over the lazy dog."),
        react_1.default.createElement(heading_1.Heading, { ...args, size: "9" }, "The quick brown fox jumps over the lazy dog."))),
};
exports.Color = {
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-3)' } },
        react_1.default.createElement(heading_1.Heading, { ...args, color: "indigo" }, "The quick brown fox jumps over the lazy dog."),
        react_1.default.createElement(heading_1.Heading, { ...args, color: "cyan" }, "The quick brown fox jumps over the lazy dog."),
        react_1.default.createElement(heading_1.Heading, { ...args, color: "orange" }, "The quick brown fox jumps over the lazy dog."),
        react_1.default.createElement(heading_1.Heading, { ...args, color: "crimson" }, "The quick brown fox jumps over the lazy dog."))),
};
exports.Align = {
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-3)', width: 500 } },
        react_1.default.createElement(heading_1.Heading, { ...args, align: "left" }, "Left-aligned"),
        react_1.default.createElement(heading_1.Heading, { ...args, align: "center" }, "Center-aligned"),
        react_1.default.createElement(heading_1.Heading, { ...args, align: "right" }, "Right-aligned"))),
};
exports.Trim = {
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-3)' } },
        react_1.default.createElement(heading_1.Heading, { ...args, trim: "normal", style: {
                background: 'var(--gray-a2)',
                borderTop: '1px dashed var(--gray-a7)',
                borderBottom: '1px dashed var(--gray-a7)',
            } }, "Without trim"),
        react_1.default.createElement(heading_1.Heading, { ...args, trim: "both", style: {
                background: 'var(--gray-a2)',
                borderTop: '1px dashed var(--gray-a7)',
                borderBottom: '1px dashed var(--gray-a7)',
            } }, "With trim"))),
};
exports.HighContrast = {
    name: 'High Contrast',
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-3)' } },
        react_1.default.createElement(heading_1.Heading, { ...args, highContrast: true, color: "indigo" }, "The quick brown fox jumps over the lazy dog."),
        react_1.default.createElement(heading_1.Heading, { ...args, highContrast: true, color: "cyan" }, "The quick brown fox jumps over the lazy dog."),
        react_1.default.createElement(heading_1.Heading, { ...args, highContrast: true, color: "orange" }, "The quick brown fox jumps over the lazy dog."),
        react_1.default.createElement(heading_1.Heading, { ...args, highContrast: true, color: "crimson" }, "The quick brown fox jumps over the lazy dog."))),
};
