// frosted-ui/dist/esm/components/scroll-area/scroll-area.stories.js
import React from 'react';
import { Heading, ScrollArea, Text, scrollAreaPropDefs } from '..';
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Components/ScrollArea',
    component: ScrollArea,
    args: {
        type: 'always',
    },
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
};
export default meta;
// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
export const Default = {
    args: {
        size: scrollAreaPropDefs.size.default,
        scrollbars: 'vertical',
    },
    render: (args) => (React.createElement(ScrollArea, { type: "always", scrollbars: "vertical", style: { height: 180 }, ...args },
        React.createElement("div", { style: { padding: '8px 8px 54px 8px' } },
            React.createElement(Heading, { size: "4", style: { marginBottom: 8 }, trim: "start" }, "Principles of the typographic craft"),
            React.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-4)' } },
                React.createElement(Text, { as: "p" }, "Three fundamental aspects of typography are legibility, readability, and aesthetics. Although in a non-technical sense \"legible\" and \"readable\" are often used synonymously, typographically they are separate but related concepts."),
                React.createElement(Text, { as: "p" }, "Legibility describes how easily individual characters can be distinguished from one another. It is described by Walter Tracy as \"the quality of being decipherable and recognisable\". For instance, if a \"b\" and an \"h\", or a \"3\" and an \"8\", are difficult to distinguish at small sizes, this is a problem of legibility."),
                React.createElement(Text, { as: "p" }, "Typographers are concerned with legibility insofar as it is their job to select the correct font to use. Brush Script is an example of a font containing many characters that might be difficult to distinguish. The selection of cases influences the legibility of typography because using only uppercase letters (all-caps) reduces legibility."))))),
};
export const Size = {
    args: {
        size: scrollAreaPropDefs.size.default,
        scrollbars: 'vertical',
    },
    render: (args) => (React.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-2)' } },
        React.createElement(ScrollArea, { ...args, size: "1", type: "always", scrollbars: "horizontal", style: { width: 300, height: 12 } },
            React.createElement("div", { style: { width: 800, height: 1 } })),
        React.createElement(ScrollArea, { ...args, size: "2", type: "always", scrollbars: "horizontal", style: { width: 350, height: 16 } },
            React.createElement("div", { style: { width: 900, height: 1 } })),
        React.createElement(ScrollArea, { ...args, size: "3", type: "always", scrollbars: "horizontal", style: { width: 400, height: 20 } },
            React.createElement("div", { style: { width: 1000, height: 1 } })))),
};
export const Scrollbars = {
    args: {
        size: scrollAreaPropDefs.size.default,
        type: 'always',
        scrollbars: 'vertical',
    },
    render: (args) => (React.createElement("div", { style: { display: 'grid', gridTemplateColumns: 'repeat(2, 1fr)', gap: 'var(--space-2)' } },
        React.createElement(ScrollArea, { ...args, type: "always", scrollbars: "vertical", style: { height: 150 } },
            React.createElement("div", { style: {
                    display: 'flex',
                    flexDirection: 'column',
                    gap: 'var(--space-4)',
                    padding: 'var(--space-2)',
                    paddingRight: 'var(--space-8)',
                } },
                React.createElement(Text, { size: "2", trim: "both" }, "Three fundamental aspects of typography are legibility, readability, and aesthetics. Although in a non-technical sense \"legible\" and \"readable\" are often used synonymously, typographically they are separate but related concepts."),
                React.createElement(Text, { size: "2", trim: "both" }, "Legibility describes how easily individual characters can be distinguished from one another. It is described by Walter Tracy as \"the quality of being decipherable and recognisable\". For instance, if a \"b\" and an \"h\", or a \"3\" and an \"8\", are difficult to distinguish at small sizes, this is a problem of legibility."))),
        React.createElement(ScrollArea, { ...args, type: "always", scrollbars: "horizontal", style: { height: 150 } },
            React.createElement("div", { style: { display: 'flex', gap: 'var(--space-4)', padding: 'var(--space-2)', width: 700 } },
                React.createElement(Text, { size: "2", trim: "both" }, "Three fundamental aspects of typography are legibility, readability, and aesthetics. Although in a non-technical sense \"legible\" and \"readable\" are often used synonymously, typographically they are separate but related concepts."),
                React.createElement(Text, { size: "2", trim: "both" }, "Legibility describes how easily individual characters can be distinguished from one another. It is described by Walter Tracy as \"the quality of being decipherable and recognisable\". For instance, if a \"b\" and an \"h\", or a \"3\" and an \"8\", are difficult to distinguish at small sizes, this is a problem of legibility."))))),
};
