// frosted-ui/dist/esm/components/text/text.stories.js
import React from 'react';
import { Checkbox, Code, Em, Kbd, Link, Text, textPropDefs } from '..';
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Typography/Text',
    component: Text,
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
};
export default meta;
// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
export const Default = {
    args: {
        children: 'The quick brown fox jumps over the lazy dog.',
        size: textPropDefs.size.default,
    },
};
export const Size = {
    render: (args) => (React.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 12 } },
        React.createElement(Text, { ...args, size: "0" }, "The quick brown fox jumps over the lazy dog."),
        React.createElement(Text, { ...args, size: "1" }, "The quick brown fox jumps over the lazy dog."),
        React.createElement(Text, { ...args, size: "2" }, "The quick brown fox jumps over the lazy dog."),
        React.createElement(Text, { ...args, size: "3" }, "The quick brown fox jumps over the lazy dog."),
        React.createElement(Text, { ...args, size: "4" }, "The quick brown fox jumps over the lazy dog."),
        React.createElement(Text, { ...args, size: "5" }, "The quick brown fox jumps over the lazy dog."),
        React.createElement(Text, { ...args, size: "6" }, "The quick brown fox jumps over the lazy dog."),
        React.createElement(Text, { ...args, size: "7" }, "The quick brown fox jumps over the lazy dog."),
        React.createElement(Text, { ...args, size: "8" }, "The quick brown fox jumps over the lazy dog."),
        React.createElement(Text, { ...args, size: "9" }, "The quick brown fox jumps over the lazy dog."))),
};
export const Weight = {
    render: (args) => (React.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 12 } },
        React.createElement(Text, { ...args, weight: "light" }, "The quick brown fox jumps over the lazy dog."),
        React.createElement(Text, { ...args, weight: "regular" }, "The quick brown fox jumps over the lazy dog."),
        React.createElement(Text, { ...args, weight: "medium" }, "The quick brown fox jumps over the lazy dog."),
        React.createElement(Text, { ...args, weight: "semi-bold" }, "The quick brown fox jumps over the lazy dog."),
        React.createElement(Text, { ...args, weight: "bold" }, "The quick brown fox jumps over the lazy dog."))),
};
export const Color = {
    render: (args) => (React.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 12 } },
        React.createElement(Text, { ...args, color: "indigo" }, "The quick brown fox jumps over the lazy dog."),
        React.createElement(Text, { ...args, color: "cyan" }, "The quick brown fox jumps over the lazy dog."),
        React.createElement(Text, { ...args, color: "orange" }, "The quick brown fox jumps over the lazy dog."),
        React.createElement(Text, { ...args, color: "crimson" }, "The quick brown fox jumps over the lazy dog."))),
};
export const Align = {
    render: (args) => (React.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 12, width: 500 } },
        React.createElement(Text, { ...args, align: "left" }, "Left-aligned"),
        React.createElement(Text, { ...args, align: "center" }, "Center-aligned"),
        React.createElement(Text, { ...args, align: "right" }, "Right-aligned"))),
};
export const Trim = {
    render: (args) => (React.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 12 } },
        React.createElement(Text, { ...args, trim: "normal", style: {
                background: 'var(--gray-a2)',
                borderTop: '1px dashed var(--gray-a7)',
                borderBottom: '1px dashed var(--gray-a7)',
            } }, "Without trim"),
        React.createElement(Text, { ...args, trim: "both", style: {
                background: 'var(--gray-a2)',
                borderTop: '1px dashed var(--gray-a7)',
                borderBottom: '1px dashed var(--gray-a7)',
            } }, "With trim"))),
};
export const Formatting = {
    render: (args) => (React.createElement(Text, { size: "5", ...args },
        "Look, such a helpful ",
        React.createElement(Link, { href: "#" }, "link"),
        ", an ",
        React.createElement(Em, null, "italic emphasis"),
        ", a piece of computer",
        ' ',
        React.createElement(Code, null, "code"),
        ", and even a hotkey combination ",
        React.createElement(Kbd, null, "\u21E7\u2318A"),
        " within the text.")),
};
export const FormControls = {
    name: 'With form controls',
    render: () => (React.createElement("div", { style: { maxWidth: 300 } },
        React.createElement(Text, { size: "3", asChild: true },
            React.createElement("div", { style: { display: 'flex', gap: 12 } },
                React.createElement(Checkbox, { defaultChecked: true }),
                " Composing Text with a form control like Checkbox, RadioGroup, or Switch automatically centers the control with the first line of text, even when the text is multi-line.")))),
};
export const HighContrast = {
    name: 'High Contrast',
    render: (args) => (React.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 12 } },
        React.createElement(Text, { ...args, highContrast: true, color: "indigo" }, "The quick brown fox jumps over the lazy dog."),
        React.createElement(Text, { ...args, highContrast: true, color: "cyan" }, "The quick brown fox jumps over the lazy dog."),
        React.createElement(Text, { ...args, highContrast: true, color: "orange" }, "The quick brown fox jumps over the lazy dog."),
        React.createElement(Text, { ...args, highContrast: true, color: "crimson" }, "The quick brown fox jumps over the lazy dog."))),
};
