// frosted-ui/dist/cjs/components/inset/inset.stories.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Default = void 0;
const tslib_1 = require("tslib");
const react_1 = tslib_1.__importDefault(require("react"));
const components_1 = require("../../../src/components/");
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Layout/Inset',
    component: components_1.Inset,
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
};
exports.default = meta;
// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
exports.Default = {
    args: {
        side: components_1.insetPropDefs.side.default,
        clip: components_1.insetPropDefs.clip.default,
    },
    render: (args) => (react_1.default.createElement(components_1.Card, { size: "2", style: { maxWidth: 240 } },
        react_1.default.createElement(components_1.Inset, { clip: "padding-box", side: "top", pb: "current", ...args },
            react_1.default.createElement("img", { src: "https://images.unsplash.com/photo-1617050318658-a9a3175e34cb?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=600&q=80", alt: "Bold typography", style: {
                    display: 'block',
                    objectFit: 'cover',
                    width: '100%',
                    height: 140,
                    backgroundColor: 'var(--gray-5)',
                } })),
        react_1.default.createElement(components_1.Text, { as: "p", size: "3", style: { marginTop: 12 } },
            react_1.default.createElement(components_1.Code, null, "Inset"),
            " component applies a negative margin to allow content to bleed into the surrounding container."))),
};
