// frosted-ui/dist/cjs/components/inset/inset.props.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.insetPropDefs = void 0;
const sides = ['all', 'x', 'y', 'top', 'bottom', 'left', 'right'];
const clipValues = ['border-box', 'padding-box'];
const paddingValues = ['current', '0'];
const insetPropDefs = {
    side: { type: 'enum', values: sides, default: 'all' },
    clip: {
        type: 'enum',
        values: clipValues,
        default: 'border-box',
    },
    p: {
        type: 'enum',
        values: paddingValues,
        default: undefined,
    },
    px: {
        type: 'enum',
        values: paddingValues,
        default: undefined,
    },
    py: {
        type: 'enum',
        values: paddingValues,
        default: undefined,
    },
    pt: {
        type: 'enum',
        values: paddingValues,
        default: undefined,
    },
    pr: {
        type: 'enum',
        values: paddingValues,
        default: undefined,
    },
    pb: {
        type: 'enum',
        values: paddingValues,
        default: undefined,
    },
    pl: {
        type: 'enum',
        values: paddingValues,
        default: undefined,
    },
};
exports.insetPropDefs = insetPropDefs;
