// frosted-ui/dist/cjs/components/spinner/spinner.stories.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WithChildren = exports.Sizes = exports.Default = void 0;
const tslib_1 = require("tslib");
const react_1 = tslib_1.__importDefault(require("react"));
const __1 = require("..");
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Components/Spinner',
    component: __1.Spinner,
    args: {
        size: __1.spinnerPropDefs.size.default,
        loading: __1.spinnerPropDefs.loading.default,
    },
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
};
exports.default = meta;
// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
exports.Default = {
    name: 'Default',
    render: (args) => react_1.default.createElement(__1.Spinner, { ...args }),
};
exports.Sizes = {
    name: 'Size',
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', alignItems: 'center', gap: 'var(--space-4)' } },
        react_1.default.createElement(__1.Spinner, { ...args, size: "1" }),
        react_1.default.createElement(__1.Spinner, { ...args, size: "2" }),
        react_1.default.createElement(__1.Spinner, { ...args, size: "3" }),
        react_1.default.createElement(__1.Spinner, { ...args, size: "4" }),
        react_1.default.createElement(__1.Spinner, { ...args, size: "5" }),
        react_1.default.createElement(__1.Spinner, { ...args, size: "6" }))),
};
exports.WithChildren = {
    name: 'With children',
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-5)', maxWidth: 640 } },
        react_1.default.createElement(__1.Text, null,
            "Use the ",
            react_1.default.createElement(__1.Code, null, "loading"),
            " prop to control whether the spinner or its children are displayed. Spinner preserves the dimensions of children when they are hidden and disables interactive elements."),
        react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-4)' } },
            react_1.default.createElement(__1.Spinner, { ...args, loading: true },
                react_1.default.createElement(__1.Switch, { defaultChecked: true })),
            react_1.default.createElement(__1.Spinner, { ...args, loading: false },
                react_1.default.createElement(__1.Switch, { defaultChecked: true }))))),
};
