// frosted-ui/dist/cjs/components/spinner/spinner.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Spinner = void 0;
const tslib_1 = require("tslib");
const classnames_1 = tslib_1.__importDefault(require("classnames"));
const React = tslib_1.__importStar(require("react"));
const spinner_props_1 = require("./spinner.props");
const Spinner = (props) => {
    const { children, className, loading = spinner_props_1.spinnerPropDefs.loading.default, size = spinner_props_1.spinnerPropDefs.size.default, ...spinnerProps } = props;
    if (!loading)
        return children;
    const spinner = (React.createElement("span", { ...spinnerProps, className: (0, classnames_1.default)('fui-Spinner', `fui-r-size-${size}`, className) },
        React.createElement("span", { className: "fui-SpinnerLeaf" }),
        React.createElement("span", { className: "fui-SpinnerLeaf" }),
        React.createElement("span", { className: "fui-SpinnerLeaf" }),
        React.createElement("span", { className: "fui-SpinnerLeaf" }),
        React.createElement("span", { className: "fui-SpinnerLeaf" }),
        React.createElement("span", { className: "fui-SpinnerLeaf" }),
        React.createElement("span", { className: "fui-SpinnerLeaf" }),
        React.createElement("span", { className: "fui-SpinnerLeaf" })));
    if (children === undefined)
        return spinner;
    return (React.createElement("span", { style: {
            position: 'relative',
            display: 'flex',
            alignItems: 'center',
            justifyContent: 'center',
        } },
        React.createElement("span", { "aria-hidden": true, style: { display: 'contents', visibility: 'hidden' }, inert: true }, children),
        React.createElement("span", { style: {
                position: 'absolute',
                display: 'flex',
                alignItems: 'center',
                justifyContent: 'center',
                inset: '0',
            } }, spinner)));
};
exports.Spinner = Spinner;
Spinner.displayName = 'Spinner';
