// frosted-ui/dist/esm/components/drawer/drawer.js
'use client';
import classNames from 'classnames';
import { Dialog as DrawerPrimitive } from 'radix-ui';
import * as React from 'react';
import { Theme } from '../../theme';
import { Heading } from '../heading';
const DrawerRoot = (props) => React.createElement(DrawerPrimitive.Root, { ...props, modal: true });
DrawerRoot.displayName = 'DrawerRoot';
const DrawerTrigger = (props) => React.createElement(DrawerPrimitive.Trigger, { ...props, asChild: true });
DrawerTrigger.displayName = 'DrawerTrigger';
const DrawerContent = (props) => {
    const { className, forceMount, container, ...contentProps } = props;
    return (React.createElement(DrawerPrimitive.Portal, { container: container, forceMount: forceMount },
        React.createElement(Theme, { asChild: true },
            React.createElement(DrawerPrimitive.Overlay, { className: "fui-DialogOverlay" },
                React.createElement(DrawerPrimitive.Content, { ...contentProps, "aria-describedby": undefined, className: classNames('fui-DrawerContent', className) })))));
};
DrawerContent.displayName = 'DrawerContent';
const DrawerTitle = (props) => (React.createElement(DrawerPrimitive.Title, { asChild: true },
    React.createElement(Heading, { size: "4", weight: "semi-bold", ...props })));
DrawerTitle.displayName = 'DrawerTitle';
const DrawerClose = (props) => React.createElement(DrawerPrimitive.Close, { ...props, asChild: true });
DrawerClose.displayName = 'DrawerClose';
const DrawerStickyFooter = ({ children, className, ...props }) => (React.createElement("div", { className: classNames('fui-DrawerStickyFooter', className), ...props }, children));
DrawerStickyFooter.displayName = 'DrawerStickyFooter';
const DrawerHeader = ({ children, className, ...props }) => (React.createElement("div", { className: classNames('fui-DrawerHeader', className), ...props }, children));
DrawerHeader.displayName = 'DrawerHeader';
const DrawerBody = ({ children, className, ...props }) => {
    const localRef = React.useRef(null);
    const contentRef = React.useRef(null);
    return (React.createElement("div", { className: "fui-BodyScrollArea", ...props, ref: localRef },
        React.createElement("div", { className: classNames('fui-DrawerBody', className), ref: contentRef }, children)));
};
DrawerBody.displayName = 'DrawerBody';
export { DrawerBody as Body, DrawerClose as Close, DrawerContent as Content, DrawerHeader as Header, DrawerRoot as Root, DrawerStickyFooter as StickyFooter, DrawerTitle as Title, DrawerTrigger as Trigger, };
