// frosted-ui/dist/cjs/components/slot/slot.stories.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Default = void 0;
const tslib_1 = require("tslib");
const react_1 = tslib_1.__importDefault(require("react"));
const __1 = require("..");
const CustomButton = ({ asChild, ...props }) => {
    const Comp = asChild ? __1.Slot.Root : 'button';
    return (react_1.default.createElement(Comp, { ...props, style: {
            padding: '8px 16px',
            border: '1px solid var(--gray-7)',
        } }));
};
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Utilities/Slot',
    component: __1.Slot.Root,
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
};
exports.default = meta;
// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
exports.Default = {
    args: {},
    render: () => (react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-4)' } },
        react_1.default.createElement(__1.Text, null,
            "Can be used to support your own `asChild` prop. See",
            ' ',
            react_1.default.createElement(__1.Link, { href: "https://www.radix-ui.com/primitives/docs/utilities/slot" }, "Radix UI docs"),
            "."),
        react_1.default.createElement("div", null,
            react_1.default.createElement(CustomButton, { asChild: true },
                react_1.default.createElement("a", { href: "/contact" }, "Contact"))))),
};
