// frosted-ui/dist/cjs/components/avatar-group/avatar-group.stories.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Color = exports.Size = exports.Default = void 0;
const tslib_1 = require("tslib");
const react_1 = tslib_1.__importDefault(require("react"));
const react_aria_components_1 = require("react-aria-components");
const __1 = require("..");
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Components/AvatarGroup',
    component: __1.AvatarGroup.Root,
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
};
exports.default = meta;
const availableUsers = [
    {
        name: 'Artur Bień',
        color: 'blue',
        src: 'https://images.unsplash.com/photo-1502823403499-6ccfcf4fb453?&w=256&h=256&q=70&crop=focalpoint&fp-x=0.5&fp-y=0.3&fp-z=1&fit=crop',
    },
    {
        name: 'Cameron Zoub',
        color: 'green',
        src: '',
    },
    {
        name: 'Carlos Whitt',
        color: 'pink',
        src: '',
    },
    {
        name: 'Ilya Miskov',
        color: 'yellow',
        src: '',
    },
    {
        name: 'Guillermo',
        color: 'bronze',
        src: '',
    },
    {
        name: 'Steven Schwartz',
        color: 'lime',
        src: '',
    },
    {
        name: 'FJ',
        color: 'tomato',
        src: '',
    },
];
// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
exports.Default = {
    args: {},
    render: (args) => {
        const [users, setUsers] = react_1.default.useState(availableUsers);
        return (react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-2)' } },
            react_1.default.createElement(__1.AvatarGroup.Root, { ...args, size: "3" }, users.map((user) => (react_1.default.createElement(__1.AvatarGroup.Avatar, { fallback: user.name, color: user.color, key: user.name, src: user.src })))),
            react_1.default.createElement(__1.AvatarGroup.Root, { ...args, size: "9" }, users.map((user) => (react_1.default.createElement(__1.AvatarGroup.Avatar, { fallback: user.name, color: user.color, key: user.name, src: user.src })))),
            react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-2)' } }, availableUsers.map((user) => (react_1.default.createElement(react_aria_components_1.Button, { key: user.name, onPress: () => {
                    setUsers((prev) => prev.map((a) => a.name).includes(user.name)
                        ? prev.filter((u) => u.name !== user.name)
                        : [...prev, user]);
                } },
                "Toggle ",
                user.name))))));
    },
};
exports.Size = {
    args: {},
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', alignItems: 'center', gap: 'var(--space-4)' } },
        react_1.default.createElement(__1.AvatarGroup.Root, { ...args, size: "1" }, availableUsers.map((user) => (react_1.default.createElement(__1.AvatarGroup.Avatar, { fallback: user.name, color: user.color, key: user.name, src: user.src })))),
        react_1.default.createElement(__1.AvatarGroup.Root, { ...args, size: "2" }, availableUsers.map((user) => (react_1.default.createElement(__1.AvatarGroup.Avatar, { fallback: user.name, color: user.color, key: user.name, src: user.src })))),
        react_1.default.createElement(__1.AvatarGroup.Root, { ...args, size: "3" }, availableUsers.map((user) => (react_1.default.createElement(__1.AvatarGroup.Avatar, { fallback: user.name, color: user.color, key: user.name, src: user.src })))),
        react_1.default.createElement(__1.AvatarGroup.Root, { ...args, size: "4" }, availableUsers.map((user) => (react_1.default.createElement(__1.AvatarGroup.Avatar, { fallback: user.name, color: user.color, key: user.name, src: user.src })))),
        react_1.default.createElement(__1.AvatarGroup.Root, { ...args, size: "5" }, availableUsers.map((user) => (react_1.default.createElement(__1.AvatarGroup.Avatar, { fallback: user.name, color: user.color, key: user.name, src: user.src })))),
        react_1.default.createElement(__1.AvatarGroup.Root, { ...args, size: "6" }, availableUsers.map((user) => (react_1.default.createElement(__1.AvatarGroup.Avatar, { fallback: user.name, color: user.color, key: user.name, src: user.src })))),
        react_1.default.createElement(__1.AvatarGroup.Root, { ...args, size: "7" }, availableUsers.map((user) => (react_1.default.createElement(__1.AvatarGroup.Avatar, { fallback: user.name, color: user.color, key: user.name, src: user.src })))),
        react_1.default.createElement(__1.AvatarGroup.Root, { ...args, size: "8" }, availableUsers.map((user) => (react_1.default.createElement(__1.AvatarGroup.Avatar, { fallback: user.name, color: user.color, key: user.name, src: user.src })))),
        react_1.default.createElement(__1.AvatarGroup.Root, { ...args, size: "9" }, availableUsers.map((user) => (react_1.default.createElement(__1.AvatarGroup.Avatar, { fallback: user.name, color: user.color, key: user.name, src: user.src })))))),
};
exports.Color = {
    args: {},
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-2)' } },
        react_1.default.createElement(__1.AvatarGroup.Root, { size: "5", ...args, color: "indigo" }, availableUsers.map((user) => (react_1.default.createElement(__1.AvatarGroup.Avatar, { fallback: user.name, color: user.color, key: user.name, src: user.src })))),
        react_1.default.createElement(__1.AvatarGroup.Root, { size: "5", ...args, color: "cyan" }, availableUsers.map((user) => (react_1.default.createElement(__1.AvatarGroup.Avatar, { fallback: user.name, color: user.color, key: user.name, src: user.src })))),
        react_1.default.createElement(__1.AvatarGroup.Root, { size: "5", ...args, color: "orange" }, availableUsers.map((user) => (react_1.default.createElement(__1.AvatarGroup.Avatar, { fallback: user.name, color: user.color, key: user.name, src: user.src })))),
        react_1.default.createElement(__1.AvatarGroup.Root, { size: "5", ...args, color: "gold" }, availableUsers.map((user) => (react_1.default.createElement(__1.AvatarGroup.Avatar, { fallback: user.name, color: user.color, key: user.name, src: user.src })))))),
};
