// frosted-ui/dist/cjs/components/hover-card/hover-card.stories.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.InsetContent = exports.Size = exports.Default = void 0;
const tslib_1 = require("tslib");
const react_1 = tslib_1.__importDefault(require("react"));
const __1 = require("..");
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Components/HoverCard',
    component: __1.HoverCard.Content,
    args: {},
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
};
exports.default = meta;
// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
exports.Default = {
    args: {
        size: __1.hoverCardContentPropDefs.size.default,
        variant: __1.hoverCardContentPropDefs.variant.default,
    },
    render: (args) => (react_1.default.createElement(__1.Text, null,
        "Follow",
        ' ',
        react_1.default.createElement(__1.HoverCard.Root, null,
            react_1.default.createElement(__1.HoverCard.Trigger, null,
                react_1.default.createElement(__1.Link, { href: "https://github.com/whopio/frosted-ui", target: "_blank" }, "frosted-ui")),
            react_1.default.createElement(__1.HoverCard.Content, { ...args },
                react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-4)' } },
                    react_1.default.createElement(__1.Avatar, { size: "3", fallback: "FUI" }),
                    react_1.default.createElement("div", null,
                        react_1.default.createElement(__1.Heading, { size: "3", as: "h3" }, "Frosted UI"),
                        react_1.default.createElement(__1.Text, { as: "div", size: "2", color: "gray" }, "frosted-ui"),
                        react_1.default.createElement(__1.Text, { as: "div", size: "2", style: { maxWidth: 300, marginTop: 'var(--space-3)' } }, "React components library built on top of Radix Themes."))))),
        ' ',
        "for updates.")),
};
exports.Size = {
    args: {
        variant: __1.hoverCardContentPropDefs.variant.default,
    },
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-4)' } },
        react_1.default.createElement(__1.HoverCard.Root, null,
            react_1.default.createElement(__1.HoverCard.Trigger, null,
                react_1.default.createElement(__1.Link, { href: "#" }, "Size one")),
            react_1.default.createElement(__1.HoverCard.Content, { ...args, size: "1" },
                react_1.default.createElement(__1.Text, { as: "div", size: "1", style: { maxWidth: 325 } },
                    react_1.default.createElement(__1.Strong, null, "Typography"),
                    " is the art and technique of arranging type to make written language legible, readable and appealing when displayed."))),
        react_1.default.createElement(__1.HoverCard.Root, null,
            react_1.default.createElement(__1.HoverCard.Trigger, null,
                react_1.default.createElement(__1.Link, { href: "#" }, "Size two")),
            react_1.default.createElement(__1.HoverCard.Content, { ...args, size: "2" },
                react_1.default.createElement(__1.Text, { as: "div", size: "2", style: { maxWidth: 350 } },
                    react_1.default.createElement(__1.Strong, null, "Typography"),
                    " is the art and technique of arranging type to make written language legible, readable and appealing when displayed."))),
        react_1.default.createElement(__1.HoverCard.Root, null,
            react_1.default.createElement(__1.HoverCard.Trigger, null,
                react_1.default.createElement(__1.Link, { href: "#" }, "Size three")),
            react_1.default.createElement(__1.HoverCard.Content, { ...args, size: "3" },
                react_1.default.createElement(__1.Text, { as: "div", size: "3", style: { maxWidth: 400 } },
                    react_1.default.createElement(__1.Strong, null, "Typography"),
                    " is the art and technique of arranging type to make written language legible, readable and appealing when displayed."))))),
};
exports.InsetContent = {
    name: 'With inset content',
    args: {
        size: __1.hoverCardContentPropDefs.size.default,
        variant: __1.hoverCardContentPropDefs.variant.default,
    },
    render: (args) => (react_1.default.createElement(__1.Text, null,
        "Technology revolutionized",
        ' ',
        react_1.default.createElement(__1.HoverCard.Root, null,
            react_1.default.createElement(__1.HoverCard.Trigger, null,
                react_1.default.createElement(__1.Link, { href: "#" }, "typography")),
            react_1.default.createElement(__1.HoverCard.Content, { ...args },
                react_1.default.createElement("div", { style: { display: 'flex' } },
                    react_1.default.createElement(__1.Inset, { side: "left", pr: "current" },
                        react_1.default.createElement("img", { src: "https://images.unsplash.com/photo-1617050318658-a9a3175e34cb?&auto=format&fit=crop&w=300&q=80", alt: "Bold typography", style: {
                                display: 'block',
                                objectFit: 'cover',
                                height: '100%',
                                width: 150,
                                backgroundColor: 'var(--gray-5)',
                            } })),
                    react_1.default.createElement(__1.Text, { size: "2", style: { maxWidth: 250 }, as: "p" },
                        react_1.default.createElement(__1.Strong, null, "Typography"),
                        " is the art and technique of arranging type to make written language legible, readable and appealing when displayed. The arrangement of type involves selecting typefaces, point sizes, line lengths, line-spacing (leading), and letter-spacing (tracking)\u2026")))),
        ' ',
        "in the latter twentieth century.")),
};
