// frosted-ui/dist/cjs/components/progress/progress.stories.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.HighContrast = exports.Color = exports.Size = exports.Default = void 0;
const tslib_1 = require("tslib");
const react_1 = tslib_1.__importDefault(require("react"));
const __1 = require("..");
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Components/Progress',
    component: __1.Progress,
    args: {
        size: __1.progressPropDefs.size.default,
        color: __1.progressPropDefs.color.default,
        value: 40,
        max: 100,
    },
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
};
exports.default = meta;
// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
exports.Default = {
    args: {
        size: __1.progressPropDefs.size.default,
        color: __1.progressPropDefs.color.default,
    },
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-3)', width: 300 } },
        react_1.default.createElement(__1.Progress, { ...args }))),
};
exports.Size = {
    args: {
        size: __1.progressPropDefs.size.default,
        color: __1.progressPropDefs.color.default,
    },
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-3)', width: 300 } },
        react_1.default.createElement(__1.Progress, { ...args, size: "6" }),
        react_1.default.createElement(__1.Progress, { ...args, size: "5" }),
        react_1.default.createElement(__1.Progress, { ...args, size: "4" }),
        react_1.default.createElement(__1.Progress, { ...args, size: "3" }),
        react_1.default.createElement(__1.Progress, { ...args, size: "2" }),
        react_1.default.createElement(__1.Progress, { ...args, size: "1" }))),
};
exports.Color = {
    args: {
        size: __1.progressPropDefs.size.default,
    },
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-3)', width: 300 } },
        react_1.default.createElement(__1.Progress, { ...args, color: "indigo" }),
        react_1.default.createElement(__1.Progress, { ...args, color: "cyan" }),
        react_1.default.createElement(__1.Progress, { ...args, color: "orange" }),
        react_1.default.createElement(__1.Progress, { ...args, color: "crimson" }))),
};
exports.HighContrast = {
    name: 'High Contrast',
    args: {},
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'row', gap: 'var(--space-2)' } },
        react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-3)', width: 300 } },
            react_1.default.createElement(__1.Progress, { ...args, highContrast: false })),
        react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-3)', width: 300 } },
            react_1.default.createElement(__1.Progress, { ...args, highContrast: true })))),
};
