// frosted-ui/dist/cjs/components/circular-progress/circular-progress.stories.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.HighContrast = exports.Color = exports.Size = exports.Default = void 0;
const tslib_1 = require("tslib");
const react_1 = tslib_1.__importDefault(require("react"));
const __1 = require("..");
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Components/CircularProgress',
    component: __1.CircularProgress,
    args: {
        size: __1.circularProgressPropDefs.size.default,
        color: __1.circularProgressPropDefs.color.default,
        value: 40,
        max: 100,
    },
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
};
exports.default = meta;
// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
exports.Default = {
    args: {
        size: __1.circularProgressPropDefs.size.default,
        color: __1.circularProgressPropDefs.color.default,
    },
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-3)', width: 300 } },
        react_1.default.createElement(__1.CircularProgress, { ...args }))),
};
exports.Size = {
    args: {
        size: __1.circularProgressPropDefs.size.default,
        color: __1.circularProgressPropDefs.color.default,
    },
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-3)', width: 300 } }, __1.circularProgressPropDefs.size.values.map((size) => (react_1.default.createElement(__1.CircularProgress, { ...args, size: size, key: size }))))),
};
exports.Color = {
    args: {
        size: __1.circularProgressPropDefs.size.default,
    },
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-3)', width: 300 } },
        react_1.default.createElement(__1.CircularProgress, { ...args, color: "indigo" }),
        react_1.default.createElement(__1.CircularProgress, { ...args, color: "cyan" }),
        react_1.default.createElement(__1.CircularProgress, { ...args, color: "orange" }),
        react_1.default.createElement(__1.CircularProgress, { ...args, color: "crimson" }))),
};
exports.HighContrast = {
    name: 'High Contrast',
    args: {},
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'row', gap: 'var(--space-2)' } },
        react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-3)', width: 300 } },
            react_1.default.createElement(__1.CircularProgress, { ...args, highContrast: false })),
        react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-3)', width: 300 } },
            react_1.default.createElement(__1.CircularProgress, { ...args, highContrast: true })))),
};
