// frosted-ui/dist/cjs/components/separator/separator.stories.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Orientation = exports.Color = exports.Size = exports.Default = void 0;
const tslib_1 = require("tslib");
const react_1 = tslib_1.__importDefault(require("react"));
const __1 = require("..");
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Layout/Separator',
    component: __1.Separator,
    args: {
        size: __1.separatorPropDefs.size.default,
        color: __1.separatorPropDefs.color.default,
    },
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
};
exports.default = meta;
// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
exports.Default = {
    render: (args) => (react_1.default.createElement(__1.Text, { size: "2" },
        "Tools for building high-quality, accessible UI.",
        react_1.default.createElement(__1.Separator, { ...args, orientation: "horizontal", size: "4", style: { margin: 'var(--space-3) 0' } }),
        react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-3)', alignItems: 'center' } },
            "Themes",
            react_1.default.createElement(__1.Separator, { ...args, orientation: "vertical" }),
            "Primitives",
            react_1.default.createElement(__1.Separator, { ...args, orientation: "vertical" }),
            "Icons",
            react_1.default.createElement(__1.Separator, { ...args, orientation: "vertical" }),
            "Colors"))),
};
exports.Size = {
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-6)' } },
        react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-4)' } },
            react_1.default.createElement(__1.Separator, { ...args, orientation: "horizontal", size: "4" }),
            react_1.default.createElement(__1.Separator, { ...args, orientation: "horizontal", size: "3" }),
            react_1.default.createElement(__1.Separator, { ...args, orientation: "horizontal", size: "2" }),
            react_1.default.createElement(__1.Separator, { ...args, orientation: "horizontal", size: "1" })),
        react_1.default.createElement("div", { style: { display: 'flex', alignItems: 'center', gap: 'var(--space-4)', height: 96 } },
            react_1.default.createElement(__1.Separator, { ...args, orientation: "vertical", size: "4" }),
            react_1.default.createElement(__1.Separator, { ...args, orientation: "vertical", size: "3" }),
            react_1.default.createElement(__1.Separator, { ...args, orientation: "vertical", size: "2" }),
            react_1.default.createElement(__1.Separator, { ...args, orientation: "vertical", size: "1" })))),
};
exports.Color = {
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-3)' } },
        react_1.default.createElement(__1.Separator, { ...args, color: "indigo", size: "3" }),
        react_1.default.createElement(__1.Separator, { ...args, color: "cyan", size: "3" }),
        react_1.default.createElement(__1.Separator, { ...args, color: "orange", size: "3" }),
        react_1.default.createElement(__1.Separator, { ...args, color: "crimson", size: "3" }))),
};
exports.Orientation = {
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', alignItems: 'center', gap: 'var(--space-4)' } },
        react_1.default.createElement(__1.Separator, { ...args, orientation: "horizontal" }),
        react_1.default.createElement(__1.Separator, { ...args, orientation: "vertical" }))),
};
