// frosted-ui/dist/esm/components/filter-chip/filter-chip.stories.js
import React from 'react';
import { FilterChip, filterChipPropDefs } from '..';
const ExampleIcon = ({ size }) => (React.createElement("svg", { width: size, height: size, viewBox: "0 0 20 20", fill: "none", xmlns: "http://www.w3.org/2000/svg" },
    React.createElement("path", { d: "M3 8C3 6 5.70833 3 9.5 3C13.2917 3 16 6 16 8M16 8H13M16 8V8C16.5523 8 17 7.55228 17 7V4", stroke: "currentColor", strokeWidth: "1.5", strokeLinecap: "round", strokeLinejoin: "round" }),
    React.createElement("path", { d: "M17 12C17 14 14.2917 17 10.5 17C6.70833 17 4 14 4 12M4 12H7M4 12V12C3.44772 12 3 12.4477 3 13V16", stroke: "currentColor", strokeWidth: "1.5", strokeLinecap: "round", strokeLinejoin: "round" })));
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Controls/FilterChip',
    component: FilterChip,
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'centered',
    },
    args: {
        children: null,
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
};
export default meta;
// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
export const Default = {
    args: {
        size: filterChipPropDefs.size.default,
        color: filterChipPropDefs.color.default,
    },
    render: (args) => (React.createElement("div", { style: { display: 'flex', gap: 'var(--space-2)', flexDirection: 'row' } },
        React.createElement(FilterChip, { defaultChecked: true, ...args },
            React.createElement(ExampleIcon, { size: 16 }),
            "Default"),
        React.createElement(FilterChip, { defaultChecked: true, disabled: true, ...args },
            React.createElement(ExampleIcon, { size: 16 }),
            "Disabled checked"),
        React.createElement(FilterChip, { defaultChecked: false, disabled: true, ...args },
            React.createElement(ExampleIcon, { size: 16 }),
            "Disabled unchecked"))),
};
export const Size = {
    render: (args) => (React.createElement("div", { style: { display: 'flex', gap: 'var(--space-2)' } },
        React.createElement(FilterChip, { defaultChecked: true, ...args, size: "1" },
            React.createElement(ExampleIcon, { size: 16 }),
            " Size 1"),
        React.createElement(FilterChip, { defaultChecked: true, ...args, size: "2" },
            React.createElement(ExampleIcon, { size: 20 }),
            " Size 2"),
        React.createElement(FilterChip, { defaultChecked: true, ...args, size: "3" },
            React.createElement(ExampleIcon, { size: 24 }),
            " Size 3"))),
};
export const Color = {
    render: (args) => (React.createElement("div", { style: { display: 'flex', gap: 'var(--space-2)' } },
        React.createElement(FilterChip, { ...args, color: "indigo", defaultChecked: true },
            React.createElement(ExampleIcon, { size: 20 }),
            "Indigo"),
        React.createElement(FilterChip, { ...args, color: "cyan", defaultChecked: true },
            React.createElement(ExampleIcon, { size: 20 }),
            "Cyan"),
        React.createElement(FilterChip, { ...args, color: "orange", defaultChecked: true },
            React.createElement(ExampleIcon, { size: 20 }),
            "Orange"),
        React.createElement(FilterChip, { ...args, color: "crimson", defaultChecked: true },
            React.createElement(ExampleIcon, { size: 20 }),
            "Crimson"))),
};
