// frosted-ui/dist/cjs/components/breadcrumbs/breadcrumbs.stories.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Truncated = exports.WithOnClick = exports.WithLinks = void 0;
const tslib_1 = require("tslib");
const react_1 = tslib_1.__importDefault(require("react"));
const __1 = require("..");
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Components/Breadcrumbs',
    component: __1.Breadcrumbs.Root,
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
};
exports.default = meta;
exports.WithLinks = {
    name: 'With links',
    args: {
        color: __1.breadcrumbsPropDefs.color.default,
    },
    render: (args) => (react_1.default.createElement(__1.Breadcrumbs.Root, { ...args },
        react_1.default.createElement(__1.Breadcrumbs.Item, { asChild: true },
            react_1.default.createElement("a", { href: "#" }, "Home")),
        react_1.default.createElement(__1.Breadcrumbs.Item, { asChild: true },
            react_1.default.createElement("a", { href: "#" }, "User Profiles")),
        react_1.default.createElement(__1.Breadcrumbs.Item, null, "Products"))),
};
exports.WithOnClick = {
    name: 'With onClick',
    args: {
        color: __1.breadcrumbsPropDefs.color.default,
    },
    render: (args) => (react_1.default.createElement(__1.Breadcrumbs.Root, { ...args },
        react_1.default.createElement(__1.Breadcrumbs.Item, { onClick: () => alert('Home') }, "Home"),
        react_1.default.createElement(__1.Breadcrumbs.Item, { onClick: () => alert('Products') }, "Products"),
        react_1.default.createElement(__1.Breadcrumbs.Item, { onClick: () => alert('Sneaker Bot') }, "Sneaker Bot"))),
};
exports.Truncated = {
    name: 'Truncated',
    args: {
        color: __1.breadcrumbsPropDefs.color.default,
    },
    render: (args) => (react_1.default.createElement(__1.Breadcrumbs.Root, { ...args },
        react_1.default.createElement(__1.Breadcrumbs.Item, { asChild: true },
            react_1.default.createElement("a", { href: "#" }, "Home")),
        react_1.default.createElement(__1.Breadcrumbs.Dropdown, null,
            react_1.default.createElement(__1.Breadcrumbs.DropdownItem, { asChild: true },
                react_1.default.createElement("a", { href: "#" }, "Products")),
            react_1.default.createElement(__1.Breadcrumbs.DropdownItem, { asChild: true },
                react_1.default.createElement("a", { href: "#" }, "Categories")),
            react_1.default.createElement(__1.Breadcrumbs.DropdownItem, { asChild: true },
                react_1.default.createElement("a", { href: "#" }, "Software"))),
        react_1.default.createElement(__1.Breadcrumbs.Item, { asChild: true },
            react_1.default.createElement("a", { href: "#" }, "Bots")),
        react_1.default.createElement(__1.Breadcrumbs.Item, null, "Sneaker Bot"))),
};
