// frosted-ui/dist/esm/components/data-list/data-list.stories.js
import { Copy12 } from '@frosted-ui/icons';
import React from 'react';
import { Badge, Code, DataList, Heading, IconButton, Link, Separator, Text, Tooltip, dataListRootPropDefs, } from '../../../src/components/';
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Data presentation/DataList',
    component: DataList.Root,
    args: {
        size: dataListRootPropDefs.size.default,
        orientation: dataListRootPropDefs.orientation.default,
        trim: dataListRootPropDefs.trim.default,
    },
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
};
export default meta;
// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
export const Default = {
    render: (args) => (React.createElement("div", null,
        React.createElement(Text, { as: "p", style: { marginBottom: 32 } },
            React.createElement(Code, null, '<DataList />'),
            " component displays metadata as a list of key-value pairs."),
        React.createElement(DataList.Root, { ...args },
            React.createElement(DataList.Item, { align: "center" },
                React.createElement(DataList.Label, null, "Status"),
                React.createElement(DataList.Value, null,
                    React.createElement(Badge, { color: "jade", variant: "soft", size: "1" }, "Active"))),
            React.createElement(DataList.Item, null,
                React.createElement(DataList.Label, null, "ID"),
                React.createElement(DataList.Value, null,
                    React.createElement("div", { style: { display: 'flex', alignItems: 'center', gap: 'var(--space-2)' } },
                        React.createElement(Code, { variant: "ghost" }, "biz_AB23XH123A"),
                        React.createElement(Tooltip, { content: "Copy" },
                            React.createElement(IconButton, { size: "1", "aria-label": "Copy value", color: "gray", variant: "ghost" },
                                React.createElement(Copy12, null)))))),
            React.createElement(DataList.Item, null,
                React.createElement(DataList.Label, null, "Name"),
                React.createElement(DataList.Value, null, "Artur Bie\u0144")),
            React.createElement(DataList.Item, null,
                React.createElement(DataList.Label, null, "Email"),
                React.createElement(DataList.Value, null,
                    React.createElement(Link, { href: "mailto:artur@whop.com" }, "artur@whop.com"))),
            React.createElement(DataList.Item, null,
                React.createElement(DataList.Label, null, "Company"),
                React.createElement(DataList.Value, null,
                    React.createElement(Link, { target: "_blank", href: "https://whop.com" }, "Whop")))))),
};
export const Size = {
    render: (args) => (React.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-8)' } },
        React.createElement(DataList.Root, { ...args, size: "1" },
            React.createElement(DataList.Item, null,
                React.createElement(DataList.Label, null, "Name"),
                React.createElement(DataList.Value, null, "Artur Bie\u0144")),
            React.createElement(DataList.Item, null,
                React.createElement(DataList.Label, null, "Email"),
                React.createElement(DataList.Value, null,
                    React.createElement(Link, { href: "mailto:artur@whop.com" }, "artur@whop.com"))),
            React.createElement(DataList.Item, null,
                React.createElement(DataList.Label, null, "Company"),
                React.createElement(DataList.Value, null,
                    React.createElement(Link, { target: "_blank", href: "https://whop.com" }, "Whop")))),
        React.createElement(DataList.Root, { ...args, size: "2" },
            React.createElement(DataList.Item, null,
                React.createElement(DataList.Label, null, "Name"),
                React.createElement(DataList.Value, null, "Artur Bie\u0144")),
            React.createElement(DataList.Item, null,
                React.createElement(DataList.Label, null, "Email"),
                React.createElement(DataList.Value, null,
                    React.createElement(Link, { href: "mailto:artur@whop.com" }, "artur@whop.com"))),
            React.createElement(DataList.Item, null,
                React.createElement(DataList.Label, null, "Company"),
                React.createElement(DataList.Value, null,
                    React.createElement(Link, { target: "_blank", href: "https://whop.com" }, "Whop")))),
        React.createElement(DataList.Root, { ...args, size: "3" },
            React.createElement(DataList.Item, null,
                React.createElement(DataList.Label, null, "Name"),
                React.createElement(DataList.Value, null, "Artur Bie\u0144")),
            React.createElement(DataList.Item, null,
                React.createElement(DataList.Label, null, "Email"),
                React.createElement(DataList.Value, null,
                    React.createElement(Link, { href: "mailto:artur@whop.com" }, "artur@whop.com"))),
            React.createElement(DataList.Item, null,
                React.createElement(DataList.Label, null, "Company"),
                React.createElement(DataList.Value, null,
                    React.createElement(Link, { target: "_blank", href: "https://whop.com" }, "Whop")))))),
};
export const Orientation = {
    render: (args) => (React.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-8)' } },
        React.createElement("div", null,
            React.createElement(Heading, { size: "3" }, "Horizontal"),
            React.createElement(Separator, { orientation: "horizontal", size: "4", style: { marginBottom: 16, marginTop: 16 } }),
            React.createElement(DataList.Root, { ...args, orientation: 'horizontal' },
                React.createElement(DataList.Item, null,
                    React.createElement(DataList.Label, null, "Name"),
                    React.createElement(DataList.Value, null, "Artur Bie\u0144")),
                React.createElement(DataList.Item, null,
                    React.createElement(DataList.Label, null, "Email"),
                    React.createElement(DataList.Value, null,
                        React.createElement(Link, { href: "mailto:artur@whop.com" }, "artur@whop.com"))),
                React.createElement(DataList.Item, null,
                    React.createElement(DataList.Label, null, "Company"),
                    React.createElement(DataList.Value, null,
                        React.createElement(Link, { target: "_blank", href: "https://whop.com" }, "Whop"))))),
        React.createElement("div", null,
            React.createElement(Heading, { size: "3" }, "Vertical"),
            React.createElement(Separator, { orientation: "horizontal", size: "4", style: { marginBottom: 16, marginTop: 16 } }),
            React.createElement(DataList.Root, { ...args, orientation: 'vertical' },
                React.createElement(DataList.Item, null,
                    React.createElement(DataList.Label, null, "Name"),
                    React.createElement(DataList.Value, null, "Artur Bie\u0144")),
                React.createElement(DataList.Item, null,
                    React.createElement(DataList.Label, null, "Email"),
                    React.createElement(DataList.Value, null,
                        React.createElement(Link, { href: "mailto:artur@whop.com" }, "artur@whop.com"))),
                React.createElement(DataList.Item, null,
                    React.createElement(DataList.Label, null, "Company"),
                    React.createElement(DataList.Value, null,
                        React.createElement(Link, { target: "_blank", href: "https://whop.com" }, "Whop"))))))),
};
export const Color = {
    render: (args) => (React.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-4)' } },
        React.createElement(Text, null,
            "Use the ",
            React.createElement(Code, null, "color"),
            " prop on the ",
            React.createElement(Code, null, '<DataList.Label />'),
            " component to assign a specific color."),
        React.createElement(DataList.Root, { orientation: "vertical", ...args },
            React.createElement(DataList.Item, null,
                React.createElement(DataList.Label, { color: "iris", style: { minWidth: 40 } }, "Color:"),
                React.createElement(DataList.Value, null, "Iris")),
            React.createElement(DataList.Item, null,
                React.createElement(DataList.Label, { color: "cyan", style: { minWidth: 40 } }, "Color:"),
                React.createElement(DataList.Value, null, "Cyan")),
            React.createElement(DataList.Item, null,
                React.createElement(DataList.Label, { color: "lime", style: { minWidth: 40 } }, "Color:"),
                React.createElement(DataList.Value, null, "Lime")),
            React.createElement(DataList.Item, null,
                React.createElement(DataList.Label, { color: "crimson", style: { minWidth: 40 } }, "Color:"),
                React.createElement(DataList.Value, null, "Crimson"))))),
};
export const HighContrast = {
    name: 'High Contrast',
    render: (args) => (React.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-4)' } },
        React.createElement(Text, null,
            "Use the ",
            React.createElement(Code, null, "highContrast"),
            " prop on the ",
            React.createElement(Code, null, '<DataList.Label />'),
            " component ",
            React.createElement("br", null),
            " to increase color contrast with the background."),
        React.createElement("div", { style: { display: 'flex', gap: 'var(--space-9)' } },
            React.createElement(DataList.Root, { orientation: "vertical", ...args },
                React.createElement(DataList.Item, null,
                    React.createElement(DataList.Label, { color: "iris" }, "Name"),
                    React.createElement(DataList.Value, null, "Iris")),
                React.createElement(DataList.Item, null,
                    React.createElement(DataList.Label, { color: "cyan" }, "Name"),
                    React.createElement(DataList.Value, null, "Cyan")),
                React.createElement(DataList.Item, null,
                    React.createElement(DataList.Label, { color: "lime" }, "Name"),
                    React.createElement(DataList.Value, null, "Lime")),
                React.createElement(DataList.Item, null,
                    React.createElement(DataList.Label, { color: "crimson" }, "Name"),
                    React.createElement(DataList.Value, null, "Crimson"))),
            React.createElement(DataList.Root, { orientation: "vertical", ...args },
                React.createElement(DataList.Item, null,
                    React.createElement(DataList.Label, { color: "iris", highContrast: true }, "Name"),
                    React.createElement(DataList.Value, null, "Iris")),
                React.createElement(DataList.Item, null,
                    React.createElement(DataList.Label, { color: "cyan", highContrast: true }, "Name"),
                    React.createElement(DataList.Value, null, "Cyan")),
                React.createElement(DataList.Item, null,
                    React.createElement(DataList.Label, { color: "lime", highContrast: true }, "Name"),
                    React.createElement(DataList.Value, null, "Lime")),
                React.createElement(DataList.Item, null,
                    React.createElement(DataList.Label, { color: "crimson", highContrast: true }, "Name"),
                    React.createElement(DataList.Value, null, "Crimson")))))),
};
