// frosted-ui/dist/esm/components/avatar/avatar.stories.js
import { fn } from '@storybook/test';
import React from 'react';
import { Avatar } from '..';
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Components/Avatar',
    component: Avatar,
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
    args: {
        onLoadingStatusChange: fn(),
    },
};
export default meta;
// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
export const Default = {
    args: {
        fallback: 'Cameron Zoub',
        color: 'blue',
    },
    render: (args) => (React.createElement("div", { style: { display: 'flex', gap: 'var(--space-2)' } },
        React.createElement(Avatar, { ...args, src: "https://images.unsplash.com/photo-1502823403499-6ccfcf4fb453?&w=256&h=256&q=70&crop=focalpoint&fp-x=0.5&fp-y=0.3&fp-z=1&fit=crop" }))),
};
export const Variant = {
    args: {
        fallback: 'Cameron Zoub',
        color: 'blue',
    },
    render: (args) => (React.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-4)' } },
        React.createElement("div", { style: { display: 'flex', gap: 'var(--space-2)' } },
            React.createElement(Avatar, { ...args, src: "https://images.unsplash.com/photo-1502823403499-6ccfcf4fb453?&w=256&h=256&q=70&crop=focalpoint&fp-x=0.5&fp-y=0.3&fp-z=1&fit=crop", variant: "round" }),
            React.createElement(Avatar, { ...args, variant: "round" })),
        React.createElement("div", { style: { display: 'flex', gap: 'var(--space-2)' } },
            React.createElement(Avatar, { ...args, src: "https://images.unsplash.com/photo-1502823403499-6ccfcf4fb453?&w=256&h=256&q=70&crop=focalpoint&fp-x=0.5&fp-y=0.3&fp-z=1&fit=crop", variant: "square" }),
            React.createElement(Avatar, { ...args, variant: "square" })))),
};
export const Size = {
    args: {
        fallback: 'Cameron Zoub',
        src: 'https://images.unsplash.com/photo-1502823403499-6ccfcf4fb453?&w=256&h=256&q=70&crop=focalpoint&fp-x=0.5&fp-y=0.3&fp-z=1&fit=crop',
        color: 'blue',
    },
    render: (args) => (React.createElement("div", { style: { display: 'flex', alignItems: 'center', gap: 'var(--space-4)' } },
        React.createElement(Avatar, { ...args, size: "1" }),
        React.createElement(Avatar, { ...args, size: "2" }),
        React.createElement(Avatar, { ...args, size: "3" }),
        React.createElement(Avatar, { ...args, size: "4" }),
        React.createElement(Avatar, { ...args, size: "5" }),
        React.createElement(Avatar, { ...args, size: "6" }),
        React.createElement(Avatar, { ...args, size: "7" }),
        React.createElement(Avatar, { ...args, size: "8" }))),
};
export const Color = {
    args: {
        fallback: 'Cameron Zoub',
    },
    render: (args) => (React.createElement("div", { style: { display: 'flex', gap: 'var(--space-2)' } },
        React.createElement(Avatar, { ...args, color: "indigo" }),
        React.createElement(Avatar, { ...args, color: "cyan" }),
        React.createElement(Avatar, { ...args, color: "orange" }),
        React.createElement(Avatar, { ...args, color: "crimson" }))),
};
export const HighContrast = {
    args: {
        fallback: 'Cameron Zoub',
    },
    render: (args) => (React.createElement("div", { style: { display: 'inline-grid', gridTemplateRows: 'repeat(2, 1fr)', gap: '8px', gridAutoFlow: 'column' } },
        React.createElement(Avatar, { ...args, color: "indigo" }),
        React.createElement(Avatar, { ...args, color: "indigo", highContrast: true }),
        React.createElement(Avatar, { ...args, color: "cyan" }),
        React.createElement(Avatar, { ...args, color: "cyan", highContrast: true }),
        React.createElement(Avatar, { ...args, color: "orange" }),
        React.createElement(Avatar, { ...args, color: "orange", highContrast: true }),
        React.createElement(Avatar, { ...args, color: "crimson" }),
        React.createElement(Avatar, { ...args, color: "crimson", highContrast: true }))),
};
export const Fallback = {
    args: {
        fallback: 'Cameron Zoub',
    },
    render: (args) => (React.createElement("div", { style: { display: 'flex', gap: 'var(--space-2)' } },
        React.createElement(Avatar, { ...args, fallback: React.createElement(React.Fragment, null, "AB") }),
        React.createElement(Avatar, { ...args, fallback: "Ilya Miskov" }),
        React.createElement(Avatar, { ...args, fallback: "John von Neumann" }),
        React.createElement(Avatar, { ...args, fallback: "Grace Hopper 123" }),
        React.createElement(Avatar, { ...args, fallback: "artur_bien" }),
        React.createElement(Avatar, { ...args, fallback: React.createElement("div", { style: { width: 20, height: 20 } },
                React.createElement("svg", { viewBox: "0 0 64 64", fill: "currentColor" },
                    React.createElement("path", { d: "M41.5 14c4.687 0 8.5 4.038 8.5 9s-3.813 9-8.5 9S33 27.962 33 23 36.813 14 41.5 14zM56.289 43.609C57.254 46.21 55.3 49 52.506 49c-2.759 0-11.035 0-11.035 0 .689-5.371-4.525-10.747-8.541-13.03 2.388-1.171 5.149-1.834 8.07-1.834C48.044 34.136 54.187 37.944 56.289 43.609zM37.289 46.609C38.254 49.21 36.3 52 33.506 52c-5.753 0-17.259 0-23.012 0-2.782 0-4.753-2.779-3.783-5.392 2.102-5.665 8.245-9.472 15.289-9.472S35.187 40.944 37.289 46.609zM21.5 17c4.687 0 8.5 4.038 8.5 9s-3.813 9-8.5 9S13 30.962 13 26 16.813 17 21.5 17z" }))) }))),
};
