// frosted-ui/dist/cjs/components/context-menu/context-menu.stories.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Default = void 0;
const tslib_1 = require("tslib");
const react_1 = tslib_1.__importDefault(require("react"));
const __1 = require("..");
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Controls/ContextMenu',
    component: __1.ContextMenu.Content,
    args: {
        variant: __1.contextMenuContentPropDefs.variant.default,
    },
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'centered',
    },
    render: (args) => (react_1.default.createElement(__1.ContextMenu.Root, null,
        react_1.default.createElement(__1.ContextMenu.Trigger, null,
            react_1.default.createElement("div", { style: {
                    background: 'var(--gray-a2)',
                    borderRadius: 'var(--radius-3)',
                    width: 200,
                    height: 200,
                    display: 'flex',
                    alignItems: 'center',
                    justifyContent: 'center',
                } }, "Right-click here")),
        react_1.default.createElement(__1.ContextMenu.Content, { ...args },
            react_1.default.createElement(__1.ContextMenu.Item, { shortcut: "\u2318 E" }, "Edit"),
            react_1.default.createElement(__1.ContextMenu.Item, { shortcut: "\u2318 D" }, "Duplicate"),
            react_1.default.createElement(__1.ContextMenu.Separator, null),
            react_1.default.createElement(__1.ContextMenu.Item, { shortcut: "\u2318 N" }, "Archive"),
            react_1.default.createElement(__1.ContextMenu.Sub, null,
                react_1.default.createElement(__1.ContextMenu.SubTrigger, null, "More"),
                react_1.default.createElement(__1.ContextMenu.SubContent, null,
                    react_1.default.createElement(__1.ContextMenu.Item, null, "Move to project\u2026"),
                    react_1.default.createElement(__1.ContextMenu.Item, null, "Move to folder\u2026"),
                    react_1.default.createElement(__1.ContextMenu.Separator, null),
                    react_1.default.createElement(__1.ContextMenu.Item, null, "Advanced options\u2026"))),
            react_1.default.createElement(__1.ContextMenu.Separator, null),
            react_1.default.createElement(__1.ContextMenu.Item, null, "Share"),
            react_1.default.createElement(__1.ContextMenu.Item, null, "Add to favorites"),
            react_1.default.createElement(__1.ContextMenu.Separator, null),
            react_1.default.createElement(__1.ContextMenu.Item, { shortcut: "\u2318 \u232B", color: "danger" }, "Delete")))),
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
};
exports.default = meta;
// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
exports.Default = {};
