// frosted-ui/dist/cjs/components/sheet/sheet.js
"use strict";
'use client';
Object.defineProperty(exports, "__esModule", { value: true });
exports.Trigger = exports.Title = exports.Root = exports.NestedRoot = exports.Header = exports.Description = exports.Content = exports.Close = exports.Body = void 0;
const tslib_1 = require("tslib");
const React = tslib_1.__importStar(require("react"));
const vaul_1 = require("vaul");
const classnames_1 = tslib_1.__importDefault(require("classnames"));
const theme_1 = require("../../theme");
const heading_1 = require("../heading");
const text_1 = require("../text");
const SheetRoot = ({ ...props }) => React.createElement(vaul_1.Drawer.Root, { ...props });
exports.Root = SheetRoot;
SheetRoot.displayName = 'SheetRoot';
const SheetNestedRoot = ({ ...props }) => React.createElement(vaul_1.Drawer.NestedRoot, { ...props });
exports.NestedRoot = SheetNestedRoot;
SheetNestedRoot.displayName = 'SheetNestedRoot';
const SheetTrigger = (props) => React.createElement(vaul_1.Drawer.Trigger, { ...props, asChild: true });
exports.Trigger = SheetTrigger;
SheetTrigger.displayName = 'SheetTrigger';
const SheetClose = (props) => React.createElement(vaul_1.Drawer.Close, { ...props, asChild: true });
exports.Close = SheetClose;
SheetClose.displayName = 'SheetClose';
const SheetPortal = vaul_1.Drawer.Portal;
const SheetOverlay = ({ className, ...props }) => (React.createElement(vaul_1.Drawer.Overlay, { className: (0, classnames_1.default)('fui-SheetOverlay', className), ...props }));
SheetOverlay.displayName = 'SheetOverlay';
const SheetContent = ({ className, children, ...props }) => (React.createElement(SheetPortal, null,
    React.createElement(React.Fragment, null,
        React.createElement(theme_1.Theme, { asChild: true },
            React.createElement(SheetOverlay, null)),
        React.createElement(theme_1.Theme, { asChild: true },
            React.createElement(vaul_1.Drawer.Content, { className: (0, classnames_1.default)('fui-SheetContent', className), ...props },
                React.createElement("div", { className: "fui-SheetContentHandle" }),
                children)))));
exports.Content = SheetContent;
SheetContent.displayName = 'SheetContent';
const SheetHeader = ({ children, className, ...props }) => (React.createElement("div", { className: (0, classnames_1.default)('fui-SheetHeader', className), ...props }, children));
exports.Header = SheetHeader;
SheetHeader.displayName = 'SheetHeader';
const SheetBody = ({ children, className, ...props }) => (React.createElement("div", { className: (0, classnames_1.default)('fui-SheetBody', className), ...props }, children));
exports.Body = SheetBody;
SheetBody.displayName = 'SheetBody';
const SheetFooter = ({ className, ...props }) => (React.createElement("div", { className: (0, classnames_1.default)('fui-SheetFooter', className), ...props }));
SheetFooter.displayName = 'SheetFooter';
const SheetTitle = ({ size = '5', weight = 'bold', ...props }) => {
    return (React.createElement(vaul_1.Drawer.Title, { asChild: true },
        React.createElement(heading_1.Heading, { weight: weight, size: size, ...props })));
};
exports.Title = SheetTitle;
SheetTitle.displayName = 'SheetTitle';
const SheetDescription = ({ size = '3', weight = 'regular', ...props }) => {
    return (React.createElement(vaul_1.Drawer.Description, { asChild: true },
        React.createElement(text_1.Text, { as: "p", size: size, weight: weight, ...props })));
};
exports.Description = SheetDescription;
SheetDescription.displayName = 'SheetDescription';
