// frosted-ui/dist/esm/components/portal/portal.stories.js
import React, { useRef, useState } from 'react';
import { Button, Code, Portal, Text } from '..';
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Utilities/Portal',
    component: Portal.Root,
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
};
export default meta;
// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
export const Default = {
    args: {},
    render: (args) => {
        const [open, setOpen] = useState(false);
        const portalContainerRef = useRef(null);
        const portalContainerElement = portalContainerRef.current;
        return (React.createElement(React.Fragment, null,
            React.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-4)' } },
                React.createElement(Text, null,
                    "Anything you put inside this component will be rendered in a separate ",
                    React.createElement(Code, null, `<div>`),
                    " element.",
                    React.createElement("br", null),
                    "By default, this element will be appended to ",
                    React.createElement(Code, null, "document.body"),
                    " but you can choose a different container by using the container prop."),
                React.createElement("div", null,
                    React.createElement(Button, { onClick: () => setOpen(true) }, "Open Portal"))),
            React.createElement("div", { id: "portal-container", ref: portalContainerRef }),
            portalContainerElement && open ? (React.createElement(Portal.Root, { ...args, container: portalContainerElement },
                React.createElement("div", { style: {
                        position: 'fixed',
                        inset: 0,
                        display: 'flex',
                        alignItems: 'center',
                        justifyContent: 'center',
                        background: 'rgba(0, 0, 0, 0.65)',
                    } },
                    React.createElement(Button, { onClick: () => setOpen(false) }, "Close Portal")))) : null));
    },
};
