// frosted-ui/dist/cjs/helpers/radix-colors.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.radixGrayScalesDesaturated = exports.radixGrayScales = exports.radixGrayScalePure = exports.radixColorScalesRegular = exports.radixColorScalesMetal = exports.radixColorScalesBright = exports.radixColorScales = void 0;
exports.radixGetMatchingGrayScale = radixGetMatchingGrayScale;
// prettier-ignore
const radixColorScalesRegular = [
    'tomato',
    'red',
    'ruby',
    'crimson',
    'pink',
    'plum',
    'purple',
    'violet',
    'iris',
    'cyan',
    'teal',
    'jade',
    'green',
    'grass',
    'brown',
    // Whop Brand Colors
    'blue',
    'orange',
    'indigo',
];
exports.radixColorScalesRegular = radixColorScalesRegular;
const radixColorScalesBright = [
    'sky',
    'mint',
    'yellow',
    'amber',
    // Whop Brand Colors
    'lime',
    'lemon',
    'magenta',
];
exports.radixColorScalesBright = radixColorScalesBright;
const radixColorScalesMetal = ['gold', 'bronze'];
exports.radixColorScalesMetal = radixColorScalesMetal;
// prettier-ignore
const radixColorScales = [...radixColorScalesRegular, ...radixColorScalesBright, ...radixColorScalesMetal];
exports.radixColorScales = radixColorScales;
const radixGrayScalePure = 'gray';
exports.radixGrayScalePure = radixGrayScalePure;
const radixGrayScalesDesaturated = ['mauve', 'slate', 'sage', 'olive', 'sand'];
exports.radixGrayScalesDesaturated = radixGrayScalesDesaturated;
const radixGrayScales = [radixGrayScalePure, ...radixGrayScalesDesaturated];
exports.radixGrayScales = radixGrayScales;
function radixGetMatchingGrayScale(colorScale) {
    switch (colorScale) {
        case 'tomato':
        case 'red':
        case 'ruby':
        case 'crimson':
        case 'pink':
        case 'plum':
        case 'purple':
        case 'violet':
            return 'mauve';
        case 'sky':
        case 'cyan':
            return 'slate';
        case 'teal':
        case 'jade':
        case 'mint':
        case 'green':
            return 'sage';
        case 'grass':
            return 'olive';
        case 'yellow':
        case 'amber':
        case 'brown':
        case 'gold':
        case 'bronze':
            return 'sand';
        case 'iris':
        case 'blue':
        case 'orange':
        case 'indigo':
        case 'magenta':
        case 'lime':
        case 'lemon':
            return 'gray';
    }
}
