// frosted-ui/dist/cjs/components/blockquote/blockquote.stories.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Default = void 0;
const blockquote_1 = require("../../../src/components/blockquote");
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Typography/Blockquote',
    component: blockquote_1.Blockquote,
    args: {
        children: 'I love how we have the freedom to explore skeuomorphism',
        size: blockquote_1.blockquotePropDefs.size.default,
        color: blockquote_1.blockquotePropDefs.color.default,
    },
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
};
exports.default = meta;
// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
exports.Default = {};
