// frosted-ui/dist/cjs/components/calendar/calendar.js
"use strict";
'use client';
Object.defineProperty(exports, "__esModule", { value: true });
exports.Calendar = Calendar;
exports.RangeCalendar = RangeCalendar;
const tslib_1 = require("tslib");
const date_1 = require("@internationalized/date");
const calendar_1 = require("@react-aria/calendar");
const focus_1 = require("@react-aria/focus");
const i18n_1 = require("@react-aria/i18n");
const utils_1 = require("@react-aria/utils");
const calendar_2 = require("@react-stately/calendar");
const classnames_1 = tslib_1.__importDefault(require("classnames"));
const react_1 = tslib_1.__importStar(require("react"));
const __1 = require("../");
const icon_button_1 = require("../icon-button");
function Calendar({ className, ...props }) {
    const { locale } = (0, i18n_1.useLocale)();
    const state = (0, calendar_2.useCalendarState)({
        ...props,
        locale,
        createCalendar: date_1.createCalendar,
    });
    // Destructuring props to remap/omit React Aria button props to avoid warnings
    const { calendarProps, prevButtonProps: { onPress: prevOnPress, isDisabled: prevIsDisabled, 
    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    onFocusChange: prevOnFocusChange, ...otherPrevButtonProps }, nextButtonProps: { onPress: nextOnPress, isDisabled: nextIsDisabled, 
    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    onFocusChange: nextOnFocusChange, ...otherNextVuttonProps }, } = (0, calendar_1.useCalendar)(props, state);
    return (react_1.default.createElement("div", { ...calendarProps, className: (0, classnames_1.default)('fui-CalendarRoot', className) },
        react_1.default.createElement("div", { className: 'fui-CalendarHeader' },
            react_1.default.createElement(icon_button_1.IconButton, { ...otherPrevButtonProps, disabled: prevIsDisabled, 
                // @ts-expect-error fix mismatching onPress and onClick types
                onClick: prevOnPress, size: "1", 
                // Default button type is "submit" so we need to explicitly set it to "button"
                // to prevent form submission
                type: "button" },
                react_1.default.createElement(CalendaChevronLeftIcon, null)),
            react_1.default.createElement("div", { className: 'fui-CalendarDropdowns' },
                react_1.default.createElement(MonthDropdown, { state: state }),
                react_1.default.createElement(YearDropdown, { state: state })),
            react_1.default.createElement(icon_button_1.IconButton, { ...otherNextVuttonProps, disabled: nextIsDisabled, 
                // @ts-expect-error fix mismatching onPress and onClick types
                onClick: nextOnPress, size: "1", 
                // Default button type is "submit" so we need to explicitly set it to "button"
                // to prevent form submission
                type: "button" },
                react_1.default.createElement(CalendaChevronLeftIcon, { style: { transform: 'rotate(180deg)' } }))),
        react_1.default.createElement(CalendarGrid, { state: state })));
}
function RangeCalendar({ className, ...props }) {
    const { locale } = (0, i18n_1.useLocale)();
    const state = (0, calendar_2.useRangeCalendarState)({
        ...props,
        locale,
        createCalendar: date_1.createCalendar,
    });
    const ref = (0, react_1.useRef)(null);
    // Destructuring props to remap/omit React Aria button props to avoid warnings
    const { calendarProps, prevButtonProps: { onPress: prevOnPress, isDisabled: prevIsDisabled, 
    // Omit "onFocusChange" because it's not recognized by FUI button (plus we don't really need it)
    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    onFocusChange: prevOnFocusChange, ...otherPrevButtonProps }, nextButtonProps: { onPress: nextOnPress, isDisabled: nextIsDisabled, 
    // Omit "onFocusChange" because it's not recognized by FUI button (plus we don't really need it)
    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    onFocusChange: nextOnFocusChange, ...otherNextVuttonProps }, } = (0, calendar_1.useRangeCalendar)(props, state, ref);
    return (react_1.default.createElement("div", { ...calendarProps, 
        // https://github.com/adobe/react-spectrum/blob/2137076772a8c7b59bc5ff6705355c21a1be750c/packages/%40react-aria/calendar/src/useRangeCalendar.ts#L67
        onBlur: undefined, className: (0, classnames_1.default)('fui-CalendarRoot', className), ref: ref },
        react_1.default.createElement("div", { className: 'fui-CalendarHeader' },
            react_1.default.createElement(icon_button_1.IconButton, { ...otherPrevButtonProps, disabled: prevIsDisabled, 
                // @ts-expect-error fix mismatching onPress and onClick types
                onClick: prevOnPress, size: "1", type: "button" },
                react_1.default.createElement(CalendaChevronLeftIcon, null)),
            react_1.default.createElement("div", { className: 'fui-CalendarDropdowns' },
                react_1.default.createElement(MonthDropdown, { state: state }),
                react_1.default.createElement(YearDropdown, { state: state })),
            react_1.default.createElement(icon_button_1.IconButton, { ...otherNextVuttonProps, disabled: nextIsDisabled, 
                // @ts-expect-error fix mismatching onPress and onClick types
                onClick: nextOnPress, size: "1", type: "button" },
                react_1.default.createElement(CalendaChevronLeftIcon, { style: { transform: 'rotate(180deg)' } }))),
        react_1.default.createElement(CalendarGrid, { state: state })));
}
function MonthDropdown({ state }) {
    const months = [];
    const formatter = (0, i18n_1.useDateFormatter)({
        month: 'long',
        timeZone: state.timeZone,
    });
    // Format the name of each month in the year according to the
    // current locale and calendar system. Note that in some calendar
    // systems, such as the Hebrew, the number of months may differ
    // between years.
    const numMonths = state.focusedDate.calendar.getMonthsInYear(state.focusedDate);
    for (let i = 1; i <= numMonths; i++) {
        const date = state.focusedDate.set({ month: i });
        months.push(formatter.format(date.toDate(state.timeZone)));
    }
    const onChange = (newValue) => {
        const value = Number(newValue);
        const date = state.focusedDate.set({ month: value });
        state.setFocusedDate(date);
        state.setFocused(false);
    };
    return (react_1.default.createElement(__1.Select.Root, { "aria-label": "Month", onValueChange: onChange, value: state.focusedDate.month.toString(), disabled: state.isDisabled, size: "1" },
        react_1.default.createElement(__1.Select.Trigger, { variant: "surface" }),
        react_1.default.createElement(__1.Select.Content, null, months.map((month, i) => (react_1.default.createElement(__1.Select.Item, { key: i, value: (i + 1).toString() }, month))))));
}
function YearDropdown({ state }) {
    const years = [];
    const formatter = (0, i18n_1.useDateFormatter)({
        year: 'numeric',
        timeZone: state.timeZone,
    });
    // Format 20 years on each side of the current year according
    // to the current locale and calendar system.
    for (let i = -20; i <= 20; i++) {
        const date = state.focusedDate.add({ years: i });
        years.push({
            value: date,
            formatted: formatter.format(date.toDate(state.timeZone)),
        });
    }
    const onChange = (newValue) => {
        const index = Number(newValue);
        const date = years[index].value;
        state.setFocusedDate(date);
        state.setFocused(false);
    };
    return (react_1.default.createElement(__1.Select.Root, { "aria-label": "Year", value: (20).toString(), onValueChange: onChange, disabled: state.isDisabled, size: "1" },
        react_1.default.createElement(__1.Select.Trigger, { variant: "surface", onClick: (e) => e.stopPropagation() }),
        react_1.default.createElement(__1.Select.Content, null, years.map((year, i) => (react_1.default.createElement(__1.Select.Item, { key: i, value: i.toString() }, year.formatted))))));
}
function CalendarGrid({ state, ...props }) {
    const { locale } = (0, i18n_1.useLocale)();
    const { gridProps, headerProps, weekDays } = (0, calendar_1.useCalendarGrid)({ ...props, weekdayStyle: 'narrow' }, state);
    // Get the number of weeks in the month so we can render the proper number of rows.
    const weeksInMonth = (0, date_1.getWeeksInMonth)(state.visibleRange.start, locale);
    const placeholderWeeks = Math.max(0, 6 - weeksInMonth);
    return (react_1.default.createElement("table", { ...gridProps, cellPadding: "0", className: "fui-CalendarGrid" },
        react_1.default.createElement("thead", { ...headerProps },
            react_1.default.createElement("tr", null, weekDays.map((day, index) => (react_1.default.createElement("th", { key: index }, day))))),
        react_1.default.createElement("tbody", null,
            [...new Array(weeksInMonth).keys()].map((weekIndex) => (react_1.default.createElement("tr", { key: weekIndex }, state
                .getDatesInWeek(weekIndex)
                .map((date, i) => (date ? react_1.default.createElement(CalendarCell, { key: i, state: state, date: date }) : react_1.default.createElement("td", { key: i })))))),
            [...new Array(placeholderWeeks).keys()].map((_, index) => (react_1.default.createElement("tr", { key: `filler-${index}` }, [...new Array(7).keys()].map((_, i) => (react_1.default.createElement("td", { key: i },
                react_1.default.createElement("div", { "aria-hidden": true, className: "fui-CalendarGridCellInner fui-CalendarGridCellInnerPlaceholder" }, "0"))))))))));
}
function CalendarCell({ state, ...props }) {
    const ref = (0, react_1.useRef)(null);
    const { cellProps, buttonProps, formattedDate, isSelected, isOutsideVisibleRange, isUnavailable } = (0, calendar_1.useCalendarCell)(props, state, ref);
    const { focusProps, isFocused, isFocusVisible } = (0, focus_1.useFocusRing)();
    // The start and end date of the selected range will have
    // an emphasized appearance.
    const isSelectionStart = 'highlightedRange' in state && state.highlightedRange
        ? (0, date_1.isSameDay)(props.date, state.highlightedRange.start)
        : isSelected;
    const isSelectionEnd = 'highlightedRange' in state && state.highlightedRange
        ? (0, date_1.isSameDay)(props.date, state.highlightedRange.end)
        : isSelected;
    let isHighlighted = false;
    if ('highlightedRange' in state && state.highlightedRange) {
        const { highlightedRange } = state;
        const { start, end } = highlightedRange;
        if (start && end) {
            isHighlighted = props.date.compare(start) >= 0 && props.date.compare(end) <= 0;
        }
    }
    let isSingleDateSelected = false;
    if (!('highlightedRange' in state)) {
        isSingleDateSelected = state.value !== null && props.date.compare(state.value) === 0;
    }
    const { locale } = (0, i18n_1.useLocale)();
    const dayOfWeek = (0, date_1.getDayOfWeek)(props.date, locale);
    const isRoundedLeft = isSelected && (isSelectionStart || dayOfWeek === 0 || props.date.day === 1);
    const isRoundedRight = isSelected &&
        (isSelectionEnd || dayOfWeek === 6 || props.date.day === props.date.calendar.getDaysInMonth(props.date));
    return (react_1.default.createElement("td", { ...cellProps, className: "fui-CalendarGridCell" },
        react_1.default.createElement("div", { ref: ref, hidden: isOutsideVisibleRange, ...(0, utils_1.mergeProps)(buttonProps, focusProps), "data-selection-state": (isSelectionStart && isSelectionEnd) || isSingleDateSelected
                ? 'selected'
                : isSelectionStart
                    ? 'start'
                    : isSelectionEnd
                        ? 'end'
                        : !(isSelectionStart || isSelectionEnd) && isSelected
                            ? 'range'
                            : isHighlighted
                                ? 'range'
                                : undefined, "data-rounded": isRoundedLeft ? 'left' : isRoundedRight ? 'right' : undefined, className: (0, classnames_1.default)('fui-CalendarGridCellInner', {
                ['fui-CalendarGridCell-selected']: isSelectionStart || isSelectionEnd || isSingleDateSelected,
                ['fui-CalendarGridCell-focusRing']: isFocused && isFocusVisible,
                ['fui-CalendarGridCell-unavailable']: isUnavailable,
            }) }, formattedDate)));
}
const CalendaChevronLeftIcon = (props) => (react_1.default.createElement("svg", { width: "16", height: "16", viewBox: "0 0 16 16", fill: "none", xmlns: "http://www.w3.org/2000/svg", ...props },
    react_1.default.createElement("path", { d: "M9.5 3.5L5.35355 7.64645C5.15829 7.84171 5.15829 8.15829 5.35355 8.35355L9.5 12.5", stroke: "currentColor", strokeWidth: "1.5", strokeLinecap: "round" })));
