// frosted-ui/dist/cjs/components/table/table.stories.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TanstackTable = exports.HorizontalScroll = exports.TableFooter = exports.Variant = exports.Default = void 0;
const tslib_1 = require("tslib");
const react_table_1 = require("@tanstack/react-table");
const react_1 = tslib_1.__importDefault(require("react"));
const __1 = require("..");
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Data presentation/Table',
    component: __1.Table.Root,
    args: {
        size: __1.tableRootPropDefs.size.default,
        variant: __1.tableRootPropDefs.variant.default,
    },
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
};
exports.default = meta;
// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
exports.Default = {
    render: (args) => (react_1.default.createElement(__1.Table.Root, { ...args },
        react_1.default.createElement(__1.Table.Table, null,
            react_1.default.createElement(__1.Table.Header, null,
                react_1.default.createElement(__1.Table.Row, null,
                    react_1.default.createElement(__1.Table.ColumnHeaderCell, null, "Full name"),
                    react_1.default.createElement(__1.Table.ColumnHeaderCell, null, "Email"),
                    react_1.default.createElement(__1.Table.ColumnHeaderCell, null, "Group"))),
            react_1.default.createElement(__1.Table.Body, null,
                react_1.default.createElement(__1.Table.Row, null,
                    react_1.default.createElement(__1.Table.RowHeaderCell, null, "Danilo Sousa"),
                    react_1.default.createElement(__1.Table.Cell, null, "danilo@example.com"),
                    react_1.default.createElement(__1.Table.Cell, null,
                        react_1.default.createElement(__1.Badge, { color: "green" }, "Developer"))),
                react_1.default.createElement(__1.Table.Row, null,
                    react_1.default.createElement(__1.Table.RowHeaderCell, null, "Zahra Ambessa"),
                    react_1.default.createElement(__1.Table.Cell, null, "zahra@example.com"),
                    react_1.default.createElement(__1.Table.Cell, null,
                        react_1.default.createElement(__1.Badge, { color: "amber" }, "Admin"))),
                react_1.default.createElement(__1.Table.Row, null,
                    react_1.default.createElement(__1.Table.RowHeaderCell, null, "Jasper Eriksson"),
                    react_1.default.createElement(__1.Table.Cell, null, "jasper@example.com"),
                    react_1.default.createElement(__1.Table.Cell, null,
                        react_1.default.createElement(__1.Badge, { color: "green" }, "Developer"))))),
        react_1.default.createElement(__1.Table.BottomBar, null, "Showing 1 to 3 of 3 entries"))),
};
exports.Variant = {
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'row', gap: 'var(--space-3)' } },
        react_1.default.createElement("div", null,
            react_1.default.createElement(__1.Table.Root, { ...args, variant: "ghost" },
                react_1.default.createElement(__1.Table.Table, null,
                    react_1.default.createElement(__1.Table.Header, null,
                        react_1.default.createElement(__1.Table.Row, null,
                            react_1.default.createElement(__1.Table.ColumnHeaderCell, null,
                                react_1.default.createElement("div", { style: { display: 'flex', alignItems: 'center', gap: 'var(--space-3)' } },
                                    react_1.default.createElement(__1.Checkbox, null),
                                    "Full name")),
                            react_1.default.createElement(__1.Table.ColumnHeaderCell, null, "Email"),
                            react_1.default.createElement(__1.Table.ColumnHeaderCell, null, "Group"))),
                    react_1.default.createElement(__1.Table.Body, null,
                        react_1.default.createElement(__1.Table.Row, null,
                            react_1.default.createElement(__1.Table.RowHeaderCell, null,
                                react_1.default.createElement("div", { style: { display: 'flex', alignItems: 'center', gap: 'var(--space-3)' } },
                                    react_1.default.createElement(__1.Checkbox, null),
                                    "Danilo Sousa")),
                            react_1.default.createElement(__1.Table.Cell, null, "danilo@example.com"),
                            react_1.default.createElement(__1.Table.Cell, null,
                                react_1.default.createElement(__1.Badge, { color: "green" }, "Developer"))),
                        react_1.default.createElement(__1.Table.Row, null,
                            react_1.default.createElement(__1.Table.RowHeaderCell, null,
                                react_1.default.createElement("div", { style: { display: 'flex', alignItems: 'center', gap: 'var(--space-3)' } },
                                    react_1.default.createElement(__1.Checkbox, null),
                                    "Zahra Ambessa")),
                            react_1.default.createElement(__1.Table.Cell, null, "zahra@example.com"),
                            react_1.default.createElement(__1.Table.Cell, null,
                                react_1.default.createElement(__1.Badge, { color: "amber" }, "Admin"))),
                        react_1.default.createElement(__1.Table.Row, null,
                            react_1.default.createElement(__1.Table.RowHeaderCell, null,
                                react_1.default.createElement("div", { style: { display: 'flex', alignItems: 'center', gap: 'var(--space-3)' } },
                                    react_1.default.createElement(__1.Checkbox, null),
                                    "Jasper Eriksson")),
                            react_1.default.createElement(__1.Table.Cell, null, "jasper@example.com"),
                            react_1.default.createElement(__1.Table.Cell, null,
                                react_1.default.createElement(__1.Badge, { color: "green" }, "Developer"))))))),
        react_1.default.createElement("div", null,
            react_1.default.createElement(__1.Table.Root, { ...args, variant: "surface" },
                react_1.default.createElement(__1.Table.Table, null,
                    react_1.default.createElement(__1.Table.Header, null,
                        react_1.default.createElement(__1.Table.Row, null,
                            react_1.default.createElement(__1.Table.ColumnHeaderCell, null,
                                react_1.default.createElement("div", { style: { display: 'flex', alignItems: 'center', gap: 'var(--space-3)' } },
                                    react_1.default.createElement(__1.Checkbox, null),
                                    "Full name")),
                            react_1.default.createElement(__1.Table.ColumnHeaderCell, null, "Email"),
                            react_1.default.createElement(__1.Table.ColumnHeaderCell, null, "Group"))),
                    react_1.default.createElement(__1.Table.Body, null,
                        react_1.default.createElement(__1.Table.Row, null,
                            react_1.default.createElement(__1.Table.RowHeaderCell, null,
                                react_1.default.createElement("div", { style: { display: 'flex', alignItems: 'center', gap: 'var(--space-3)' } },
                                    react_1.default.createElement(__1.Checkbox, null),
                                    "Danilo Sousa")),
                            react_1.default.createElement(__1.Table.Cell, null, "danilo@example.com"),
                            react_1.default.createElement(__1.Table.Cell, null,
                                react_1.default.createElement(__1.Badge, { color: "green" }, "Developer"))),
                        react_1.default.createElement(__1.Table.Row, null,
                            react_1.default.createElement(__1.Table.RowHeaderCell, null,
                                react_1.default.createElement("div", { style: { display: 'flex', alignItems: 'center', gap: 'var(--space-3)' } },
                                    react_1.default.createElement(__1.Checkbox, null),
                                    "Zahra Ambessa")),
                            react_1.default.createElement(__1.Table.Cell, null, "zahra@example.com"),
                            react_1.default.createElement(__1.Table.Cell, null,
                                react_1.default.createElement(__1.Badge, { color: "amber" }, "Admin"))),
                        react_1.default.createElement(__1.Table.Row, null,
                            react_1.default.createElement(__1.Table.RowHeaderCell, null,
                                react_1.default.createElement("div", { style: { display: 'flex', alignItems: 'center', gap: 'var(--space-3)' } },
                                    react_1.default.createElement(__1.Checkbox, null),
                                    "Jasper Eriksson")),
                            react_1.default.createElement(__1.Table.Cell, null, "jasper@example.com"),
                            react_1.default.createElement(__1.Table.Cell, null,
                                react_1.default.createElement(__1.Badge, { color: "green" }, "Developer"))))))))),
};
exports.TableFooter = {
    render: (args) => (react_1.default.createElement("div", { style: {
            display: 'flex',
            flexDirection: 'column',
            gap: 'var(--space-6)',
            justifyContent: 'center',
            alignItems: 'center',
        } },
        react_1.default.createElement(__1.Text, { style: { maxWidth: 700 } },
            "The ",
            react_1.default.createElement(__1.Code, null, '<TableFooter />'),
            " component is usually used as summary of the columns, e.g., a sum of the given numbers in a column."),
        react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'row', gap: 'var(--space-8)' } },
            react_1.default.createElement("div", null,
                react_1.default.createElement(__1.Table.Root, { ...args, variant: "ghost" },
                    react_1.default.createElement(__1.Table.Table, null,
                        react_1.default.createElement(__1.Table.Header, null,
                            react_1.default.createElement(__1.Table.Row, null,
                                react_1.default.createElement(__1.Table.ColumnHeaderCell, null, "Product"),
                                react_1.default.createElement(__1.Table.ColumnHeaderCell, null, "Price"))),
                        react_1.default.createElement(__1.Table.Body, null,
                            react_1.default.createElement(__1.Table.Row, null,
                                react_1.default.createElement(__1.Table.RowHeaderCell, null, "Pizza"),
                                react_1.default.createElement(__1.Table.Cell, null, "$20")),
                            react_1.default.createElement(__1.Table.Row, null,
                                react_1.default.createElement(__1.Table.RowHeaderCell, null, "Confetti"),
                                react_1.default.createElement(__1.Table.Cell, null, "$30")),
                            react_1.default.createElement(__1.Table.Row, null,
                                react_1.default.createElement(__1.Table.RowHeaderCell, null, "Beer"),
                                react_1.default.createElement(__1.Table.Cell, null, "$43"))),
                        react_1.default.createElement(__1.Table.Footer, null,
                            react_1.default.createElement(__1.Table.Row, null,
                                react_1.default.createElement(__1.Table.RowHeaderCell, null, "Total"),
                                react_1.default.createElement(__1.Table.Cell, null,
                                    react_1.default.createElement(__1.Text, { color: "green" }, "$93"))))))),
            react_1.default.createElement("div", null,
                react_1.default.createElement(__1.Table.Root, { ...args, variant: "surface" },
                    react_1.default.createElement(__1.Table.Table, null,
                        react_1.default.createElement(__1.Table.Header, null,
                            react_1.default.createElement(__1.Table.Row, null,
                                react_1.default.createElement(__1.Table.ColumnHeaderCell, null, "Product"),
                                react_1.default.createElement(__1.Table.ColumnHeaderCell, null, "Price"))),
                        react_1.default.createElement(__1.Table.Body, null,
                            react_1.default.createElement(__1.Table.Row, null,
                                react_1.default.createElement(__1.Table.RowHeaderCell, null, "Pizza"),
                                react_1.default.createElement(__1.Table.Cell, null, "$20")),
                            react_1.default.createElement(__1.Table.Row, null,
                                react_1.default.createElement(__1.Table.RowHeaderCell, null, "Confetti"),
                                react_1.default.createElement(__1.Table.Cell, null, "$30")),
                            react_1.default.createElement(__1.Table.Row, null,
                                react_1.default.createElement(__1.Table.RowHeaderCell, null, "Beer"),
                                react_1.default.createElement(__1.Table.Cell, null, "$43"))),
                        react_1.default.createElement(__1.Table.Footer, null,
                            react_1.default.createElement(__1.Table.Row, null,
                                react_1.default.createElement(__1.Table.RowHeaderCell, null, "Total"),
                                react_1.default.createElement(__1.Table.Cell, null,
                                    react_1.default.createElement(__1.Text, { color: "green" }, "$93")))))))))),
};
exports.HorizontalScroll = {
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'row', gap: 'var(--space-8)' } },
        react_1.default.createElement("div", null,
            react_1.default.createElement(__1.Table.Root, { ...args, variant: "ghost", style: { maxWidth: 400, whiteSpace: 'nowrap' } },
                react_1.default.createElement(__1.ScrollArea, { scrollbars: "horizontal" },
                    react_1.default.createElement(__1.Table.Table, null,
                        react_1.default.createElement(__1.Table.Header, null,
                            react_1.default.createElement(__1.Table.Row, null,
                                react_1.default.createElement(__1.Table.ColumnHeaderCell, null, "Product"),
                                react_1.default.createElement(__1.Table.ColumnHeaderCell, null, "Availability"),
                                react_1.default.createElement(__1.Table.ColumnHeaderCell, null, "Store"),
                                react_1.default.createElement(__1.Table.ColumnHeaderCell, null, "Price"),
                                react_1.default.createElement(__1.Table.ColumnHeaderCell, null, "Tax"))),
                        react_1.default.createElement(__1.Table.Body, null,
                            react_1.default.createElement(__1.Table.Row, null,
                                react_1.default.createElement(__1.Table.RowHeaderCell, null, "Pizza"),
                                react_1.default.createElement(__1.Table.Cell, null,
                                    react_1.default.createElement(__1.Badge, { color: "green" }, "In stock")),
                                react_1.default.createElement(__1.Table.Cell, null, "Tesco"),
                                react_1.default.createElement(__1.Table.Cell, null, "$20"),
                                react_1.default.createElement(__1.Table.Cell, null, "$2")),
                            react_1.default.createElement(__1.Table.Row, null,
                                react_1.default.createElement(__1.Table.RowHeaderCell, null, "Confetti"),
                                react_1.default.createElement(__1.Table.Cell, null,
                                    react_1.default.createElement(__1.Badge, { color: "green" }, "In stock")),
                                react_1.default.createElement(__1.Table.Cell, null, "Wallmart"),
                                react_1.default.createElement(__1.Table.Cell, null, "$30"),
                                react_1.default.createElement(__1.Table.Cell, null, "$3.50")),
                            react_1.default.createElement(__1.Table.Row, null,
                                react_1.default.createElement(__1.Table.RowHeaderCell, null, "Beer"),
                                react_1.default.createElement(__1.Table.Cell, null,
                                    react_1.default.createElement(__1.Badge, { color: "red" }, "Out of stock")),
                                react_1.default.createElement(__1.Table.Cell, null, "Beer Enthusiast"),
                                react_1.default.createElement(__1.Table.Cell, null, "$43"),
                                react_1.default.createElement(__1.Table.Cell, null, "$13.75"))),
                        react_1.default.createElement(__1.Table.Footer, null,
                            react_1.default.createElement(__1.Table.Row, null,
                                react_1.default.createElement(__1.Table.RowHeaderCell, { colSpan: 3 }, "Total"),
                                react_1.default.createElement(__1.Table.Cell, null,
                                    react_1.default.createElement(__1.Text, { color: "green" }, "$93")),
                                react_1.default.createElement(__1.Table.Cell, null,
                                    react_1.default.createElement(__1.Text, { color: "red" }, "$19.25")))))),
                react_1.default.createElement(__1.Table.BottomBar, null,
                    react_1.default.createElement(__1.Text, { size: "1", color: "gray", align: "center" }, "Birthday party shopping list.")))),
        react_1.default.createElement("div", null,
            react_1.default.createElement(__1.Table.Root, { ...args, variant: "surface", style: { maxWidth: 400, whiteSpace: 'nowrap' } },
                react_1.default.createElement(__1.ScrollArea, { scrollbars: "horizontal" },
                    react_1.default.createElement(__1.Table.Table, null,
                        react_1.default.createElement(__1.Table.Header, null,
                            react_1.default.createElement(__1.Table.Row, null,
                                react_1.default.createElement(__1.Table.ColumnHeaderCell, null, "Product"),
                                react_1.default.createElement(__1.Table.ColumnHeaderCell, null, "Availability"),
                                react_1.default.createElement(__1.Table.ColumnHeaderCell, null, "Store"),
                                react_1.default.createElement(__1.Table.ColumnHeaderCell, null, "Price"),
                                react_1.default.createElement(__1.Table.ColumnHeaderCell, null, "Tax"))),
                        react_1.default.createElement(__1.Table.Body, null,
                            react_1.default.createElement(__1.Table.Row, null,
                                react_1.default.createElement(__1.Table.RowHeaderCell, null, "Pizza"),
                                react_1.default.createElement(__1.Table.Cell, null,
                                    react_1.default.createElement(__1.Badge, { color: "green" }, "In stock")),
                                react_1.default.createElement(__1.Table.Cell, null, "Tesco"),
                                react_1.default.createElement(__1.Table.Cell, null, "$20"),
                                react_1.default.createElement(__1.Table.Cell, null, "$2")),
                            react_1.default.createElement(__1.Table.Row, null,
                                react_1.default.createElement(__1.Table.RowHeaderCell, null, "Confetti"),
                                react_1.default.createElement(__1.Table.Cell, null,
                                    react_1.default.createElement(__1.Badge, { color: "green" }, "In stock")),
                                react_1.default.createElement(__1.Table.Cell, null, "Wallmart"),
                                react_1.default.createElement(__1.Table.Cell, null, "$30"),
                                react_1.default.createElement(__1.Table.Cell, null, "$3.50")),
                            react_1.default.createElement(__1.Table.Row, null,
                                react_1.default.createElement(__1.Table.RowHeaderCell, null, "Beer"),
                                react_1.default.createElement(__1.Table.Cell, null,
                                    react_1.default.createElement(__1.Badge, { color: "red" }, "Out of stock")),
                                react_1.default.createElement(__1.Table.Cell, null, "Beer Enthusiast"),
                                react_1.default.createElement(__1.Table.Cell, null, "$43"),
                                react_1.default.createElement(__1.Table.Cell, null, "$13.75"))),
                        react_1.default.createElement(__1.Table.Footer, null,
                            react_1.default.createElement(__1.Table.Row, null,
                                react_1.default.createElement(__1.Table.RowHeaderCell, { colSpan: 3 }, "Total"),
                                react_1.default.createElement(__1.Table.Cell, null,
                                    react_1.default.createElement(__1.Text, { color: "green" }, "$93")),
                                react_1.default.createElement(__1.Table.Cell, null,
                                    react_1.default.createElement(__1.Text, { color: "red" }, "$19.25")))))),
                react_1.default.createElement(__1.Table.BottomBar, null,
                    react_1.default.createElement(__1.Text, { size: "1", color: "gray", align: "center" }, "Birthday party shopping list.")))))),
};
const data = [
    {
        id: '3u1reuv4',
        amount: 242,
        status: 'success',
        email: 'ilya@gmail.com',
    },
    {
        id: 'derv1ws0',
        amount: 837,
        status: 'processing',
        email: 'jamie@gmail.com',
    },
    {
        id: 'm5gr84i9',
        amount: 316,
        status: 'success',
        email: 'artur@yahoo.com',
    },
    {
        id: '5kma53ae',
        amount: 874,
        status: 'success',
        email: 'fj@gmail.com',
    },
    {
        id: 'bhqecj4p',
        amount: 721,
        status: 'failed',
        email: 'steven@whop.com',
    },
    {
        id: '5kma23ae',
        amount: 874,
        status: 'success',
        email: 'es@whop.com',
    },
    {
        id: '8jqecj4p',
        amount: 721,
        status: 'failed',
        email: 'dima@yahoo.com',
    },
    {
        id: 'asma53ae',
        amount: 174,
        status: 'success',
        email: 'tester@wp.pl',
    },
    {
        id: 'bhqokj4p',
        amount: 721,
        status: 'failed',
        email: 'someone@gmail.com',
    },
    {
        id: '5kma93ae',
        amount: 894,
        status: 'success',
        email: 'tristan@whop.com',
    },
];
const columns = [
    {
        id: 'select',
        header: ({ table }) => (react_1.default.createElement(__1.Checkbox, { checked: table.getIsAllRowsSelected() || (table.getIsSomeRowsSelected() && 'indeterminate'), onCheckedChange: (value) => table.toggleAllRowsSelected(!!value), "aria-label": "Select all" })),
        cell: ({ row }) => (react_1.default.createElement(__1.Checkbox, { checked: row.getIsSelected(), onCheckedChange: (value) => row.toggleSelected(!!value), "aria-label": "Select row" })),
        enableSorting: false,
        enableHiding: false,
    },
    {
        accessorKey: 'status',
        header: ({ column }) => (react_1.default.createElement(__1.Table.ColumnHeaderCellButton, { isSortable: column.getCanSort(), sortDirection: column.getIsSorted(), onClick: () => column.toggleSorting() }, "Status")),
        cell: ({ row }) => row.getValue('status'),
    },
    {
        accessorKey: 'email',
        header: ({ column }) => {
            return (react_1.default.createElement(__1.Table.ColumnHeaderCellButton, { isSortable: column.getCanSort(), sortDirection: column.getIsSorted(), onClick: () => column.toggleSorting() }, "Email"));
        },
        cell: ({ row }) => row.getValue('email'),
    },
    {
        enableSorting: true,
        accessorKey: 'amount',
        header: ({ column }) => {
            return (react_1.default.createElement(__1.Table.ColumnHeaderCellButton, { isSortable: column.getCanSort(), sortDirection: column.getIsSorted(), onClick: () => column.toggleSorting() }, "Amount"));
        },
        cell: ({ row }) => {
            const amount = parseFloat(row.getValue('amount'));
            // Format the amount as a dollar amount
            const formatted = new Intl.NumberFormat('en-US', {
                style: 'currency',
                currency: 'USD',
            }).format(amount);
            return react_1.default.createElement(__1.Text, { color: "green" }, formatted);
        },
    },
    {
        id: 'actions',
        enableHiding: false,
        cell: ({ row }) => {
            const payment = row.original;
            return (react_1.default.createElement("div", { style: { display: 'flex', justifyContent: 'end' } },
                react_1.default.createElement(__1.DropdownMenu.Root, null,
                    react_1.default.createElement(__1.DropdownMenu.Trigger, null,
                        react_1.default.createElement(__1.IconButton, { variant: "ghost", color: "gray", "aria-label": "Open menu", style: { margin: 0 } },
                            react_1.default.createElement("svg", { xmlns: "http://www.w3.org/2000/svg", width: "24", height: "24", viewBox: "0 0 24 24", fill: "none", stroke: "currentColor", strokeWidth: "2", strokeLinecap: "round", strokeLinejoin: "round", style: { width: 16, height: 16 } },
                                react_1.default.createElement("circle", { cx: "12", cy: "12", r: "1" }),
                                react_1.default.createElement("circle", { cx: "19", cy: "12", r: "1" }),
                                react_1.default.createElement("circle", { cx: "5", cy: "12", r: "1" })))),
                    react_1.default.createElement(__1.DropdownMenu.Content, { align: "end" },
                        react_1.default.createElement(__1.DropdownMenu.Item, null, "View customer"),
                        react_1.default.createElement(__1.DropdownMenu.Item, null, "View payment details"),
                        react_1.default.createElement(__1.DropdownMenu.Separator, null),
                        react_1.default.createElement(__1.DropdownMenu.Label, null, "Actions"),
                        react_1.default.createElement(__1.DropdownMenu.Item, { onClick: () => navigator.clipboard.writeText(payment.id) }, "Copy payment ID"),
                        react_1.default.createElement(__1.DropdownMenu.Item, { color: "danger" }, "Refund")))));
        },
    },
];
const TanstackTableExample = (props) => {
    var _a, _b, _c;
    const [sorting, setSorting] = react_1.default.useState([]);
    const [columnFilters, setColumnFilters] = react_1.default.useState([]);
    const [columnVisibility, setColumnVisibility] = react_1.default.useState({});
    const [rowSelection, setRowSelection] = react_1.default.useState({});
    const [pagination, setPagination] = react_1.default.useState({
        pageSize: 5,
        pageIndex: 0,
    });
    const table = (0, react_table_1.useReactTable)({
        data,
        columns,
        onSortingChange: setSorting,
        onColumnFiltersChange: setColumnFilters,
        getCoreRowModel: (0, react_table_1.getCoreRowModel)(),
        getPaginationRowModel: (0, react_table_1.getPaginationRowModel)(),
        getSortedRowModel: (0, react_table_1.getSortedRowModel)(),
        getFilteredRowModel: (0, react_table_1.getFilteredRowModel)(),
        onColumnVisibilityChange: setColumnVisibility,
        onRowSelectionChange: setRowSelection,
        onPaginationChange: setPagination,
        state: {
            sorting,
            columnFilters,
            columnVisibility,
            rowSelection,
            pagination,
        },
    });
    return (react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-4)', width: 580 } },
        react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-4)', width: '100%' } },
            react_1.default.createElement("div", { style: { width: '100%' } },
                react_1.default.createElement(__1.TextField.Input, { placeholder: "Filter emails...", value: (_b = (_a = table.getColumn('email')) === null || _a === void 0 ? void 0 : _a.getFilterValue()) !== null && _b !== void 0 ? _b : '', onChange: (event) => { var _a; return (_a = table.getColumn('email')) === null || _a === void 0 ? void 0 : _a.setFilterValue(event.target.value); }, style: { width: '100%' }, width: '100%' })),
            react_1.default.createElement(__1.DropdownMenu.Root, null,
                react_1.default.createElement(__1.DropdownMenu.Trigger, null,
                    react_1.default.createElement(__1.Button, { variant: "surface", className: "ml-auto" }, "Columns")),
                react_1.default.createElement(__1.DropdownMenu.Content, { align: "end" }, table
                    .getAllColumns()
                    .filter((column) => column.getCanHide())
                    .map((column) => {
                    return (react_1.default.createElement(__1.DropdownMenu.CheckboxItem, { key: column.id, className: "capitalize", checked: column.getIsVisible(), onCheckedChange: (value) => column.toggleVisibility(!!value), onSelect: (event) => event.preventDefault() }, column.id));
                })))),
        react_1.default.createElement(__1.Table.Root, { ...props },
            react_1.default.createElement(__1.Table.Table, null,
                react_1.default.createElement(__1.Table.Header, null, table.getHeaderGroups().map((headerGroup) => (react_1.default.createElement(__1.Table.Row, { key: headerGroup.id }, headerGroup.headers.map((header) => {
                    return (react_1.default.createElement(__1.Table.ColumnHeaderCell, { key: header.id }, header.isPlaceholder
                        ? null
                        : (0, react_table_1.flexRender)(header.column.columnDef.header, header.getContext())));
                }))))),
                react_1.default.createElement(__1.Table.Body, null, ((_c = table.getRowModel().rows) === null || _c === void 0 ? void 0 : _c.length) ? (table.getRowModel().rows.map((row) => (react_1.default.createElement(__1.Table.Row, { key: row.id, "data-state": row.getIsSelected() && 'selected' }, row.getVisibleCells().map((cell) => (react_1.default.createElement(__1.Table.Cell, { key: cell.id }, (0, react_table_1.flexRender)(cell.column.columnDef.cell, cell.getContext())))))))) : (react_1.default.createElement(__1.Table.Row, null,
                    react_1.default.createElement(__1.Table.Cell, { colSpan: columns.length }, "No results."))))),
            react_1.default.createElement(__1.Table.BottomBar, null,
                react_1.default.createElement("div", { style: { display: 'flex', alignItems: 'center', justifyContent: 'space-between' } },
                    react_1.default.createElement(__1.Text, { color: "gray", size: "2" },
                        table.getFilteredSelectedRowModel().rows.length,
                        " of ",
                        table.getFilteredRowModel().rows.length,
                        " row(s) selected."),
                    react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-2)' } },
                        react_1.default.createElement(__1.Button, { variant: "surface", onClick: () => table.previousPage(), disabled: !table.getCanPreviousPage() }, "Previous"),
                        react_1.default.createElement(__1.Button, { variant: "surface", onClick: () => table.nextPage(), disabled: !table.getCanNextPage() }, "Next"),
                        react_1.default.createElement(__1.Select.Root, { onValueChange: (newPerPage) => table.setPageSize(Number(newPerPage)), size: "2", value: table.getState().pagination.pageSize.toString(), "aria-label": "Rows per page" },
                            react_1.default.createElement(__1.Select.Trigger, { color: "gray", variant: "soft" }),
                            react_1.default.createElement(__1.Select.Content, null, [2, 5, 10].map((option) => (react_1.default.createElement(__1.Select.Item, { key: option, value: option.toString() }, option)))))))))));
};
exports.TanstackTable = {
    render: (args) => react_1.default.createElement(TanstackTableExample, { ...args }),
};
