// frosted-ui/dist/cjs/components/date-field/date-field.stories.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WithTime = exports.MinValue = exports.Size = exports.Default = void 0;
const tslib_1 = require("tslib");
const date_1 = require("@internationalized/date");
const react_1 = tslib_1.__importDefault(require("react"));
const __1 = require("..");
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Controls/Dates/DateField',
    component: __1.DateField,
    args: {
        size: __1.dateFieldPropDefs.size.default,
        color: __1.dateFieldPropDefs.color.default,
        'aria-label': 'Birth date',
    },
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
};
exports.default = meta;
// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
exports.Default = {
    args: {},
    render: (args) => (react_1.default.createElement("div", { style: { width: 300 } },
        react_1.default.createElement(__1.DateField, { ...args, defaultValue: (0, date_1.parseDate)('2020-02-03'), onChange: (date) => console.log(date === null || date === void 0 ? void 0 : date.toString()) }))),
};
exports.Size = {
    args: {},
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-4)', width: 300 } },
        react_1.default.createElement(__1.DateField, { ...args, defaultValue: (0, date_1.parseDate)('2020-02-03'), size: "1" }),
        react_1.default.createElement(__1.DateField, { ...args, defaultValue: (0, date_1.parseDate)('2020-02-03'), size: "2" }),
        react_1.default.createElement(__1.DateField, { ...args, defaultValue: (0, date_1.parseDate)('2020-02-03'), size: "3" }))),
};
exports.MinValue = {
    args: {},
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-4)', width: 300 } },
        react_1.default.createElement(__1.DateField, { ...args, minValue: (0, date_1.today)((0, date_1.getLocalTimeZone)()), defaultValue: (0, date_1.parseDate)('2020-02-03'), validationBehavior: "aria" }))),
};
exports.WithTime = {
    name: 'With time',
    args: {},
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-4)', width: 300 } },
        react_1.default.createElement(__1.DateField, { ...args, defaultValue: (0, date_1.parseZonedDateTime)('2022-11-07T00:45[America/Los_Angeles]') }),
        react_1.default.createElement(__1.DateField, { ...args, defaultValue: (0, date_1.parseZonedDateTime)('2022-11-07T00:45[America/Los_Angeles]'), hideTimeZone: true }),
        react_1.default.createElement(__1.DateField, { ...args, defaultValue: (0, date_1.parseZonedDateTime)('2022-11-07T00:45[America/Los_Angeles]'), hideTimeZone: true, granularity: "second" }))),
};
