// frosted-ui/dist/cjs/components/text-area/text-area.stories.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Color = exports.Variant = exports.Size = exports.Default = void 0;
const tslib_1 = require("tslib");
const react_1 = tslib_1.__importDefault(require("react"));
const __1 = require("..");
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Controls/TextArea',
    component: __1.TextArea,
    args: {
        size: __1.textAreaPropDefs.size.default,
        variant: __1.textAreaPropDefs.variant.default,
        color: __1.textAreaPropDefs.color.default,
        disabled: false,
    },
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
};
exports.default = meta;
// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
exports.Default = {
    render: (args) => (react_1.default.createElement("div", { style: { width: 300 } },
        react_1.default.createElement(__1.TextArea, { placeholder: "Reply to comment\u2026", ...args }))),
};
exports.Size = {
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-3)', maxWidth: 500 } },
        react_1.default.createElement(__1.TextArea, { placeholder: "Reply to comment\u2026", ...args, size: "1" }),
        react_1.default.createElement(__1.TextArea, { placeholder: "Reply to comment\u2026", ...args, size: "2" }),
        react_1.default.createElement(__1.TextArea, { placeholder: "Reply to comment\u2026", ...args, size: "3" }))),
};
exports.Variant = {
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-3)', maxWidth: 500 } },
        react_1.default.createElement(__1.TextArea, { placeholder: "Reply to comment\u2026", ...args, variant: "surface" }),
        react_1.default.createElement(__1.TextArea, { placeholder: "Reply to comment\u2026", ...args, variant: "soft" }))),
};
exports.Color = {
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-3)', maxWidth: 500 } },
        react_1.default.createElement(__1.TextArea, { variant: "soft", placeholder: "Reply to comment\u2026", ...args, color: "blue" }),
        react_1.default.createElement(__1.TextArea, { variant: "soft", placeholder: "Reply to comment\u2026", ...args, color: "green" }),
        react_1.default.createElement(__1.TextArea, { variant: "soft", placeholder: "Reply to comment\u2026", ...args, color: "red" }))),
};
