// frosted-ui/dist/cjs/components/code/code.stories.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.HighContrast = exports.Color = exports.Size = exports.Variant = exports.Default = void 0;
const tslib_1 = require("tslib");
const react_1 = tslib_1.__importDefault(require("react"));
const __1 = require("..");
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Typography/Code',
    component: __1.Code,
    args: {
        children: 'Code',
        size: __1.codePropDefs.size.default,
    },
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
};
exports.default = meta;
// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
exports.Default = {
    args: {
        children: 'Code',
        size: __1.codePropDefs.size.default,
    },
};
exports.Variant = {
    args: {
        size: __1.codePropDefs.size.default,
    },
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', alignItems: 'start', gap: 'var(--space-2)' } },
        react_1.default.createElement(__1.Code, { ...args, variant: "solid" }, "console.log()"),
        react_1.default.createElement(__1.Code, { ...args, variant: "soft" }, "console.log()"),
        react_1.default.createElement(__1.Code, { ...args, variant: "outline" }, "console.log()"),
        react_1.default.createElement(__1.Code, { ...args, variant: "ghost" }, "console.log()"))),
};
exports.Size = {
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-3)' } },
        react_1.default.createElement(__1.Code, { ...args, size: "1" }, "console.log()"),
        react_1.default.createElement(__1.Code, { ...args, size: "2" }, "console.log()"),
        react_1.default.createElement(__1.Code, { ...args, size: "3" }, "console.log()"),
        react_1.default.createElement(__1.Code, { ...args, size: "4" }, "console.log()"),
        react_1.default.createElement(__1.Code, { ...args, size: "5" }, "console.log()"),
        react_1.default.createElement(__1.Code, { ...args, size: "6" }, "console.log()"),
        react_1.default.createElement(__1.Code, { ...args, size: "7" }, "console.log()"),
        react_1.default.createElement(__1.Code, { ...args, size: "8" }, "console.log()"),
        react_1.default.createElement(__1.Code, { ...args, size: "9" }, "console.log()"))),
};
exports.Color = {
    args: {
        size: __1.codePropDefs.size.default,
    },
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', alignItems: 'start', gap: 'var(--space-2)' } },
        react_1.default.createElement(__1.Code, { ...args, color: "indigo" }, "console.log()"),
        react_1.default.createElement(__1.Code, { ...args, color: "crimson" }, "console.log()"),
        react_1.default.createElement(__1.Code, { ...args, color: "cyan" }, "console.log()"),
        react_1.default.createElement(__1.Code, { ...args, color: "orange" }, "console.log()"))),
};
exports.HighContrast = {
    name: 'High Contrast',
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-3)' } },
        react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', alignItems: 'start', gap: 'var(--space-2)' } },
            react_1.default.createElement(__1.Code, { ...args, variant: "solid" }, "console.log()"),
            react_1.default.createElement(__1.Code, { ...args, variant: "soft" }, "console.log()"),
            react_1.default.createElement(__1.Code, { ...args, variant: "outline" }, "console.log()"),
            react_1.default.createElement(__1.Code, { ...args, variant: "ghost" }, "console.log()")),
        react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', alignItems: 'start', gap: 'var(--space-2)' } },
            react_1.default.createElement(__1.Code, { ...args, variant: "solid", highContrast: true }, "console.log()"),
            react_1.default.createElement(__1.Code, { ...args, variant: "soft", highContrast: true }, "console.log()"),
            react_1.default.createElement(__1.Code, { ...args, variant: "outline", highContrast: true }, "console.log()"),
            react_1.default.createElement(__1.Code, { ...args, variant: "ghost", highContrast: true }, "console.log()")))),
};
