// frosted-ui/dist/esm/components/text-area/text-area.stories.js
import React from 'react';
import { TextArea, textAreaPropDefs } from '..';
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Controls/TextArea',
    component: TextArea,
    args: {
        size: textAreaPropDefs.size.default,
        variant: textAreaPropDefs.variant.default,
        color: textAreaPropDefs.color.default,
        disabled: false,
    },
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
};
export default meta;
// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
export const Default = {
    render: (args) => (React.createElement("div", { style: { width: 300 } },
        React.createElement(TextArea, { placeholder: "Reply to comment\u2026", ...args }))),
};
export const Size = {
    render: (args) => (React.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-3)', maxWidth: 500 } },
        React.createElement(TextArea, { placeholder: "Reply to comment\u2026", ...args, size: "1" }),
        React.createElement(TextArea, { placeholder: "Reply to comment\u2026", ...args, size: "2" }),
        React.createElement(TextArea, { placeholder: "Reply to comment\u2026", ...args, size: "3" }))),
};
export const Variant = {
    render: (args) => (React.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-3)', maxWidth: 500 } },
        React.createElement(TextArea, { placeholder: "Reply to comment\u2026", ...args, variant: "surface" }),
        React.createElement(TextArea, { placeholder: "Reply to comment\u2026", ...args, variant: "soft" }))),
};
export const Color = {
    render: (args) => (React.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-3)', maxWidth: 500 } },
        React.createElement(TextArea, { variant: "soft", placeholder: "Reply to comment\u2026", ...args, color: "blue" }),
        React.createElement(TextArea, { variant: "soft", placeholder: "Reply to comment\u2026", ...args, color: "green" }),
        React.createElement(TextArea, { variant: "soft", placeholder: "Reply to comment\u2026", ...args, color: "red" }))),
};
