// frosted-ui/dist/cjs/components/kbd/kbd.stories.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Size = exports.Default = void 0;
const tslib_1 = require("tslib");
const react_1 = tslib_1.__importDefault(require("react"));
const components_1 = require("../../../src/components/");
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Typography/Kbd',
    component: components_1.Kbd,
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
};
exports.default = meta;
// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
exports.Default = {
    args: {
        children: 'Shift + Tab',
        size: components_1.kbdPropDefs.size.default,
    },
};
exports.Size = {
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-3)' } },
        react_1.default.createElement("div", null,
            react_1.default.createElement(components_1.Kbd, { ...args, size: "1" }, "Shift + Tab")),
        react_1.default.createElement("div", null,
            react_1.default.createElement(components_1.Kbd, { ...args, size: "2" }, "Shift + Tab")),
        react_1.default.createElement("div", null,
            react_1.default.createElement(components_1.Kbd, { ...args, size: "3" }, "Shift + Tab")),
        react_1.default.createElement("div", null,
            react_1.default.createElement(components_1.Kbd, { ...args, size: "4" }, "Shift + Tab")),
        react_1.default.createElement("div", null,
            react_1.default.createElement(components_1.Kbd, { ...args, size: "5" }, "Shift + Tab")),
        react_1.default.createElement("div", null,
            react_1.default.createElement(components_1.Kbd, { ...args, size: "6" }, "Shift + Tab")),
        react_1.default.createElement("div", null,
            react_1.default.createElement(components_1.Kbd, { ...args, size: "7" }, "Shift + Tab")),
        react_1.default.createElement("div", null,
            react_1.default.createElement(components_1.Kbd, { ...args, size: "8" }, "Shift + Tab")),
        react_1.default.createElement("div", null,
            react_1.default.createElement(components_1.Kbd, { ...args, size: "9" }, "Shift + Tab")))),
};
