// frosted-ui/dist/cjs/components/callout/callout.stories.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AsAlert = exports.HighContrast = exports.SemanticColor = exports.Color = exports.Variant = exports.Size = exports.Default = void 0;
const tslib_1 = require("tslib");
const icons_1 = require("@frosted-ui/icons");
const react_1 = tslib_1.__importDefault(require("react"));
const __1 = require("..");
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Components/Callout',
    component: __1.Callout.Root,
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
};
exports.default = meta;
// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
exports.Default = {
    args: {
        color: __1.calloutRootPropDefs.color.default,
        variant: __1.calloutRootPropDefs.variant.default,
        children: (react_1.default.createElement(react_1.default.Fragment, null,
            "You will need to upgrade to the ",
            react_1.default.createElement(__1.Link, { href: "#" }, "newest Frosted-UI version"),
            " now.")),
    },
    render: ({ children, ...args }) => (react_1.default.createElement(__1.Callout.Root, { ...args },
        react_1.default.createElement(__1.Callout.Icon, null,
            react_1.default.createElement(icons_1.InfoCircle16, null)),
        react_1.default.createElement(__1.Callout.Text, null, children))),
};
exports.Size = {
    args: {
        color: __1.calloutRootPropDefs.color.default,
        variant: __1.calloutRootPropDefs.variant.default,
        children: (react_1.default.createElement(react_1.default.Fragment, null,
            "You will need to upgrade to the ",
            react_1.default.createElement(__1.Link, { href: "#" }, "newest Frosted-UI version"),
            " now.")),
    },
    render: ({ children, ...args }) => (react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-3)', alignItems: 'flex-start' } },
        react_1.default.createElement(__1.Callout.Root, { ...args, size: "3" },
            react_1.default.createElement(__1.Callout.Icon, null,
                react_1.default.createElement(__1.Callout.Icon, null,
                    react_1.default.createElement(icons_1.InfoCircle16, null))),
            react_1.default.createElement(__1.Callout.Text, null, children)),
        react_1.default.createElement(__1.Callout.Root, { ...args, size: "2" },
            react_1.default.createElement(__1.Callout.Icon, null,
                react_1.default.createElement(__1.Callout.Icon, null,
                    react_1.default.createElement(icons_1.InfoCircle16, null))),
            react_1.default.createElement(__1.Callout.Text, null, children)),
        react_1.default.createElement(__1.Callout.Root, { ...args, size: "1" },
            react_1.default.createElement(__1.Callout.Icon, null,
                react_1.default.createElement(__1.Callout.Icon, null,
                    react_1.default.createElement(icons_1.InfoCircle16, null))),
            react_1.default.createElement(__1.Callout.Text, null, children)))),
};
exports.Variant = {
    args: {
        color: __1.calloutRootPropDefs.color.default,
        children: (react_1.default.createElement(react_1.default.Fragment, null,
            "You will need to upgrade to the ",
            react_1.default.createElement(__1.Link, { href: "#" }, "newest Frosted-UI version"),
            " now.")),
    },
    render: ({ children, ...args }) => (react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-3)' } },
        react_1.default.createElement(__1.Callout.Root, { ...args, variant: "soft" },
            react_1.default.createElement(__1.Callout.Icon, null,
                react_1.default.createElement(__1.Callout.Icon, null,
                    react_1.default.createElement(icons_1.InfoCircle16, null))),
            react_1.default.createElement(__1.Callout.Text, null, children)),
        react_1.default.createElement(__1.Callout.Root, { ...args, variant: "surface" },
            react_1.default.createElement(__1.Callout.Icon, null,
                react_1.default.createElement(__1.Callout.Icon, null,
                    react_1.default.createElement(icons_1.InfoCircle16, null))),
            react_1.default.createElement(__1.Callout.Text, null, children)),
        react_1.default.createElement(__1.Callout.Root, { ...args, variant: "outline" },
            react_1.default.createElement(__1.Callout.Icon, null,
                react_1.default.createElement(__1.Callout.Icon, null,
                    react_1.default.createElement(icons_1.InfoCircle16, null))),
            react_1.default.createElement(__1.Callout.Text, null, children)))),
};
exports.Color = {
    args: {
        color: __1.calloutRootPropDefs.color.default,
        variant: __1.calloutRootPropDefs.variant.default,
        children: (react_1.default.createElement(react_1.default.Fragment, null,
            "You will need to upgrade to the ",
            react_1.default.createElement(__1.Link, { href: "#" }, "newest Frosted-UI version"),
            " now.")),
    },
    render: ({ children, ...args }) => (react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-3)' } },
        react_1.default.createElement(__1.Callout.Root, { ...args, color: "blue" },
            react_1.default.createElement(__1.Callout.Icon, null,
                react_1.default.createElement(__1.Callout.Icon, null,
                    react_1.default.createElement(icons_1.InfoCircle16, null))),
            react_1.default.createElement(__1.Callout.Text, null, children)),
        react_1.default.createElement(__1.Callout.Root, { ...args, color: "green" },
            react_1.default.createElement(__1.Callout.Icon, null,
                react_1.default.createElement(__1.Callout.Icon, null,
                    react_1.default.createElement(icons_1.InfoCircle16, null))),
            react_1.default.createElement(__1.Callout.Text, null, children)),
        react_1.default.createElement(__1.Callout.Root, { ...args, color: "red" },
            react_1.default.createElement(__1.Callout.Icon, null,
                react_1.default.createElement(__1.Callout.Icon, null,
                    react_1.default.createElement(icons_1.InfoCircle16, null))),
            react_1.default.createElement(__1.Callout.Text, null, children)))),
};
exports.SemanticColor = {
    name: 'Semantic color',
    args: {
        color: __1.calloutRootPropDefs.color.default,
        variant: __1.calloutRootPropDefs.variant.default,
        children: (react_1.default.createElement(react_1.default.Fragment, null,
            "You will need to upgrade to the ",
            react_1.default.createElement(__1.Link, { href: "#" }, "newest Frosted-UI version"),
            " now.")),
    },
    render: ({ children, ...args }) => (react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-3)' } },
        react_1.default.createElement(__1.Callout.Root, { ...args, color: "info" },
            react_1.default.createElement(__1.Callout.Icon, null,
                react_1.default.createElement(__1.Callout.Icon, null,
                    react_1.default.createElement(icons_1.InfoCircle16, null))),
            react_1.default.createElement(__1.Callout.Text, null, children)),
        react_1.default.createElement(__1.Callout.Root, { ...args, color: "success" },
            react_1.default.createElement(__1.Callout.Icon, null,
                react_1.default.createElement(__1.Callout.Icon, null,
                    react_1.default.createElement(icons_1.InfoCircle16, null))),
            react_1.default.createElement(__1.Callout.Text, null, children)),
        react_1.default.createElement(__1.Callout.Root, { ...args, color: "warning" },
            react_1.default.createElement(__1.Callout.Icon, null,
                react_1.default.createElement(__1.Callout.Icon, null,
                    react_1.default.createElement(icons_1.InfoCircle16, null))),
            react_1.default.createElement(__1.Callout.Text, null, children)),
        react_1.default.createElement(__1.Callout.Root, { ...args, color: "danger" },
            react_1.default.createElement(__1.Callout.Icon, null,
                react_1.default.createElement(__1.Callout.Icon, null,
                    react_1.default.createElement(icons_1.InfoCircle16, null))),
            react_1.default.createElement(__1.Callout.Text, null, children)))),
};
exports.HighContrast = {
    name: 'High Contrast',
    args: {
        color: __1.calloutRootPropDefs.color.default,
        variant: __1.calloutRootPropDefs.variant.default,
        children: (react_1.default.createElement(react_1.default.Fragment, null,
            "You will need to upgrade to the ",
            react_1.default.createElement(__1.Link, { href: "#" }, "newest Frosted-UI version"),
            " now.")),
    },
    render: ({ children, ...args }) => (react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-3)' } },
        react_1.default.createElement(__1.Callout.Root, { ...args, variant: "soft" },
            react_1.default.createElement(__1.Callout.Icon, null,
                react_1.default.createElement(__1.Callout.Icon, null,
                    react_1.default.createElement(icons_1.InfoCircle16, null))),
            react_1.default.createElement(__1.Callout.Text, null, children)),
        react_1.default.createElement(__1.Callout.Root, { ...args, variant: "soft", highContrast: true },
            react_1.default.createElement(__1.Callout.Icon, null,
                react_1.default.createElement(__1.Callout.Icon, null,
                    react_1.default.createElement(icons_1.InfoCircle16, null))),
            react_1.default.createElement(__1.Callout.Text, null, children)))),
};
exports.AsAlert = {
    name: 'As Alert',
    render: (args) => (react_1.default.createElement(react_1.default.Fragment, null,
        react_1.default.createElement("div", { style: { maxWidth: 500 } },
            react_1.default.createElement(__1.Text, null,
                "Add a native",
                ' ',
                react_1.default.createElement(__1.Link, { href: "https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/alert_role", target: "_blank" },
                    "WAI-ARIA ",
                    react_1.default.createElement(__1.Code, null, "alert"),
                    " role"),
                ' ',
                "to the callout when the user's immediate attention is required, like when an error message appears. The screen reader will be interrupted, announcing the new content immediately.")),
        react_1.default.createElement("br", null),
        react_1.default.createElement("div", { style: { display: 'inline-block' } },
            react_1.default.createElement(__1.Callout.Root, { ...args, color: "red", role: "alert" },
                react_1.default.createElement(__1.Callout.Icon, null,
                    react_1.default.createElement(__1.Callout.Icon, null, "\uD83D\uDEA8")),
                react_1.default.createElement(__1.Callout.Text, null, "Access denied. Please contact the network administrator to view this page."))))),
};
