// frosted-ui/dist/cjs/components/radio-group/radio-group.stories.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Alignment = exports.HighContrast = exports.Color = exports.Size = exports.Composed = exports.Default = void 0;
const tslib_1 = require("tslib");
const react_1 = tslib_1.__importDefault(require("react"));
const __1 = require("..");
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Controls/RadioGroup',
    component: __1.RadioGroup.Root,
    args: {
        disabled: false,
    },
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
};
exports.default = meta;
// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
exports.Default = {
    args: {
        size: __1.radioGroupPropDefs.size.default,
        color: __1.radioGroupPropDefs.color.default,
        highContrast: __1.radioGroupPropDefs.highContrast.default,
    },
    render: (args) => (react_1.default.createElement(__1.RadioGroup.Root, { defaultValue: "1", ...args },
        react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-2)', flexDirection: 'column' } },
            react_1.default.createElement(__1.RadioGroup.Item, { value: "1" }, "Default"),
            react_1.default.createElement(__1.RadioGroup.Item, { value: "2" }, "Comfortable"),
            react_1.default.createElement(__1.RadioGroup.Item, { value: "3" }, "Compact")))),
};
exports.Composed = {
    args: {
        size: __1.radioGroupPropDefs.size.default,
        color: __1.radioGroupPropDefs.color.default,
        highContrast: __1.radioGroupPropDefs.highContrast.default,
    },
    render: (args) => (react_1.default.createElement(__1.RadioGroup.Root, { defaultValue: "1", ...args },
        react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-2)', flexDirection: 'column' } },
            react_1.default.createElement(__1.Text, { as: "label", size: "2" },
                react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-2)' } },
                    react_1.default.createElement(__1.RadioGroup.Item, { value: "1" }),
                    " Default")),
            react_1.default.createElement(__1.Text, { as: "label", size: "2" },
                react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-2)' } },
                    react_1.default.createElement(__1.RadioGroup.Item, { value: "2" }),
                    " Comfortable")),
            react_1.default.createElement(__1.Text, { as: "label", size: "2" },
                react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-2)' } },
                    react_1.default.createElement(__1.RadioGroup.Item, { value: "3" }),
                    " Compact"))))),
};
exports.Size = {
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', alignItems: 'center', gap: 'var(--space-2)' } },
        react_1.default.createElement(__1.RadioGroup.Root, { ...args, size: "1", defaultValue: "1" },
            react_1.default.createElement(__1.RadioGroup.Item, { value: "1" })),
        react_1.default.createElement(__1.RadioGroup.Root, { ...args, size: "2", defaultValue: "1" },
            react_1.default.createElement(__1.RadioGroup.Item, { value: "1" })),
        react_1.default.createElement(__1.RadioGroup.Root, { ...args, size: "3", defaultValue: "1" },
            react_1.default.createElement(__1.RadioGroup.Item, { value: "1" })))),
};
exports.Color = {
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-2)' } },
        react_1.default.createElement(__1.RadioGroup.Root, { ...args, color: "indigo", defaultValue: "1" },
            react_1.default.createElement(__1.RadioGroup.Item, { value: "1" })),
        react_1.default.createElement(__1.RadioGroup.Root, { ...args, color: "cyan", defaultValue: "1" },
            react_1.default.createElement(__1.RadioGroup.Item, { value: "1" })),
        react_1.default.createElement(__1.RadioGroup.Root, { ...args, color: "orange", defaultValue: "1" },
            react_1.default.createElement(__1.RadioGroup.Item, { value: "1" })),
        react_1.default.createElement(__1.RadioGroup.Root, { ...args, color: "crimson", defaultValue: "1" },
            react_1.default.createElement(__1.RadioGroup.Item, { value: "1" })))),
};
exports.HighContrast = {
    name: 'High Contrast',
    render: (args) => (react_1.default.createElement("div", { style: {
            display: 'inline-grid',
            gridTemplateRows: 'repeat(2, 1fr)',
            gap: 'var(--space-2)',
            gridAutoFlow: 'column',
        } },
        react_1.default.createElement(__1.RadioGroup.Root, { ...args, color: "indigo", defaultValue: "1" },
            react_1.default.createElement(__1.RadioGroup.Item, { value: "1" })),
        react_1.default.createElement(__1.RadioGroup.Root, { ...args, color: "indigo", defaultValue: "1", highContrast: true },
            react_1.default.createElement(__1.RadioGroup.Item, { value: "1" })),
        react_1.default.createElement(__1.RadioGroup.Root, { ...args, color: "cyan", defaultValue: "1" },
            react_1.default.createElement(__1.RadioGroup.Item, { value: "1" })),
        react_1.default.createElement(__1.RadioGroup.Root, { ...args, color: "cyan", defaultValue: "1", highContrast: true },
            react_1.default.createElement(__1.RadioGroup.Item, { value: "1" })),
        react_1.default.createElement(__1.RadioGroup.Root, { ...args, color: "orange", defaultValue: "1" },
            react_1.default.createElement(__1.RadioGroup.Item, { value: "1" })),
        react_1.default.createElement(__1.RadioGroup.Root, { ...args, color: "orange", defaultValue: "1", highContrast: true },
            react_1.default.createElement(__1.RadioGroup.Item, { value: "1" })),
        react_1.default.createElement(__1.RadioGroup.Root, { ...args, color: "crimson", defaultValue: "1" },
            react_1.default.createElement(__1.RadioGroup.Item, { value: "1" })),
        react_1.default.createElement(__1.RadioGroup.Root, { ...args, color: "crimson", defaultValue: "1", highContrast: true },
            react_1.default.createElement(__1.RadioGroup.Item, { value: "1" })))),
};
exports.Alignment = {
    name: 'Alignment with text',
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-3)' } },
        react_1.default.createElement(__1.Text, { style: { marginBottom: 12 } },
            "Composing ",
            react_1.default.createElement(__1.Code, null, "RadioGroup"),
            " within ",
            react_1.default.createElement(__1.Code, null, "Text"),
            " automatically centers it with the first line of text."),
        react_1.default.createElement(__1.RadioGroup.Root, { ...args, size: "1", defaultValue: "1" },
            react_1.default.createElement(__1.Text, { as: "label", size: "2" },
                react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-2)' } },
                    react_1.default.createElement(__1.RadioGroup.Item, { value: "1" }),
                    " Default")),
            react_1.default.createElement(__1.Text, { as: "label", size: "2" },
                react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-2)' } },
                    react_1.default.createElement(__1.RadioGroup.Item, { value: "2" }),
                    " Compact"))),
        react_1.default.createElement(__1.RadioGroup.Root, { ...args, size: "2", defaultValue: "1", style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-1)' } },
            react_1.default.createElement(__1.Text, { as: "label", size: "3" },
                react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-2)' } },
                    react_1.default.createElement(__1.RadioGroup.Item, { value: "1" }),
                    " Default")),
            react_1.default.createElement(__1.Text, { as: "label", size: "3" },
                react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-2)' } },
                    react_1.default.createElement(__1.RadioGroup.Item, { value: "2" }),
                    " Compact"))),
        react_1.default.createElement(__1.RadioGroup.Root, { ...args, size: "3", defaultValue: "1", style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-2)' } },
            react_1.default.createElement(__1.Text, { as: "label", size: "4" },
                react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-2)' } },
                    react_1.default.createElement(__1.RadioGroup.Item, { value: "1" }),
                    " Default")),
            react_1.default.createElement(__1.Text, { as: "label", size: "4" },
                react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-2)' } },
                    react_1.default.createElement(__1.RadioGroup.Item, { value: "2" }),
                    " Compact"))))),
};
