// frosted-ui/dist/cjs/components/alert-dialog/alert-dialog.stories.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.InsetContent = exports.Sizes = exports.Default = void 0;
const tslib_1 = require("tslib");
const react_1 = tslib_1.__importDefault(require("react"));
const __1 = require("..");
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Components/AlertDialog',
    component: __1.AlertDialog.Content,
    args: {
        size: __1.alertDialogContentPropDefs.size.default,
    },
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
};
exports.default = meta;
// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
exports.Default = {
    render: (args) => (react_1.default.createElement(__1.AlertDialog.Root, null,
        react_1.default.createElement(__1.AlertDialog.Trigger, null,
            react_1.default.createElement(__1.Button, { variant: "classic", color: "red" }, "Revoke access")),
        react_1.default.createElement(__1.AlertDialog.Content, { style: { maxWidth: 450 }, ...args },
            react_1.default.createElement(__1.AlertDialog.Title, null, "Revoke access"),
            react_1.default.createElement(__1.AlertDialog.Description, null, "Are you sure? This application will no longer be accessible and any existing sessions will be expired."),
            react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-3)', marginTop: 'var(--space-4)', justifyContent: 'flex-end' } },
                react_1.default.createElement(__1.AlertDialog.Cancel, null,
                    react_1.default.createElement(__1.Button, { variant: "soft", color: "gray" }, "Cancel")),
                react_1.default.createElement(__1.AlertDialog.Action, null,
                    react_1.default.createElement(__1.Button, { variant: "classic", color: "red" }, "Revoke access")))))),
};
exports.Sizes = {
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-4)' } },
        react_1.default.createElement(__1.AlertDialog.Root, null,
            react_1.default.createElement(__1.AlertDialog.Trigger, null,
                react_1.default.createElement(__1.Button, { variant: "classic", color: "red" }, "Size 1")),
            react_1.default.createElement(__1.AlertDialog.Content, { style: { maxWidth: 350 }, ...args, size: "1" },
                react_1.default.createElement(__1.AlertDialog.Title, null, "Revoke access"),
                react_1.default.createElement(__1.AlertDialog.Description, null, "Are you sure? This application will no longer be accessible and any existing sessions will be expired."),
                react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-2)', justifyContent: 'flex-end' } },
                    react_1.default.createElement(__1.AlertDialog.Cancel, null,
                        react_1.default.createElement(__1.Button, { size: "1", variant: "soft", color: "gray" }, "Cancel")),
                    react_1.default.createElement(__1.AlertDialog.Action, null,
                        react_1.default.createElement(__1.Button, { size: "1", variant: "classic", color: "red" }, "Revoke access"))))),
        react_1.default.createElement(__1.AlertDialog.Root, null,
            react_1.default.createElement(__1.AlertDialog.Trigger, null,
                react_1.default.createElement(__1.Button, { variant: "classic", color: "red" }, "Size 2")),
            react_1.default.createElement(__1.AlertDialog.Content, { style: { maxWidth: 350 }, size: "2" },
                react_1.default.createElement(__1.AlertDialog.Title, null, "Revoke access"),
                react_1.default.createElement(__1.AlertDialog.Description, null, "Are you sure? This application will no longer be accessible and any existing sessions will be expired."),
                react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-2)', justifyContent: 'flex-end' } },
                    react_1.default.createElement(__1.AlertDialog.Cancel, null,
                        react_1.default.createElement(__1.Button, { size: "2", variant: "soft", color: "gray" }, "Cancel")),
                    react_1.default.createElement(__1.AlertDialog.Action, null,
                        react_1.default.createElement(__1.Button, { size: "2", variant: "classic", color: "red" }, "Revoke access"))))),
        react_1.default.createElement(__1.AlertDialog.Root, null,
            react_1.default.createElement(__1.AlertDialog.Trigger, null,
                react_1.default.createElement(__1.Button, { variant: "classic", color: "red" }, "Size 3")),
            react_1.default.createElement(__1.AlertDialog.Content, { style: { maxWidth: 350 }, size: "3" },
                react_1.default.createElement(__1.AlertDialog.Title, null, "Revoke access"),
                react_1.default.createElement(__1.AlertDialog.Description, null, "Are you sure? This application will no longer be accessible and any existing sessions will be expired."),
                react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-3)', justifyContent: 'flex-end' } },
                    react_1.default.createElement(__1.AlertDialog.Cancel, null,
                        react_1.default.createElement(__1.Button, { size: "2", variant: "soft", color: "gray" }, "Cancel")),
                    react_1.default.createElement(__1.AlertDialog.Action, null,
                        react_1.default.createElement(__1.Button, { size: "2", variant: "classic", color: "red" }, "Revoke access"))))),
        react_1.default.createElement(__1.AlertDialog.Root, null,
            react_1.default.createElement(__1.AlertDialog.Trigger, null,
                react_1.default.createElement(__1.Button, { variant: "classic", color: "red" }, "Size 4")),
            react_1.default.createElement(__1.AlertDialog.Content, { style: { maxWidth: 350 }, size: "4" },
                react_1.default.createElement(__1.AlertDialog.Title, null, "Revoke access"),
                react_1.default.createElement(__1.AlertDialog.Description, null, "Are you sure? This application will no longer be accessible and any existing sessions will be expired."),
                react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-3)', justifyContent: 'flex-end' } },
                    react_1.default.createElement(__1.AlertDialog.Cancel, null,
                        react_1.default.createElement(__1.Button, { size: "3", variant: "soft", color: "gray" }, "Cancel")),
                    react_1.default.createElement(__1.AlertDialog.Action, null,
                        react_1.default.createElement(__1.Button, { size: "3", variant: "classic", color: "red" }, "Revoke access"))))))),
};
exports.InsetContent = {
    name: 'With inset content',
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', alignItems: 'center', gap: 'var(--space-6)' } },
        react_1.default.createElement(__1.Text, null,
            "Use the ",
            react_1.default.createElement(__1.Code, null, "Inset"),
            " component to align content flush with the sides of the dialog (bypassing its padding)."),
        react_1.default.createElement("div", null,
            react_1.default.createElement(__1.AlertDialog.Root, null,
                react_1.default.createElement(__1.AlertDialog.Trigger, null,
                    react_1.default.createElement(__1.Button, { color: "red", variant: "classic" }, "Delete users")),
                react_1.default.createElement(__1.AlertDialog.Content, { style: { maxWidth: 500 }, ...args },
                    react_1.default.createElement(__1.AlertDialog.Title, null, "Delete Users"),
                    react_1.default.createElement(__1.AlertDialog.Description, null, "Are you sure you want to delete these users? This action is permanent and cannot be undone."),
                    react_1.default.createElement(__1.Inset, { side: "x", style: { marginTop: 'var(--space-5)', marginBottom: 'var(--space-5)' } },
                        react_1.default.createElement(__1.Table.Root, null,
                            react_1.default.createElement(__1.Table.Header, null,
                                react_1.default.createElement(__1.Table.Row, null,
                                    react_1.default.createElement(__1.Table.ColumnHeaderCell, null, "Full name"),
                                    react_1.default.createElement(__1.Table.ColumnHeaderCell, null, "Email"),
                                    react_1.default.createElement(__1.Table.ColumnHeaderCell, null, "Group"))),
                            react_1.default.createElement(__1.Table.Body, null,
                                react_1.default.createElement(__1.Table.Row, null,
                                    react_1.default.createElement(__1.Table.RowHeaderCell, null, "Danilo Sousa"),
                                    react_1.default.createElement(__1.Table.Cell, null, "danilo@example.com"),
                                    react_1.default.createElement(__1.Table.Cell, null, "Developer")),
                                react_1.default.createElement(__1.Table.Row, null,
                                    react_1.default.createElement(__1.Table.RowHeaderCell, null, "Zahra Ambessa"),
                                    react_1.default.createElement(__1.Table.Cell, null, "zahra@example.com"),
                                    react_1.default.createElement(__1.Table.Cell, null, "Admin"))))),
                    react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-3)', justifyContent: 'flex-end' } },
                        react_1.default.createElement(__1.AlertDialog.Cancel, null,
                            react_1.default.createElement(__1.Button, { variant: "soft", color: "gray" }, "Cancel")),
                        react_1.default.createElement(__1.AlertDialog.Action, null,
                            react_1.default.createElement(__1.Button, { color: "red", variant: "classic" }, "Delete users")))))))),
};
