// frosted-ui/dist/cjs/components/text-field/text-field.stories.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Color = exports.Variant = exports.Size = exports.Default = void 0;
const tslib_1 = require("tslib");
const icons_1 = require("@frosted-ui/icons");
const react_1 = tslib_1.__importDefault(require("react"));
const __1 = require("..");
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Controls/TextField',
    component: __1.TextField.Root,
    args: {
        size: __1.textFieldPropDefs.size.default,
        variant: __1.textFieldPropDefs.variant.default,
        color: __1.textFieldPropDefs.color.default,
    },
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
};
exports.default = meta;
// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
exports.Default = {
    render: (args) => (react_1.default.createElement("div", { style: { width: 300 } },
        react_1.default.createElement(__1.TextField.Root, { ...args },
            react_1.default.createElement(__1.TextField.Slot, null,
                react_1.default.createElement(icons_1.MagnifyingGlass16, null)),
            react_1.default.createElement(__1.TextField.Input, { placeholder: "Search the docs\u2026" })))),
};
exports.Size = {
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-3)', maxWidth: 400 } },
        react_1.default.createElement("div", { style: { display: 'flex', alignItems: 'center', gap: 'var(--space-2)' } },
            react_1.default.createElement(__1.TextField.Root, { ...args },
                react_1.default.createElement(__1.TextField.Slot, null,
                    react_1.default.createElement(icons_1.MagnifyingGlass12, null)),
                react_1.default.createElement(__1.TextField.Input, { placeholder: "Search the docs\u2026", size: "1" })),
            react_1.default.createElement(__1.Button, { size: "1", variant: args.variant }, "Search")),
        react_1.default.createElement("div", { style: { display: 'flex', alignItems: 'center', gap: 'var(--space-2)' } },
            react_1.default.createElement(__1.TextField.Root, { ...args },
                react_1.default.createElement(__1.TextField.Slot, null,
                    react_1.default.createElement(icons_1.MagnifyingGlass16, null)),
                react_1.default.createElement(__1.TextField.Input, { placeholder: "Search the docs\u2026", size: "2" }),
                react_1.default.createElement(__1.TextField.Slot, null,
                    react_1.default.createElement(__1.IconButton, { color: "gray", size: "1", variant: "ghost" },
                        react_1.default.createElement(icons_1.ThreeDotsHorizontal16, null)))),
            react_1.default.createElement(__1.Button, { size: "2", variant: args.variant }, "Search")),
        react_1.default.createElement("div", { style: { display: 'flex', alignItems: 'center', gap: 'var(--space-2)' } },
            react_1.default.createElement(__1.TextField.Root, { ...args },
                react_1.default.createElement(__1.TextField.Slot, null,
                    react_1.default.createElement(icons_1.MagnifyingGlass20, null)),
                react_1.default.createElement(__1.TextField.Input, { placeholder: "Search the docs\u2026", size: "3" }),
                react_1.default.createElement(__1.TextField.Slot, { style: { paddingRight: 'var(--space-3)' } },
                    react_1.default.createElement(__1.IconButton, { color: "gray", size: "2", variant: "ghost" },
                        react_1.default.createElement(icons_1.ThreeDotsHorizontal20, null)))),
            react_1.default.createElement(__1.Button, { size: "3", variant: args.variant }, "Search")))),
};
exports.Variant = {
    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    render: ({ ref, ...args }) => (react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'row', gap: 'var(--space-5)' } },
        react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-3)', maxWidth: 400 } },
            react_1.default.createElement(__1.TextField.Input, { placeholder: "Search the docs\u2026", ...args, variant: "surface" }),
            react_1.default.createElement(__1.TextField.Input, { placeholder: "Search the docs\u2026", ...args, variant: "soft" })),
        react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-3)', maxWidth: 400 } },
            react_1.default.createElement(__1.TextField.Input, { disabled: true, placeholder: "Search the docs\u2026", ...args, variant: "surface" }),
            react_1.default.createElement(__1.TextField.Input, { disabled: true, placeholder: "Search the docs\u2026", ...args, variant: "soft" })))),
};
exports.Color = {
    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    render: ({ ref, ...args }) => (react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-3)', maxWidth: 400 } },
        react_1.default.createElement(__1.TextField.Input, { ...args, placeholder: "Search the docs\u2026", color: "indigo" }),
        react_1.default.createElement(__1.TextField.Input, { ...args, placeholder: "Search the docs\u2026", color: "green" }),
        react_1.default.createElement(__1.TextField.Input, { ...args, placeholder: "Search the docs\u2026", color: "red" }))),
};
