// frosted-ui/dist/cjs/components/widget-stack/widget-stack.stories.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AutoPlay = exports.WithControls = exports.SingleWidget = exports.Orientation = exports.Default = void 0;
const tslib_1 = require("tslib");
const react_1 = tslib_1.__importDefault(require("react"));
const __1 = require("..");
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Components/WidgetStack',
    component: __1.WidgetStack.Stack,
    args: {
        orientation: __1.widgetStackRootPropDefs.orientation.default,
    },
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
};
exports.default = meta;
// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
exports.Default = {
    args: {},
    render: (args) => (react_1.default.createElement(__1.WidgetStack.Root, { ...args },
        react_1.default.createElement(__1.WidgetStack.Stack, { style: {
                width: 400,
                height: 200,
            } },
            react_1.default.createElement(__1.WidgetStack.Item, null,
                react_1.default.createElement("div", { style: {
                        display: 'flex',
                        flexDirection: 'column',
                        width: '100%',
                        height: '100%',
                        justifyContent: 'space-between',
                        padding: 'var(--space-4)',
                        background: 'linear-gradient(var(--blue-9), var(--blue-6))',
                        color: 'var(--blue-9-contrast)',
                    } },
                    react_1.default.createElement("div", { style: {
                            display: 'flex',
                            flexDirection: 'column',
                            width: '100%',
                            justifyContent: 'space-between',
                            gap: 'var(--space-1)',
                        } },
                        react_1.default.createElement(__1.Text, { weight: "bold", size: "5" }, "Wroc\u0142aw"),
                        react_1.default.createElement(__1.Text, { weight: "light", size: "9" }, "24")),
                    react_1.default.createElement("div", { style: {
                            display: 'flex',
                            flexDirection: 'column',
                            width: '100%',
                            justifyContent: 'space-between',
                            gap: 'var(--space-1)',
                        } },
                        react_1.default.createElement(__1.Text, { weight: "medium", size: "3" }, "Sunny"),
                        react_1.default.createElement(__1.Text, { weight: "medium", size: "3" }, "H:73 L:55")))),
            react_1.default.createElement(__1.WidgetStack.Item, null,
                react_1.default.createElement("div", { style: {
                        display: 'flex',
                        flexDirection: 'column',
                        width: '100%',
                        height: '100%',
                        justifyContent: 'space-between',
                    } },
                    react_1.default.createElement("div", { style: {
                            display: 'flex',
                            flexDirection: 'column',
                            width: '100%',
                            justifyContent: 'space-between',
                            gap: 'var(--space-1)',
                            backgroundImage: 'url(https://images.unsplash.com/photo-1502823403499-6ccfcf4fb453?&w=256&h=256&q=70&crop=focalpoint&fp-x=0.5&fp-y=0.3&fp-z=1&fit=crop)',
                            backgroundSize: 'cover',
                            height: '50%',
                        } }),
                    react_1.default.createElement("div", { style: {
                            display: 'flex',
                            flexDirection: 'column',
                            width: '100%',
                            padding: 'var(--space-4)',
                            height: '50%',
                            background: 'var(--gray-2)',
                        } },
                        react_1.default.createElement(__1.Text, { weight: "medium", size: "4", trim: "start" }, "Travis S."),
                        react_1.default.createElement(__1.Text, { weight: "medium", size: "2", color: "gray" }, "4am in Toronto"),
                        react_1.default.createElement(__1.IconButton, { variant: "soft", style: {
                                position: 'absolute',
                                right: 16,
                                bottom: 16,
                                borderRadius: '50%',
                            }, color: "blue", size: "4" }, "A")))),
            react_1.default.createElement(__1.WidgetStack.Item, null,
                react_1.default.createElement("div", { style: {
                        display: 'flex',
                        flexDirection: 'column',
                        width: '100%',
                        height: '100%',
                        justifyContent: 'center',
                        alignItems: 'center',
                        background: 'var(--grass-9)',
                    } },
                    react_1.default.createElement("span", { style: {
                            fontSize: 96,
                            lineHeight: 1,
                        } }, "\uD83C\uDFDD\uFE0F"))),
            react_1.default.createElement(__1.WidgetStack.Item, { style: { fontSize: 60 } }, "\uD83C\uDFDD\uFE0F\uD83C\uDFF4\u200D\u2620\uFE0F\u269C\uFE0F")))),
};
exports.Orientation = {
    args: {},
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-8)' } },
        react_1.default.createElement(__1.WidgetStack.Root, { ...args, orientation: "horizontal" },
            react_1.default.createElement(__1.WidgetStack.Stack, { style: {
                    width: 200,
                    height: 200,
                } },
                react_1.default.createElement(__1.WidgetStack.Item, null,
                    react_1.default.createElement("div", { style: {
                            display: 'flex',
                            flexDirection: 'column',
                            width: '100%',
                            height: '100%',
                            justifyContent: 'space-between',
                            padding: 'var(--space-4)',
                            background: 'linear-gradient(var(--blue-9), var(--blue-6))',
                            color: 'var(--blue-9-contrast)',
                        } },
                        react_1.default.createElement("div", { style: {
                                display: 'flex',
                                flexDirection: 'column',
                                width: '100%',
                                justifyContent: 'space-between',
                                gap: 'var(--space-1)',
                            } },
                            react_1.default.createElement(__1.Text, { weight: "bold", size: "5" }, "Wroc\u0142aw"),
                            react_1.default.createElement(__1.Text, { weight: "light", size: "9" }, "24")),
                        react_1.default.createElement("div", { style: {
                                display: 'flex',
                                flexDirection: 'column',
                                width: '100%',
                                justifyContent: 'space-between',
                                gap: 'var(--space-1)',
                            } },
                            react_1.default.createElement(__1.Text, { weight: "medium", size: "3" }, "Sunny"),
                            react_1.default.createElement(__1.Text, { weight: "medium", size: "3" }, "H:73 L:55")))),
                react_1.default.createElement(__1.WidgetStack.Item, null,
                    react_1.default.createElement("div", { style: {
                            display: 'flex',
                            flexDirection: 'column',
                            width: '100%',
                            height: '100%',
                            justifyContent: 'space-between',
                        } },
                        react_1.default.createElement("div", { style: {
                                display: 'flex',
                                flexDirection: 'column',
                                width: '100%',
                                justifyContent: 'space-between',
                                gap: 'var(--space-1)',
                                backgroundImage: 'url(https://images.unsplash.com/photo-1502823403499-6ccfcf4fb453?&w=256&h=256&q=70&crop=focalpoint&fp-x=0.5&fp-y=0.3&fp-z=1&fit=crop)',
                                backgroundSize: 'cover',
                                height: '50%',
                            } }),
                        react_1.default.createElement("div", { style: {
                                display: 'flex',
                                flexDirection: 'column',
                                width: '100%',
                                padding: 'var(--space-4)',
                                height: '50%',
                                background: 'var(--gray-2)',
                            } },
                            react_1.default.createElement(__1.Text, { weight: "medium", size: "4", trim: "start" }, "Travis S."),
                            react_1.default.createElement(__1.Text, { weight: "medium", size: "2", color: "gray" }, "4am in Toronto"),
                            react_1.default.createElement(__1.IconButton, { variant: "soft", style: {
                                    position: 'absolute',
                                    right: 16,
                                    bottom: 16,
                                    borderRadius: '50%',
                                }, color: "blue", size: "4" }, "A")))))),
        react_1.default.createElement(__1.WidgetStack.Root, { orientation: "vertical" },
            react_1.default.createElement(__1.WidgetStack.Stack, { style: {
                    width: 200,
                    height: 200,
                } },
                react_1.default.createElement(__1.WidgetStack.Item, null,
                    react_1.default.createElement("div", { style: {
                            display: 'flex',
                            flexDirection: 'column',
                            width: '100%',
                            height: '100%',
                            justifyContent: 'space-between',
                            padding: 'var(--space-4)',
                            background: 'linear-gradient(var(--blue-9), var(--blue-6))',
                            color: 'var(--blue-9-contrast)',
                        } },
                        react_1.default.createElement("div", { style: {
                                display: 'flex',
                                flexDirection: 'column',
                                width: '100%',
                                justifyContent: 'space-between',
                                gap: 'var(--space-1)',
                            } },
                            react_1.default.createElement(__1.Text, { weight: "bold", size: "5" }, "Wroc\u0142aw"),
                            react_1.default.createElement(__1.Text, { weight: "light", size: "9" }, "24")),
                        react_1.default.createElement("div", { style: {
                                display: 'flex',
                                flexDirection: 'column',
                                width: '100%',
                                justifyContent: 'space-between',
                                gap: 'var(--space-1)',
                            } },
                            react_1.default.createElement(__1.Text, { weight: "medium", size: "3" }, "Sunny"),
                            react_1.default.createElement(__1.Text, { weight: "medium", size: "3" }, "H:73 L:55")))),
                react_1.default.createElement(__1.WidgetStack.Item, null,
                    react_1.default.createElement("div", { style: {
                            display: 'flex',
                            flexDirection: 'column',
                            width: '100%',
                            height: '100%',
                            justifyContent: 'space-between',
                        } },
                        react_1.default.createElement("div", { style: {
                                display: 'flex',
                                flexDirection: 'column',
                                width: '100%',
                                justifyContent: 'space-between',
                                gap: 'var(--space-1)',
                                backgroundImage: 'url(https://images.unsplash.com/photo-1502823403499-6ccfcf4fb453?&w=256&h=256&q=70&crop=focalpoint&fp-x=0.5&fp-y=0.3&fp-z=1&fit=crop)',
                                backgroundSize: 'cover',
                                height: '50%',
                            } }),
                        react_1.default.createElement("div", { style: {
                                display: 'flex',
                                flexDirection: 'column',
                                width: '100%',
                                padding: 'var(--space-4)',
                                height: '50%',
                                background: 'var(--gray-2)',
                            } },
                            react_1.default.createElement(__1.Text, { weight: "medium", size: "4", trim: "start" }, "Travis S."),
                            react_1.default.createElement(__1.Text, { weight: "medium", size: "2", color: "gray" }, "4am in Toronto"),
                            react_1.default.createElement(__1.IconButton, { variant: "soft", style: {
                                    position: 'absolute',
                                    right: 16,
                                    bottom: 16,
                                    borderRadius: '50%',
                                }, color: "blue", size: "4" }, "A")))),
                react_1.default.createElement(__1.WidgetStack.Item, null,
                    react_1.default.createElement("div", { style: {
                            display: 'flex',
                            flexDirection: 'column',
                            width: '100%',
                            height: '100%',
                            justifyContent: 'center',
                            alignItems: 'center',
                            background: 'var(--grass-9)',
                        } },
                        react_1.default.createElement("span", { style: {
                                fontSize: 96,
                                lineHeight: 1,
                            } }, "\uD83C\uDFDD\uFE0F"))),
                react_1.default.createElement(__1.WidgetStack.Item, { style: { fontSize: 60 } }, "\uD83C\uDFDD\uFE0F\uD83C\uDFF4\u200D\u2620\uFE0F\u269C\uFE0F"))))),
};
exports.SingleWidget = {
    args: {},
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-8)' } },
        react_1.default.createElement(__1.Text, null, "When single widget is inside, the scrolling and animations are blocked automatically."),
        react_1.default.createElement(__1.WidgetStack.Root, { ...args },
            react_1.default.createElement(__1.WidgetStack.Stack, { style: {
                    width: 400,
                    height: 200,
                } },
                react_1.default.createElement(__1.WidgetStack.Item, null,
                    react_1.default.createElement("div", { style: {
                            display: 'flex',
                            flexDirection: 'column',
                            width: '100%',
                            height: '100%',
                            justifyContent: 'space-between',
                            padding: 'var(--space-4)',
                            background: 'linear-gradient(var(--blue-9), var(--blue-6))',
                            color: 'var(--blue-9-contrast)',
                        } },
                        react_1.default.createElement("div", { style: {
                                display: 'flex',
                                flexDirection: 'column',
                                width: '100%',
                                justifyContent: 'space-between',
                                gap: 'var(--space-1)',
                            } },
                            react_1.default.createElement(__1.Text, { weight: "bold", size: "5" }, "Wroc\u0142aw"),
                            react_1.default.createElement(__1.Text, { weight: "light", size: "9" }, "24")),
                        react_1.default.createElement("div", { style: {
                                display: 'flex',
                                flexDirection: 'column',
                                width: '100%',
                                justifyContent: 'space-between',
                                gap: 'var(--space-1)',
                            } },
                            react_1.default.createElement(__1.Text, { weight: "medium", size: "3" }, "Sunny"),
                            react_1.default.createElement(__1.Text, { weight: "medium", size: "3" }, "H:73 L:55")))))))),
};
exports.WithControls = {
    args: {},
    render: (args) => (react_1.default.createElement(__1.WidgetStack.Root, { orientation: "horizontal", ...args },
        react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'row', gap: 'var(--space-4)', alignItems: 'center' } },
            react_1.default.createElement(__1.WidgetStack.Prev, null,
                react_1.default.createElement(__1.IconButton, { variant: "soft", color: "gray", style: { borderRadius: '50%' } }, '<')),
            react_1.default.createElement(__1.WidgetStack.Stack, { style: {
                    width: 400,
                    height: 200,
                } },
                react_1.default.createElement(__1.WidgetStack.Item, null,
                    react_1.default.createElement("div", { style: {
                            display: 'flex',
                            flexDirection: 'column',
                            width: '100%',
                            height: '100%',
                            justifyContent: 'space-between',
                            padding: 'var(--space-4)',
                            background: 'linear-gradient(var(--blue-9), var(--blue-6))',
                            color: 'var(--blue-9-contrast)',
                        } },
                        react_1.default.createElement("div", { style: {
                                display: 'flex',
                                flexDirection: 'column',
                                width: '100%',
                                justifyContent: 'space-between',
                                gap: 'var(--space-1)',
                            } },
                            react_1.default.createElement(__1.Text, { weight: "bold", size: "5" }, "Wroc\u0142aw"),
                            react_1.default.createElement(__1.Text, { weight: "light", size: "9" }, "24")),
                        react_1.default.createElement("div", { style: {
                                display: 'flex',
                                flexDirection: 'column',
                                width: '100%',
                                justifyContent: 'space-between',
                                gap: 'var(--space-1)',
                            } },
                            react_1.default.createElement(__1.Text, { weight: "medium", size: "3" }, "Sunny"),
                            react_1.default.createElement(__1.Text, { weight: "medium", size: "3" }, "H:73 L:55")))),
                react_1.default.createElement(__1.WidgetStack.Item, null,
                    react_1.default.createElement("div", { style: {
                            display: 'flex',
                            flexDirection: 'column',
                            width: '100%',
                            height: '100%',
                            justifyContent: 'space-between',
                        } },
                        react_1.default.createElement("div", { style: {
                                display: 'flex',
                                flexDirection: 'column',
                                width: '100%',
                                justifyContent: 'space-between',
                                gap: 'var(--space-1)',
                                backgroundImage: 'url(https://images.unsplash.com/photo-1502823403499-6ccfcf4fb453?&w=256&h=256&q=70&crop=focalpoint&fp-x=0.5&fp-y=0.3&fp-z=1&fit=crop)',
                                backgroundSize: 'cover',
                                height: '50%',
                            } }),
                        react_1.default.createElement("div", { style: {
                                display: 'flex',
                                flexDirection: 'column',
                                width: '100%',
                                padding: 'var(--space-4)',
                                height: '50%',
                                background: 'var(--gray-2)',
                            } },
                            react_1.default.createElement(__1.Text, { weight: "medium", size: "4", trim: "start" }, "Travis S."),
                            react_1.default.createElement(__1.Text, { weight: "medium", size: "2", color: "gray" }, "4am in Toronto"),
                            react_1.default.createElement(__1.IconButton, { variant: "soft", style: {
                                    position: 'absolute',
                                    right: 16,
                                    bottom: 16,
                                    borderRadius: '50%',
                                }, color: "blue", size: "4" }, "A")))),
                react_1.default.createElement(__1.WidgetStack.Item, null,
                    react_1.default.createElement("div", { style: {
                            display: 'flex',
                            flexDirection: 'column',
                            width: '100%',
                            height: '100%',
                            justifyContent: 'center',
                            alignItems: 'center',
                            background: 'var(--grass-9)',
                        } },
                        react_1.default.createElement("span", { style: {
                                fontSize: 96,
                                lineHeight: 1,
                            } }, "\uD83C\uDFDD\uFE0F"))),
                react_1.default.createElement(__1.WidgetStack.Item, { style: { fontSize: 60 } }, "\uD83C\uDFDD\uFE0F\uD83C\uDFF4\u200D\u2620\uFE0F\u269C\uFE0F")),
            react_1.default.createElement(__1.WidgetStack.Next, null,
                react_1.default.createElement(__1.IconButton, { variant: "soft", color: "gray", style: { borderRadius: '50%' } }, '>'))))),
};
exports.AutoPlay = {
    args: {},
    render: (args) => (react_1.default.createElement(__1.WidgetStack.Root, { orientation: "horizontal", ...args, autoPlay: 4000 },
        react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'row', gap: 'var(--space-4)', alignItems: 'center' } },
            react_1.default.createElement(__1.WidgetStack.Prev, null,
                react_1.default.createElement(__1.IconButton, { variant: "soft", color: "gray", style: { borderRadius: '50%' } }, '<')),
            react_1.default.createElement(__1.WidgetStack.Stack, { style: {
                    width: 400,
                    height: 200,
                } },
                react_1.default.createElement(__1.WidgetStack.Item, null,
                    react_1.default.createElement("div", { style: {
                            display: 'flex',
                            flexDirection: 'column',
                            width: '100%',
                            height: '100%',
                            justifyContent: 'space-between',
                            padding: 'var(--space-4)',
                            background: 'linear-gradient(var(--blue-9), var(--blue-6))',
                            color: 'var(--blue-9-contrast)',
                        } },
                        react_1.default.createElement("div", { style: {
                                display: 'flex',
                                flexDirection: 'column',
                                width: '100%',
                                justifyContent: 'space-between',
                                gap: 'var(--space-1)',
                            } },
                            react_1.default.createElement(__1.Text, { weight: "bold", size: "5" }, "Wroc\u0142aw"),
                            react_1.default.createElement(__1.Text, { weight: "light", size: "9" }, "24")),
                        react_1.default.createElement("div", { style: {
                                display: 'flex',
                                flexDirection: 'column',
                                width: '100%',
                                justifyContent: 'space-between',
                                gap: 'var(--space-1)',
                            } },
                            react_1.default.createElement(__1.Text, { weight: "medium", size: "3" }, "Sunny"),
                            react_1.default.createElement(__1.Text, { weight: "medium", size: "3" }, "H:73 L:55")))),
                react_1.default.createElement(__1.WidgetStack.Item, null,
                    react_1.default.createElement("div", { style: {
                            display: 'flex',
                            flexDirection: 'column',
                            width: '100%',
                            height: '100%',
                            justifyContent: 'space-between',
                        } },
                        react_1.default.createElement("div", { style: {
                                display: 'flex',
                                flexDirection: 'column',
                                width: '100%',
                                justifyContent: 'space-between',
                                gap: 'var(--space-1)',
                                backgroundImage: 'url(https://images.unsplash.com/photo-1502823403499-6ccfcf4fb453?&w=256&h=256&q=70&crop=focalpoint&fp-x=0.5&fp-y=0.3&fp-z=1&fit=crop)',
                                backgroundSize: 'cover',
                                height: '50%',
                            } }),
                        react_1.default.createElement("div", { style: {
                                display: 'flex',
                                flexDirection: 'column',
                                width: '100%',
                                padding: 'var(--space-4)',
                                height: '50%',
                                background: 'var(--gray-2)',
                            } },
                            react_1.default.createElement(__1.Text, { weight: "medium", size: "4", trim: "start" }, "Travis S."),
                            react_1.default.createElement(__1.Text, { weight: "medium", size: "2", color: "gray" }, "4am in Toronto"),
                            react_1.default.createElement(__1.IconButton, { variant: "soft", style: {
                                    position: 'absolute',
                                    right: 16,
                                    bottom: 16,
                                    borderRadius: '50%',
                                }, color: "blue", size: "4" }, "A")))),
                react_1.default.createElement(__1.WidgetStack.Item, null,
                    react_1.default.createElement("div", { style: {
                            display: 'flex',
                            flexDirection: 'column',
                            width: '100%',
                            height: '100%',
                            justifyContent: 'center',
                            alignItems: 'center',
                            background: 'var(--grass-9)',
                        } },
                        react_1.default.createElement("span", { style: {
                                fontSize: 96,
                                lineHeight: 1,
                            } }, "\uD83C\uDFDD\uFE0F"))),
                react_1.default.createElement(__1.WidgetStack.Item, { style: { fontSize: 60 } }, "\uD83C\uDFDD\uFE0F\uD83C\uDFF4\u200D\u2620\uFE0F\u269C\uFE0F")),
            react_1.default.createElement(__1.WidgetStack.Next, null,
                react_1.default.createElement(__1.IconButton, { variant: "soft", color: "gray", style: { borderRadius: '50%' } }, '>'))))),
};
