// frosted-ui/dist/esm/components/sheet/sheet.stories.js
import React from 'react';
import { Badge, Button, Checkbox, Inset, ScrollArea, Sheet, Table, Text, TextField } from '..';
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Components/Sheet',
    component: Sheet.Content,
    args: {
    // size: dialogContentPropDefs.size.default,
    },
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
};
export default meta;
// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
export const Default = {
    render: (args) => (React.createElement(Sheet.Root, null,
        React.createElement(Sheet.Trigger, null,
            React.createElement(Button, null, "Edit profile")),
        React.createElement(Sheet.Content, { ...args },
            React.createElement(Sheet.Header, null,
                React.createElement(Sheet.Title, null, "Edit profile"),
                React.createElement(Sheet.Description, null, "Make changes to your profile.")),
            React.createElement(Sheet.Body, null,
                React.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-3)' } },
                    React.createElement("label", null,
                        React.createElement(Text, { as: "div", size: "2", style: { marginBottom: 'var(--space-1)' }, weight: "bold" }, "Name"),
                        React.createElement(TextField.Input, { size: "3", defaultValue: "Freja Johnsen", placeholder: "Enter your full name" })),
                    React.createElement("label", null,
                        React.createElement(Text, { as: "div", size: "2", style: { marginBottom: 'var(--space-1)' }, weight: "bold" }, "Email"),
                        React.createElement(TextField.Input, { size: "3", defaultValue: "freja@example.com", placeholder: "Enter your email" }))),
                React.createElement("div", { style: { display: 'flex', gap: 'var(--space-3)', marginTop: 'var(--space-4)', justifyContent: 'flex-end' } },
                    React.createElement(Sheet.Close, null,
                        React.createElement(Button, { variant: "soft", color: "gray", onClick: () => alert('Cancel') }, "Cancel")),
                    React.createElement(Sheet.Close, null,
                        React.createElement(Button, { onClick: () => alert('Save') }, "Save"))))))),
};
export const Nested = {
    render: (args) => (React.createElement(Sheet.Root, null,
        React.createElement(Sheet.Trigger, null,
            React.createElement(Button, null, "Edit profile")),
        React.createElement(Sheet.Content, { ...args },
            React.createElement(Sheet.Header, null,
                React.createElement(Sheet.Title, null, "Message request"),
                React.createElement(Sheet.Description, null, "Artur wants to message you")),
            React.createElement(Sheet.Body, null,
                React.createElement("div", { style: { display: 'flex', gap: 'var(--space-3)', marginTop: 'var(--space-4)', justifyContent: 'flex-end' } },
                    React.createElement(Sheet.NestedRoot, null,
                        React.createElement(Sheet.Trigger, null,
                            React.createElement(Button, { variant: "soft", color: "danger" }, "Block")),
                        React.createElement(Sheet.Content, null,
                            React.createElement(Sheet.Header, null,
                                React.createElement(Sheet.Title, null, "Are you sure you want to block Artur?"),
                                React.createElement(Sheet.Description, null, "Artur won't be able to message you unless you unblock him.")),
                            React.createElement(Sheet.Body, null,
                                React.createElement("div", { style: {
                                        display: 'flex',
                                        gap: 'var(--space-3)',
                                        marginTop: 'var(--space-4)',
                                        justifyContent: 'flex-end',
                                    } },
                                    React.createElement(Sheet.Close, null,
                                        React.createElement(Button, { size: "3" }, "Cancel")),
                                    React.createElement(Sheet.Close, null,
                                        React.createElement(Button, { variant: "classic", color: "danger", size: "3" }, "Block")))))),
                    React.createElement(Sheet.Close, null,
                        React.createElement(Button, { onClick: () => alert('Save') }, "Save"))))))),
};
export const InsetContent = {
    name: 'With inset content',
    render: (args) => (React.createElement(Sheet.Root, null,
        React.createElement(Sheet.Trigger, null,
            React.createElement(Button, null, "View users")),
        React.createElement(Sheet.Content, { ...args },
            React.createElement(Sheet.Header, null,
                React.createElement(Sheet.Title, null, "Users"),
                React.createElement(Sheet.Description, null, "The following users have access to this project.")),
            React.createElement(Sheet.Body, null,
                React.createElement(Inset, { side: "x", style: { marginBottom: 'var(--space-4)' } },
                    React.createElement(Table.Root, { variant: "ghost", size: "1" },
                        React.createElement(ScrollArea, { scrollbars: "horizontal" },
                            React.createElement(Table.Table, null,
                                React.createElement(Table.Header, null,
                                    React.createElement(Table.Row, null,
                                        React.createElement(Table.ColumnHeaderCell, null,
                                            React.createElement("div", { style: { display: 'flex', alignItems: 'center', gap: 'var(--space-3)' } },
                                                React.createElement(Checkbox, null),
                                                "Full name")),
                                        React.createElement(Table.ColumnHeaderCell, null, "Email"),
                                        React.createElement(Table.ColumnHeaderCell, null, "Group"))),
                                React.createElement(Table.Body, null,
                                    React.createElement(Table.Row, null,
                                        React.createElement(Table.RowHeaderCell, null,
                                            React.createElement("div", { style: { display: 'flex', alignItems: 'center', gap: 'var(--space-3)' } },
                                                React.createElement(Checkbox, null),
                                                "Danilo Sousa")),
                                        React.createElement(Table.Cell, null, "danilo@example.com"),
                                        React.createElement(Table.Cell, null,
                                            React.createElement(Badge, { color: "green" }, "Developer"))),
                                    React.createElement(Table.Row, null,
                                        React.createElement(Table.RowHeaderCell, null,
                                            React.createElement("div", { style: { display: 'flex', alignItems: 'center', gap: 'var(--space-3)' } },
                                                React.createElement(Checkbox, null),
                                                "Zahra Ambessa")),
                                        React.createElement(Table.Cell, null, "zahra@example.com"),
                                        React.createElement(Table.Cell, null,
                                            React.createElement(Badge, { color: "amber" }, "Admin"))),
                                    React.createElement(Table.Row, null,
                                        React.createElement(Table.RowHeaderCell, null,
                                            React.createElement("div", { style: { display: 'flex', alignItems: 'center', gap: 'var(--space-3)' } },
                                                React.createElement(Checkbox, null),
                                                "Jasper Eriksson")),
                                        React.createElement(Table.Cell, null, "jasper@example.com"),
                                        React.createElement(Table.Cell, null,
                                            React.createElement(Badge, { color: "green" }, "Developer")))))))),
                React.createElement("div", { style: { display: 'flex', gap: 'var(--space-3)', justifyContent: 'flex-end' } },
                    React.createElement(Sheet.Close, null,
                        React.createElement(Button, { variant: "soft", color: "gray" }, "Close"))))))),
};
export const ScrollableContent = {
    name: 'Scrollable content',
    render: (args) => (React.createElement(Sheet.Root, null,
        React.createElement(Sheet.Trigger, null,
            React.createElement(Button, null, "View users")),
        React.createElement(Sheet.Content, { ...args },
            React.createElement(Sheet.Header, null,
                React.createElement(Sheet.Title, null, "Users"),
                React.createElement(Sheet.Description, null, "The following users have access to this project.")),
            React.createElement(Sheet.Body, null,
                React.createElement(Inset, { side: "all", px: "current", pt: "current", style: {
                        flex: 1,
                        overflowY: 'auto',
                        borderTop: '1px solid var(--gray-a4)',
                        borderBottom: '1px solid var(--gray-a4)',
                    } },
                    React.createElement(Text, { as: "p", size: "2", style: { marginBottom: 'var(--space-4)' } }, "Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes"),
                    React.createElement(Text, { as: "p", size: "2", style: { marginBottom: 'var(--space-4)' } }, "Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes"),
                    React.createElement(Text, { as: "p", size: "2", style: { marginBottom: 'var(--space-4)' } }, "Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes to your profile. Make changes")))))),
};
export const Controlled = {
    name: 'Controlled (non dismissable)',
    render: (args) => {
        const [open, setOpen] = React.useState(false);
        return (React.createElement(Sheet.Root, { dismissible: false, open: open },
            React.createElement(Sheet.Trigger, { onClick: () => setOpen(true) },
                React.createElement(Button, null, "Delete account")),
            React.createElement(Sheet.Content, { ...args },
                React.createElement(Sheet.Header, null,
                    React.createElement(Sheet.Title, null, "Delete account"),
                    React.createElement(Sheet.Description, { color: "gray" }, "Are you sure you want to delete your account?")),
                React.createElement(Sheet.Body, null,
                    React.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-3)', width: '100%' } },
                        React.createElement(Sheet.Close, { onClick: () => setOpen(false) },
                            React.createElement(Button, { variant: "surface", style: { width: '100%' }, size: "3" }, "Cancel")),
                        React.createElement(Sheet.Close, { onClick: () => setOpen(false) },
                            React.createElement(Button, { variant: "classic", color: "danger", style: { width: '100%' }, size: "3" }, "Delete account")))))));
    },
};
// TODO: add support for snapPoints
// export const SnapPoints: Story = {
//   name: 'Snap points',
//   render: ({ children, ...args }) => {
//     const snapPoints = ['120px', '240px', 1];
//     const [snap, setSnap] = React.useState<string | number | null>('420px');
//     return (
//       <Sheet.Root
//       snapPoints={snapPoints}
//       activeSnapPoint={snap}
//       setActiveSnapPoint={(a) => console.log({ a })}
//       >
//         <Sheet.Trigger>
//           <Button>Open Sheet</Button>
//         </Sheet.Trigger>
//         <Sheet.Content {...args}>
//           <Sheet.Header>
//             <Sheet.Title>Delete account</Sheet.Title>
//             <Sheet.Description>
//               Are you sure you want to delete your account?
//             </Sheet.Description>
//           </Sheet.Header>
//           <Sheet.Body>
//             <div style={{ height: 120, width: '100%', background: 'red' }} />
//             <div style={{ height: 120, width: '100%', background: 'blue' }} />
//             <div style={{ height: 260, width: '100%', background: 'green' }} />
//           </Sheet.Body>
//         </Sheet.Content>
//       </Sheet.Root>
//     );
//   },
// };
