// frosted-ui/dist/esm/components/text-field/text-field.stories.js
import { MagnifyingGlass12, MagnifyingGlass16, MagnifyingGlass20, ThreeDotsHorizontal16, ThreeDotsHorizontal20, } from '@frosted-ui/icons';
import React from 'react';
import { Button, IconButton, TextField, textFieldPropDefs } from '..';
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Controls/TextField',
    component: TextField.Root,
    args: {
        size: textFieldPropDefs.size.default,
        variant: textFieldPropDefs.variant.default,
        color: textFieldPropDefs.color.default,
    },
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
};
export default meta;
// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
export const Default = {
    render: (args) => (React.createElement("div", { style: { width: 300 } },
        React.createElement(TextField.Root, { ...args },
            React.createElement(TextField.Slot, null,
                React.createElement(MagnifyingGlass16, null)),
            React.createElement(TextField.Input, { placeholder: "Search the docs\u2026" })))),
};
export const Size = {
    render: (args) => (React.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-3)', maxWidth: 400 } },
        React.createElement("div", { style: { display: 'flex', alignItems: 'center', gap: 'var(--space-2)' } },
            React.createElement(TextField.Root, { ...args },
                React.createElement(TextField.Slot, null,
                    React.createElement(MagnifyingGlass12, null)),
                React.createElement(TextField.Input, { placeholder: "Search the docs\u2026", size: "1" })),
            React.createElement(Button, { size: "1", variant: args.variant }, "Search")),
        React.createElement("div", { style: { display: 'flex', alignItems: 'center', gap: 'var(--space-2)' } },
            React.createElement(TextField.Root, { ...args },
                React.createElement(TextField.Slot, null,
                    React.createElement(MagnifyingGlass16, null)),
                React.createElement(TextField.Input, { placeholder: "Search the docs\u2026", size: "2" }),
                React.createElement(TextField.Slot, null,
                    React.createElement(IconButton, { color: "gray", size: "1", variant: "ghost" },
                        React.createElement(ThreeDotsHorizontal16, null)))),
            React.createElement(Button, { size: "2", variant: args.variant }, "Search")),
        React.createElement("div", { style: { display: 'flex', alignItems: 'center', gap: 'var(--space-2)' } },
            React.createElement(TextField.Root, { ...args },
                React.createElement(TextField.Slot, null,
                    React.createElement(MagnifyingGlass20, null)),
                React.createElement(TextField.Input, { placeholder: "Search the docs\u2026", size: "3" }),
                React.createElement(TextField.Slot, { style: { paddingRight: 'var(--space-3)' } },
                    React.createElement(IconButton, { color: "gray", size: "2", variant: "ghost" },
                        React.createElement(ThreeDotsHorizontal20, null)))),
            React.createElement(Button, { size: "3", variant: args.variant }, "Search")))),
};
export const Variant = {
    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    render: ({ ref, ...args }) => (React.createElement("div", { style: { display: 'flex', flexDirection: 'row', gap: 'var(--space-5)' } },
        React.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-3)', maxWidth: 400 } },
            React.createElement(TextField.Input, { placeholder: "Search the docs\u2026", ...args, variant: "surface" }),
            React.createElement(TextField.Input, { placeholder: "Search the docs\u2026", ...args, variant: "soft" })),
        React.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-3)', maxWidth: 400 } },
            React.createElement(TextField.Input, { disabled: true, placeholder: "Search the docs\u2026", ...args, variant: "surface" }),
            React.createElement(TextField.Input, { disabled: true, placeholder: "Search the docs\u2026", ...args, variant: "soft" })))),
};
export const Color = {
    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    render: ({ ref, ...args }) => (React.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-3)', maxWidth: 400 } },
        React.createElement(TextField.Input, { ...args, placeholder: "Search the docs\u2026", color: "indigo" }),
        React.createElement(TextField.Input, { ...args, placeholder: "Search the docs\u2026", color: "green" }),
        React.createElement(TextField.Input, { ...args, placeholder: "Search the docs\u2026", color: "red" }))),
};
