// frosted-ui/dist/cjs/components/switch/switch.stories.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Alignment = exports.HighContrast = exports.Color = exports.Size = exports.Default = void 0;
const tslib_1 = require("tslib");
const react_1 = tslib_1.__importDefault(require("react"));
const __1 = require("..");
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Controls/Switch',
    component: __1.Switch,
    args: {
        disabled: false,
    },
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
};
exports.default = meta;
// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
exports.Default = {
    args: {
        size: __1.switchPropDefs.size.default,
        color: __1.switchPropDefs.color.default,
        highContrast: __1.switchPropDefs.highContrast.default,
    },
};
exports.Size = {
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', alignItems: 'center', gap: 'var(--space-2)' } },
        react_1.default.createElement(__1.Switch, { ...args, size: "1", defaultChecked: true }),
        react_1.default.createElement(__1.Switch, { ...args, size: "2", defaultChecked: true }),
        react_1.default.createElement(__1.Switch, { ...args, size: "3", defaultChecked: true }))),
};
exports.Color = {
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', alignItems: 'center', gap: 'var(--space-2)' } },
        react_1.default.createElement(__1.Switch, { ...args, color: "indigo", defaultChecked: true }),
        react_1.default.createElement(__1.Switch, { ...args, color: "cyan", defaultChecked: true }),
        react_1.default.createElement(__1.Switch, { ...args, color: "orange", defaultChecked: true }),
        react_1.default.createElement(__1.Switch, { ...args, color: "crimson", defaultChecked: true }))),
};
exports.HighContrast = {
    name: 'High Contrast',
    render: (args) => (react_1.default.createElement("div", { style: {
            display: 'inline-grid',
            gridTemplateRows: 'repeat(2, 1fr)',
            gap: 'var(--space-2)',
            gridAutoFlow: 'column',
        } },
        react_1.default.createElement(__1.Switch, { ...args, color: "indigo", defaultChecked: true }),
        react_1.default.createElement(__1.Switch, { ...args, color: "indigo", defaultChecked: true, highContrast: true }),
        react_1.default.createElement(__1.Switch, { ...args, color: "cyan", defaultChecked: true }),
        react_1.default.createElement(__1.Switch, { ...args, color: "cyan", defaultChecked: true, highContrast: true }),
        react_1.default.createElement(__1.Switch, { ...args, color: "orange", defaultChecked: true }),
        react_1.default.createElement(__1.Switch, { ...args, color: "orange", defaultChecked: true, highContrast: true }),
        react_1.default.createElement(__1.Switch, { ...args, color: "crimson", defaultChecked: true }),
        react_1.default.createElement(__1.Switch, { ...args, color: "crimson", defaultChecked: true, highContrast: true }))),
};
exports.Alignment = {
    name: 'Alignment with text',
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-3)' } },
        react_1.default.createElement(__1.Text, { as: "label", size: "2" },
            react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-2)' } },
                react_1.default.createElement(__1.Switch, { ...args, size: "1", defaultChecked: true }),
                " Sync settings")),
        react_1.default.createElement(__1.Text, { as: "label", size: "3" },
            react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-2)' } },
                react_1.default.createElement(__1.Switch, { ...args, size: "2", defaultChecked: true }),
                " Sync settings")),
        react_1.default.createElement(__1.Text, { as: "label", size: "4" },
            react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-2)' } },
                react_1.default.createElement(__1.Switch, { ...args, size: "3", defaultChecked: true }),
                " Sync settings")))),
};
