// frosted-ui/dist/esm/components/quote/quote.stories.js
import React from 'react';
import { Quote, Text } from '..';
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Typography/Quote',
    component: Quote,
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
};
export default meta;
// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
export const Default = {
    render: (args) => (React.createElement(Text, null,
        "His famous quote, ",
        React.createElement(Quote, { ...args }, "Styles come and go. Good design is a language, not a style"),
        ", elegantly summs up Massimo\u2019s philosophy of design.")),
};
