// frosted-ui/dist/esm/components/accordion/accordion.stories.js
import React from 'react';
import { Accordion } from '..';
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Layout/Accordion',
    component: Accordion.Root,
    args: {},
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
};
export default meta;
// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
export const Single = {
    args: {
        defaultValue: 'item-1',
        type: 'single',
        collapsible: false,
    },
    render: (args) => (React.createElement("div", null,
        React.createElement(Accordion.Root
        // collapsible
        , { 
            // collapsible
            style: { width: 600 }, ...args },
            React.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-4)' } },
                React.createElement(Accordion.Item, { value: "item-1" },
                    React.createElement(Accordion.Trigger, null, "Is it accessible?"),
                    React.createElement(Accordion.Content, null, "Yes. It adheres to the WAI-ARIA design pattern.")),
                React.createElement(Accordion.Item, { value: "item-2" },
                    React.createElement(Accordion.Trigger, null, "Is it unstyled?"),
                    React.createElement(Accordion.Content, null, "Yes. It's unstyled by default, giving you freedom over the look and feel.")),
                React.createElement(Accordion.Item, { value: "item-3" },
                    React.createElement(Accordion.Trigger, null, "Can it be animated?"),
                    React.createElement(Accordion.Content, null, "Yes! You can animate the Accordion with CSS or JavaScript.")))))),
};
export const Multiple = {
    args: {
        defaultValue: ['item-1', 'item-2'],
        type: 'multiple',
    },
    render: (args) => (React.createElement("div", null,
        React.createElement(Accordion.Root, { style: { width: 600 }, ...args },
            React.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-4)' } },
                React.createElement(Accordion.Item, { value: "item-1" },
                    React.createElement(Accordion.Trigger, null, "Is it accessible?"),
                    React.createElement(Accordion.Content, null, "Yes. It adheres to the WAI-ARIA design pattern.")),
                React.createElement(Accordion.Item, { value: "item-2" },
                    React.createElement(Accordion.Trigger, null, "Is it unstyled?"),
                    React.createElement(Accordion.Content, null, "Yes. It's unstyled by default, giving you freedom over the look and feel.")),
                React.createElement(Accordion.Item, { value: "item-3" },
                    React.createElement(Accordion.Trigger, null, "Can it be animated?"),
                    React.createElement(Accordion.Content, null, "Yes! You can animate the Accordion with CSS or JavaScript.")))))),
};
