// frosted-ui/dist/cjs/components/skeleton/skeleton.props.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.skeletonTextPropDefs = exports.skeletonRectPropDefs = exports.skeletonAvatarPropDefs = void 0;
const helpers_1 = require("../../helpers");
const avatar_props_1 = require("../avatar/avatar.props");
const text_props_1 = require("../text/text.props");
const sizes = ['1', '2', '3', '4', '5', '6', '7', '8', '9'];
const skeletonAvatarPropDefs = {
    size: avatar_props_1.avatarPropDefs.size,
    color: { ...helpers_1.colorProp, default: 'gray' },
    highContrast: helpers_1.highContrastProp,
};
exports.skeletonAvatarPropDefs = skeletonAvatarPropDefs;
const skeletonTextPropDefs = {
    size: { ...text_props_1.textPropDefs.size, default: '3' },
    color: { ...helpers_1.colorProp, default: 'gray' },
    highContrast: helpers_1.highContrastProp,
};
exports.skeletonTextPropDefs = skeletonTextPropDefs;
const skeletonRectPropDefs = {
    color: { ...helpers_1.colorProp, default: 'gray' },
    highContrast: helpers_1.highContrastProp,
};
exports.skeletonRectPropDefs = skeletonRectPropDefs;
