// frosted-ui/dist/cjs/components/skeleton/skeleton.stories.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ComposedColor = exports.Composed = exports.RectHighContrast = exports.RectColors = exports.RectDefault = exports.TextHighContrast = exports.TextColors = exports.TextSizes = exports.TextDefault = exports.AvatarHighContrast = exports.AvatarColors = exports.AvatarSizes = exports.AvatarDefault = void 0;
const tslib_1 = require("tslib");
const react_1 = tslib_1.__importDefault(require("react"));
const __1 = require("..");
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Components/Skeleton',
    component: __1.Skeleton.Avatar,
    args: {
        highContrast: false,
    },
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
};
exports.default = meta;
// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
// Avatar
exports.AvatarDefault = {
    name: 'Avatar',
    args: {
        size: __1.skeletonAvatarPropDefs.size.default,
        color: __1.skeletonAvatarPropDefs.color.default,
    },
    render: (args) => react_1.default.createElement(__1.Skeleton.Avatar, { ...args }),
};
exports.AvatarSizes = {
    name: 'Avatar sizes',
    args: {
        size: __1.skeletonAvatarPropDefs.size.default,
        color: __1.skeletonAvatarPropDefs.color.default,
    },
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'row', gap: 'var(--space-3)' } },
        react_1.default.createElement(__1.Skeleton.Avatar, { ...args, size: "1" }),
        react_1.default.createElement(__1.Skeleton.Avatar, { ...args, size: "2" }),
        react_1.default.createElement(__1.Skeleton.Avatar, { ...args, size: "3" }),
        react_1.default.createElement(__1.Skeleton.Avatar, { ...args, size: "4" }),
        react_1.default.createElement(__1.Skeleton.Avatar, { ...args, size: "5" }),
        react_1.default.createElement(__1.Skeleton.Avatar, { ...args, size: "6" }),
        react_1.default.createElement(__1.Skeleton.Avatar, { ...args, size: "7" }),
        react_1.default.createElement(__1.Skeleton.Avatar, { ...args, size: "8" }),
        react_1.default.createElement(__1.Skeleton.Avatar, { ...args, size: "9" }))),
};
exports.AvatarColors = {
    name: 'Avatar colors',
    args: {
        size: __1.skeletonAvatarPropDefs.size.default,
        color: __1.skeletonAvatarPropDefs.color.default,
    },
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'row', gap: 'var(--space-3)' } },
        react_1.default.createElement(__1.Skeleton.Avatar, { ...args, color: "indigo" }),
        react_1.default.createElement(__1.Skeleton.Avatar, { ...args, color: "cyan" }),
        react_1.default.createElement(__1.Skeleton.Avatar, { ...args, color: "orange" }),
        react_1.default.createElement(__1.Skeleton.Avatar, { ...args, color: "crimson" }))),
};
exports.AvatarHighContrast = {
    name: 'Avatar high contrast',
    args: {
        size: __1.skeletonAvatarPropDefs.size.default,
        color: __1.skeletonAvatarPropDefs.color.default,
    },
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-2)' } },
        react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-2)' } },
            react_1.default.createElement(__1.Skeleton.Avatar, { ...args, color: "gray", highContrast: false }),
            react_1.default.createElement(__1.Skeleton.Avatar, { ...args, color: "indigo", highContrast: false }),
            react_1.default.createElement(__1.Skeleton.Avatar, { ...args, color: "cyan", highContrast: false }),
            react_1.default.createElement(__1.Skeleton.Avatar, { ...args, color: "orange", highContrast: false }),
            react_1.default.createElement(__1.Skeleton.Avatar, { ...args, color: "crimson", highContrast: false })),
        react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-2)' } },
            react_1.default.createElement(__1.Skeleton.Avatar, { ...args, color: "gray", highContrast: true }),
            react_1.default.createElement(__1.Skeleton.Avatar, { ...args, color: "indigo", highContrast: true }),
            react_1.default.createElement(__1.Skeleton.Avatar, { ...args, color: "cyan", highContrast: true }),
            react_1.default.createElement(__1.Skeleton.Avatar, { ...args, color: "orange", highContrast: true }),
            react_1.default.createElement(__1.Skeleton.Avatar, { ...args, color: "crimson", highContrast: true })))),
};
// Text
exports.TextDefault = {
    name: 'Text',
    args: {
        size: __1.skeletonTextPropDefs.size.default,
        color: __1.skeletonTextPropDefs.color.default,
    },
    render: (args) => react_1.default.createElement(__1.Skeleton.Text, { ...args, style: { width: 400 } }),
};
exports.TextSizes = {
    name: 'Text sizes',
    args: {
        size: __1.skeletonTextPropDefs.size.default,
        color: __1.skeletonTextPropDefs.color.default,
    },
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-3)' } },
        react_1.default.createElement(__1.Skeleton.Text, { ...args, size: "0", style: { width: 400 } }),
        react_1.default.createElement(__1.Skeleton.Text, { ...args, size: "1", style: { width: 400 } }),
        react_1.default.createElement(__1.Skeleton.Text, { ...args, size: "2", style: { width: 400 } }),
        react_1.default.createElement(__1.Skeleton.Text, { ...args, size: "3", style: { width: 400 } }),
        react_1.default.createElement(__1.Skeleton.Text, { ...args, size: "4", style: { width: 400 } }),
        react_1.default.createElement(__1.Skeleton.Text, { ...args, size: "5", style: { width: 400 } }),
        react_1.default.createElement(__1.Skeleton.Text, { ...args, size: "6", style: { width: 400 } }),
        react_1.default.createElement(__1.Skeleton.Text, { ...args, size: "7", style: { width: 400 } }),
        react_1.default.createElement(__1.Skeleton.Text, { ...args, size: "8", style: { width: 400 } }),
        react_1.default.createElement(__1.Skeleton.Text, { ...args, size: "9", style: { width: 400 } }))),
};
exports.TextColors = {
    name: 'Text colors',
    args: {
        size: __1.skeletonTextPropDefs.size.default,
        color: __1.skeletonTextPropDefs.color.default,
    },
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-3)' } },
        react_1.default.createElement(__1.Skeleton.Text, { ...args, color: "indigo", style: { width: 400 } }),
        react_1.default.createElement(__1.Skeleton.Text, { ...args, color: "cyan", style: { width: 400 } }),
        react_1.default.createElement(__1.Skeleton.Text, { ...args, color: "orange", style: { width: 400 } }),
        react_1.default.createElement(__1.Skeleton.Text, { ...args, color: "crimson", style: { width: 400 } }))),
};
exports.TextHighContrast = {
    name: 'Text high contrast',
    args: {
        size: __1.skeletonTextPropDefs.size.default,
        color: __1.skeletonTextPropDefs.color.default,
    },
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'row', gap: 'var(--space-2)' } },
        react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-2)' } },
            react_1.default.createElement(__1.Skeleton.Text, { ...args, color: "gray", highContrast: false, style: { width: 400 } }),
            react_1.default.createElement(__1.Skeleton.Text, { ...args, color: "indigo", highContrast: false, style: { width: 400 } }),
            react_1.default.createElement(__1.Skeleton.Text, { ...args, color: "cyan", highContrast: false, style: { width: 400 } }),
            react_1.default.createElement(__1.Skeleton.Text, { ...args, color: "orange", highContrast: false, style: { width: 400 } }),
            react_1.default.createElement(__1.Skeleton.Text, { ...args, color: "crimson", highContrast: false, style: { width: 400 } })),
        react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-2)' } },
            react_1.default.createElement(__1.Skeleton.Text, { ...args, color: "gray", highContrast: true, style: { width: 400 } }),
            react_1.default.createElement(__1.Skeleton.Text, { ...args, color: "indigo", highContrast: true, style: { width: 400 } }),
            react_1.default.createElement(__1.Skeleton.Text, { ...args, color: "cyan", highContrast: true, style: { width: 400 } }),
            react_1.default.createElement(__1.Skeleton.Text, { ...args, color: "orange", highContrast: true, style: { width: 400 } }),
            react_1.default.createElement(__1.Skeleton.Text, { ...args, color: "crimson", highContrast: true, style: { width: 400 } })))),
};
// Rect
exports.RectDefault = {
    name: 'Rect',
    args: {
        color: __1.skeletonRectPropDefs.color.default,
    },
    render: (args) => react_1.default.createElement(__1.Skeleton.Rect, { style: { width: 140, height: 140 }, ...args }),
};
exports.RectColors = {
    name: 'Rect colors',
    args: {
        color: __1.skeletonRectPropDefs.color.default,
    },
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'row', gap: 'var(--space-3)' } },
        react_1.default.createElement(__1.Skeleton.Rect, { style: { width: 140, height: 140 }, ...args, color: "indigo" }),
        react_1.default.createElement(__1.Skeleton.Rect, { style: { width: 140, height: 140 }, ...args, color: "cyan" }),
        react_1.default.createElement(__1.Skeleton.Rect, { style: { width: 140, height: 140 }, ...args, color: "orange" }),
        react_1.default.createElement(__1.Skeleton.Rect, { style: { width: 140, height: 140 }, ...args, color: "crimson" }))),
};
exports.RectHighContrast = {
    name: 'Rect high contrast',
    args: {
        color: __1.skeletonRectPropDefs.color.default,
    },
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-2)' } },
        react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-2)' } },
            react_1.default.createElement(__1.Skeleton.Rect, { style: { width: 140, height: 140 }, ...args, color: "gray", highContrast: false }),
            react_1.default.createElement(__1.Skeleton.Rect, { style: { width: 140, height: 140 }, ...args, color: "indigo", highContrast: false }),
            react_1.default.createElement(__1.Skeleton.Rect, { style: { width: 140, height: 140 }, ...args, color: "cyan", highContrast: false }),
            react_1.default.createElement(__1.Skeleton.Rect, { style: { width: 140, height: 140 }, ...args, color: "orange", highContrast: false }),
            react_1.default.createElement(__1.Skeleton.Rect, { style: { width: 140, height: 140 }, ...args, color: "crimson", highContrast: false })),
        react_1.default.createElement("div", { style: { display: 'flex', gap: 'var(--space-2)' } },
            react_1.default.createElement(__1.Skeleton.Rect, { style: { width: 140, height: 140 }, ...args, color: "gray", highContrast: true }),
            react_1.default.createElement(__1.Skeleton.Rect, { style: { width: 140, height: 140 }, ...args, color: "indigo", highContrast: true }),
            react_1.default.createElement(__1.Skeleton.Rect, { style: { width: 140, height: 140 }, ...args, color: "cyan", highContrast: true }),
            react_1.default.createElement(__1.Skeleton.Rect, { style: { width: 140, height: 140 }, ...args, color: "orange", highContrast: true }),
            react_1.default.createElement(__1.Skeleton.Rect, { style: { width: 140, height: 140 }, ...args, color: "crimson", highContrast: true })))),
};
exports.Composed = {
    args: {
        color: __1.skeletonTextPropDefs.color.default,
        highContrast: __1.skeletonTextPropDefs.highContrast.default,
    },
    render: (args) => {
        const [show, setShow] = react_1.default.useState(false);
        return (react_1.default.createElement(__1.Card, { variant: "surface", size: "3", onMouseOver: () => setShow(true), onMouseOut: () => setShow(false), style: { minWidth: 400 } },
            react_1.default.createElement("div", { style: {
                    objectFit: 'cover',
                    width: 400,
                    aspectRatio: '2 / 1',
                    borderRadius: 'var(--radius-2)',
                    overflow: 'hidden',
                    display: 'flex',
                } }, show ? (react_1.default.createElement("img", { className: "Image", src: "https://images.unsplash.com/photo-1535025183041-0991a977e25b?w=300&dpr=2&q=80", alt: "Landscape photograph by Tobias Tullius", style: {
                    objectFit: 'cover',
                    display: 'inline-block',
                    verticalAlign: 'center',
                    flex: 1,
                } })) : (react_1.default.createElement(__1.Skeleton.Rect, { highContrast: args.highContrast, color: args.color, style: {
                    flex: 1,
                } }))),
            react_1.default.createElement("div", { style: {
                    display: 'flex',
                    flexDirection: 'row',
                    gap: 'var(--space-4)',
                    alignItems: 'center',
                    marginTop: 'var(--space-4)',
                } },
                show ? (react_1.default.createElement(__1.Avatar, { highContrast: args.highContrast, color: args.color, fallback: "AB", size: "6" })) : (react_1.default.createElement(__1.Skeleton.Avatar, { ...args, size: "6" })),
                react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-2)' } }, show ? (react_1.default.createElement(react_1.default.Fragment, null,
                    react_1.default.createElement(__1.Text, { highContrast: args.highContrast, size: "6" }, "Artur"),
                    react_1.default.createElement(__1.Text, { highContrast: args.highContrast, size: "2" }, "Design Engineer at Whop"),
                    react_1.default.createElement(__1.Text, { highContrast: args.highContrast, size: "2" }, "Professional button polisher."))) : (react_1.default.createElement(react_1.default.Fragment, null,
                    react_1.default.createElement(__1.Skeleton.Text, { ...args, size: "6", style: { width: 150 } }),
                    react_1.default.createElement(__1.Skeleton.Text, { ...args, size: "2", style: { width: 250 } }),
                    react_1.default.createElement(__1.Skeleton.Text, { ...args, size: "2", style: { width: 200 } })))))));
    },
};
exports.ComposedColor = {
    name: 'Composed color',
    args: {
        color: 'iris',
        highContrast: __1.skeletonTextPropDefs.highContrast.default,
    },
    render: (args) => {
        const [show, setShow] = react_1.default.useState(false);
        return (react_1.default.createElement("div", { style: {
                minWidth: 400,
                boxSizing: 'border-box',
                padding: 24,
                background: `var(--${args.color}-a2)`,
                border: `1px solid var(--${args.color}-a6)`,
                borderRadius: 'var(--radius-5)',
            }, onMouseOver: () => setShow(true), onMouseOut: () => setShow(false) },
            react_1.default.createElement("div", { style: {
                    objectFit: 'cover',
                    width: 400,
                    aspectRatio: '2 / 1',
                    borderRadius: 'var(--radius-2)',
                    overflow: 'hidden',
                    display: 'flex',
                } }, show ? (react_1.default.createElement("img", { className: "Image", src: "https://images.unsplash.com/photo-1535025183041-0991a977e25b?w=300&dpr=2&q=80", alt: "Landscape photograph by Tobias Tullius", style: {
                    objectFit: 'cover',
                    display: 'inline-block',
                    verticalAlign: 'center',
                    flex: 1,
                } })) : (react_1.default.createElement(__1.Skeleton.Rect, { highContrast: args.highContrast, color: args.color, style: {
                    flex: 1,
                } }))),
            react_1.default.createElement("div", { style: {
                    display: 'flex',
                    flexDirection: 'row',
                    gap: 'var(--space-4)',
                    alignItems: 'center',
                    marginTop: 'var(--space-4)',
                } },
                show ? (react_1.default.createElement(__1.Avatar, { highContrast: args.highContrast, color: args.color, fallback: "AB", size: "6" })) : (react_1.default.createElement(__1.Skeleton.Avatar, { ...args, size: "6" })),
                react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-2)' } }, show ? (react_1.default.createElement(react_1.default.Fragment, null,
                    react_1.default.createElement(__1.Text, { color: args.color, highContrast: args.highContrast, size: "6" }, "Artur"),
                    react_1.default.createElement(__1.Text, { color: args.color, highContrast: args.highContrast, size: "2" }, "Design Engineer at Whop"),
                    react_1.default.createElement(__1.Text, { color: args.color, highContrast: args.highContrast, size: "2" }, "Professional button polisher."))) : (react_1.default.createElement(react_1.default.Fragment, null,
                    react_1.default.createElement(__1.Skeleton.Text, { ...args, size: "6", style: { width: 150 } }),
                    react_1.default.createElement(__1.Skeleton.Text, { ...args, size: "2", style: { width: 250 } }),
                    react_1.default.createElement(__1.Skeleton.Text, { ...args, size: "2", style: { width: 200 } })))))));
    },
};
