// frosted-ui/dist/esm/components/segmented-control-nav/segmented-control-nav.stories.js
import React from 'react';
import { SegmentedControlNav } from '..';
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Components/SegmentedControlNav',
    component: SegmentedControlNav.Root,
    args: {},
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
};
export default meta;
// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
export const Default = {
    render: (args) => (React.createElement("div", { style: { width: 600 } },
        React.createElement(SegmentedControlNav.Root, { ...args },
            React.createElement(SegmentedControlNav.Link, { active: true, href: "#" }, "Account"),
            React.createElement(SegmentedControlNav.Link, { href: "#" }, "Documents"),
            React.createElement(SegmentedControlNav.Link, { href: "#" }, "Settings")))),
};
