// frosted-ui/dist/cjs/components/segmented-control/segmented-control.stories.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.InDrawer = exports.Default = void 0;
const tslib_1 = require("tslib");
const react_1 = tslib_1.__importDefault(require("react"));
const __1 = require("..");
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Components/SegmentedControl',
    component: __1.SegmentedControl.List,
    args: {},
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
};
exports.default = meta;
// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
exports.Default = {
    render: (args) => (react_1.default.createElement("div", { style: { width: 'auto' } },
        react_1.default.createElement(__1.SegmentedControl.Root, { defaultValue: "account" },
            react_1.default.createElement(__1.SegmentedControl.List, { ...args },
                react_1.default.createElement(__1.SegmentedControl.Trigger, { value: "account" }, "Account"),
                react_1.default.createElement(__1.SegmentedControl.Trigger, { value: "documents" }, "Documents"),
                react_1.default.createElement(__1.SegmentedControl.Trigger, { value: "settings" }, "Settings")),
            react_1.default.createElement("div", { style: { padding: '12px 16px 8px 16px' } },
                react_1.default.createElement(__1.SegmentedControl.Content, { value: "account" },
                    react_1.default.createElement(__1.Text, { size: "2" }, "Your account.")),
                react_1.default.createElement(__1.SegmentedControl.Content, { value: "documents" },
                    react_1.default.createElement(__1.Text, { size: "2" }, "Your documents.")),
                react_1.default.createElement(__1.SegmentedControl.Content, { value: "settings" },
                    react_1.default.createElement(__1.Text, { size: "2" }, "Your profile.")))))),
};
exports.InDrawer = {
    name: 'In a drawer',
    render: (args) => (react_1.default.createElement(__1.Drawer.Root, { defaultOpen: true },
        react_1.default.createElement(__1.Drawer.Trigger, null,
            react_1.default.createElement(__1.Button, null, "View users")),
        react_1.default.createElement(__1.Drawer.Content, { style: { width: 400 } },
            react_1.default.createElement(__1.Drawer.Header, null,
                react_1.default.createElement(__1.Drawer.Title, null, "Users")),
            react_1.default.createElement(__1.Drawer.Body, null,
                react_1.default.createElement(__1.SegmentedControl.Root, { defaultValue: "account" },
                    react_1.default.createElement(__1.SegmentedControl.List, { ...args },
                        react_1.default.createElement(__1.SegmentedControl.Trigger, { value: "account" }, "Account"),
                        react_1.default.createElement(__1.SegmentedControl.Trigger, { value: "documents" }, "Documents"),
                        react_1.default.createElement(__1.SegmentedControl.Trigger, { value: "settings" }, "Settings")),
                    react_1.default.createElement("div", { style: { padding: '12px 16px 8px 16px' } },
                        react_1.default.createElement(__1.SegmentedControl.Content, { value: "account" },
                            react_1.default.createElement(__1.Text, { size: "2" }, "Make changes to your account.")),
                        react_1.default.createElement(__1.SegmentedControl.Content, { value: "documents" },
                            react_1.default.createElement(__1.Text, { size: "2" }, "Access and update your documents.")),
                        react_1.default.createElement(__1.SegmentedControl.Content, { value: "settings" },
                            react_1.default.createElement(__1.Text, { size: "2" }, "Edit your profile.")))))))),
};
