// frosted-ui/dist/cjs/components/drawer/drawer.js
"use strict";
'use client';
Object.defineProperty(exports, "__esModule", { value: true });
exports.Trigger = exports.Title = exports.StickyFooter = exports.Root = exports.Header = exports.Content = exports.Close = exports.Body = void 0;
const tslib_1 = require("tslib");
const classnames_1 = tslib_1.__importDefault(require("classnames"));
const radix_ui_1 = require("radix-ui");
const React = tslib_1.__importStar(require("react"));
const theme_1 = require("../../theme");
const heading_1 = require("../heading");
const DrawerRoot = (props) => React.createElement(radix_ui_1.Dialog.Root, { ...props, modal: true });
exports.Root = DrawerRoot;
DrawerRoot.displayName = 'DrawerRoot';
const DrawerTrigger = (props) => React.createElement(radix_ui_1.Dialog.Trigger, { ...props, asChild: true });
exports.Trigger = DrawerTrigger;
DrawerTrigger.displayName = 'DrawerTrigger';
const DrawerContent = (props) => {
    const { className, forceMount, container, ...contentProps } = props;
    return (React.createElement(radix_ui_1.Dialog.Portal, { container: container, forceMount: forceMount },
        React.createElement(theme_1.Theme, { asChild: true },
            React.createElement(radix_ui_1.Dialog.Overlay, { className: "fui-DialogOverlay" },
                React.createElement(radix_ui_1.Dialog.Content, { ...contentProps, "aria-describedby": undefined, className: (0, classnames_1.default)('fui-DrawerContent', className) })))));
};
exports.Content = DrawerContent;
DrawerContent.displayName = 'DrawerContent';
const DrawerTitle = (props) => (React.createElement(radix_ui_1.Dialog.Title, { asChild: true },
    React.createElement(heading_1.Heading, { size: "4", weight: "semi-bold", ...props })));
exports.Title = DrawerTitle;
DrawerTitle.displayName = 'DrawerTitle';
const DrawerClose = (props) => React.createElement(radix_ui_1.Dialog.Close, { ...props, asChild: true });
exports.Close = DrawerClose;
DrawerClose.displayName = 'DrawerClose';
const DrawerStickyFooter = ({ children, className, ...props }) => (React.createElement("div", { className: (0, classnames_1.default)('fui-DrawerStickyFooter', className), ...props }, children));
exports.StickyFooter = DrawerStickyFooter;
DrawerStickyFooter.displayName = 'DrawerStickyFooter';
const DrawerHeader = ({ children, className, ...props }) => (React.createElement("div", { className: (0, classnames_1.default)('fui-DrawerHeader', className), ...props }, children));
exports.Header = DrawerHeader;
DrawerHeader.displayName = 'DrawerHeader';
const DrawerBody = ({ children, className, ...props }) => {
    const localRef = React.useRef(null);
    const contentRef = React.useRef(null);
    return (React.createElement("div", { className: "fui-BodyScrollArea", ...props, ref: localRef },
        React.createElement("div", { className: (0, classnames_1.default)('fui-DrawerBody', className), ref: contentRef }, children)));
};
exports.Body = DrawerBody;
DrawerBody.displayName = 'DrawerBody';
