// frosted-ui/dist/esm/components/segmented-control-radio-group/segmented-control-radio-group.stories.js
import { Monitor20, Moon20, Sun20 } from '@frosted-ui/icons';
import React from 'react';
import { SegmentedControlRadioGroup } from '..';
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Controls/SegmentedControlRadioGroup',
    component: SegmentedControlRadioGroup.Root,
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
};
export default meta;
// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
export const Default = {
    render: (args) => (React.createElement(SegmentedControlRadioGroup.Root, { defaultValue: "system", ...args, onValueChange: (value) => console.log('🟢 onValueChange ', value) },
        React.createElement(SegmentedControlRadioGroup.Item, { value: "system" },
            React.createElement(Monitor20, null)),
        React.createElement(SegmentedControlRadioGroup.Item, { value: "light-mode" },
            React.createElement(Sun20, null)),
        React.createElement(SegmentedControlRadioGroup.Item, { value: "dark-mode" },
            React.createElement(Moon20, null)))),
};
