// frosted-ui/dist/esm/components/circular-progress/circular-progress.stories.js
import React from 'react';
import { CircularProgress, circularProgressPropDefs } from '..';
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Components/CircularProgress',
    component: CircularProgress,
    args: {
        size: circularProgressPropDefs.size.default,
        color: circularProgressPropDefs.color.default,
        value: 40,
        max: 100,
    },
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
};
export default meta;
// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
export const Default = {
    args: {
        size: circularProgressPropDefs.size.default,
        color: circularProgressPropDefs.color.default,
    },
    render: (args) => (React.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-3)', width: 300 } },
        React.createElement(CircularProgress, { ...args }))),
};
export const Size = {
    args: {
        size: circularProgressPropDefs.size.default,
        color: circularProgressPropDefs.color.default,
    },
    render: (args) => (React.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-3)', width: 300 } }, circularProgressPropDefs.size.values.map((size) => (React.createElement(CircularProgress, { ...args, size: size, key: size }))))),
};
export const Color = {
    args: {
        size: circularProgressPropDefs.size.default,
    },
    render: (args) => (React.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-3)', width: 300 } },
        React.createElement(CircularProgress, { ...args, color: "indigo" }),
        React.createElement(CircularProgress, { ...args, color: "cyan" }),
        React.createElement(CircularProgress, { ...args, color: "orange" }),
        React.createElement(CircularProgress, { ...args, color: "crimson" }))),
};
export const HighContrast = {
    name: 'High Contrast',
    args: {},
    render: (args) => (React.createElement("div", { style: { display: 'flex', flexDirection: 'row', gap: 'var(--space-2)' } },
        React.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-3)', width: 300 } },
            React.createElement(CircularProgress, { ...args, highContrast: false })),
        React.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-3)', width: 300 } },
            React.createElement(CircularProgress, { ...args, highContrast: true })))),
};
