// frosted-ui/dist/cjs/components/slider/slider.stories.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Range = exports.HighContrast = exports.Color = exports.Size = exports.Default = void 0;
const tslib_1 = require("tslib");
const react_1 = tslib_1.__importDefault(require("react"));
const __1 = require("..");
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Controls/Slider',
    component: __1.Slider,
    args: {
        size: __1.sliderPropDefs.size.default,
        color: __1.sliderPropDefs.color.default,
        highContrast: __1.sliderPropDefs.highContrast.default,
        disabled: false,
    },
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
};
exports.default = meta;
// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
exports.Default = {
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-4)', width: 300 } },
        react_1.default.createElement(__1.Slider, { ...args, defaultValue: [50] }))),
};
exports.Size = {
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-4)', width: 300 } },
        react_1.default.createElement(__1.Slider, { ...args, defaultValue: [25], size: "1" }),
        react_1.default.createElement(__1.Slider, { ...args, defaultValue: [50], size: "2" }),
        react_1.default.createElement(__1.Slider, { ...args, defaultValue: [75], size: "3" }))),
};
exports.Color = {
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-4)', width: 300 } },
        react_1.default.createElement(__1.Slider, { ...args, defaultValue: [20], color: "indigo" }),
        react_1.default.createElement(__1.Slider, { ...args, defaultValue: [40], color: "cyan" }),
        react_1.default.createElement(__1.Slider, { ...args, defaultValue: [60], color: "orange" }),
        react_1.default.createElement(__1.Slider, { ...args, defaultValue: [80], color: "crimson" }))),
};
exports.HighContrast = {
    name: 'High Contrast',
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-4)', width: 300 } },
        react_1.default.createElement(__1.Slider, { ...args, defaultValue: [25], highContrast: false }),
        react_1.default.createElement(__1.Slider, { ...args, defaultValue: [50], highContrast: true }))),
};
exports.Range = {
    render: (args) => (react_1.default.createElement("div", { style: { display: 'flex', flexDirection: 'column', gap: 'var(--space-4)', width: 300 } },
        react_1.default.createElement(__1.Slider, { ...args, defaultValue: [25, 75] }))),
};
